<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">

<head>
  <meta charset="utf-8"/>
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta http-equiv="X-UA-Compatible" content="IE=Edge,chrome=1"/>
  <meta name="generator" content="distill" />

  <style type="text/css">
  /* Hide doc at startup (prevent jankiness while JS renders/transforms) */
  body {
    visibility: hidden;
  }
  </style>

 <!--radix_placeholder_import_source-->
 <!--/radix_placeholder_import_source-->

  <!--radix_placeholder_meta_tags-->
  <title>C084: Multi-sample comparisons of the Hayman/Pasricha (C084) ZIPT mini-bulk data set</title>
  
  
  
  <!--  https://schema.org/Article -->
  <meta property="article:published" itemprop="datePublished" content="2020-08-27"/>
  <meta property="article:created" itemprop="dateCreated" content="2020-08-27"/>
  <meta name="article:author" content="Peter Hickey"/>
  
  <!--  https://developers.facebook.com/docs/sharing/webmasters#markup -->
  <meta property="og:title" content="C084: Multi-sample comparisons of the Hayman/Pasricha (C084) ZIPT mini-bulk data set"/>
  <meta property="og:type" content="article"/>
  <meta property="og:locale" content="en_US"/>
  <meta property="og:site_name" content="C084"/>
  
  <!--  https://dev.twitter.com/cards/types/summary -->
  <meta property="twitter:card" content="summary"/>
  <meta property="twitter:title" content="C084: Multi-sample comparisons of the Hayman/Pasricha (C084) ZIPT mini-bulk data set"/>
  
  <!--/radix_placeholder_meta_tags-->
  
  <meta name="citation_reference" content="citation_title=Voom: Precision weights unlock linear model analysis tools for rna-seq read counts;citation_publication_date=2014;citation_volume=15;citation_author=C. W. Law;citation_author=Y. Chen;citation_author=W. Shi;citation_author=G. K. Smyth"/>
  <meta name="citation_reference" content="citation_title=Linear models and empirical bayes methods for assessing differential expression in microarray experiments;citation_publication_date=2004;citation_publisher=De Gruyter;citation_volume=3;citation_author=Gordon K Smyth"/>
  <!--radix_placeholder_rmarkdown_metadata-->
  
  <script type="text/json" id="radix-rmarkdown-metadata">
  {"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["title","description","author","date","editor_options","bibliography"]}},"value":[{"type":"character","attributes":{},"value":["Multi-sample comparisons of the Hayman/Pasricha (C084) ZIPT mini-bulk data set"]},{"type":"NULL"},{"type":"list","attributes":{},"value":[{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","url","affiliation","affiliation_url"]}},"value":[{"type":"character","attributes":{},"value":["Peter Hickey"]},{"type":"character","attributes":{},"value":["https://peterhickey.org"]},{"type":"character","attributes":{},"value":["WEHI SCORE"]},{"type":"character","attributes":{},"value":["https://www.wehi.edu.au/people/shalin-naik/3310/score"]}]}]},{"type":"character","attributes":{},"value":["2020-08-27"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["chunk_output_type"]}},"value":[{"type":"character","attributes":{},"value":["console"]}]},{"type":"character","attributes":{},"value":["ref.bib"]}]}
  </script>
  <!--/radix_placeholder_rmarkdown_metadata-->
  <!--radix_placeholder_navigation_in_header-->
  
  <script type="application/javascript">
  
    window.headroom_prevent_pin = false;
  
    window.document.addEventListener("DOMContentLoaded", function (event) {
  
      // initialize headroom for banner
      var header = $('header').get(0);
      var headerHeight = header.offsetHeight;
      var headroom = new Headroom(header, {
        onPin : function() {
          if (window.headroom_prevent_pin) {
            window.headroom_prevent_pin = false;
            headroom.unpin();
          }
        }
      });
      headroom.init();
      if(window.location.hash)
        headroom.unpin();
      $(header).addClass('headroom--transition');
  
      // offset scroll location for banner on hash change
      // (see: https://github.com/WickyNilliams/headroom.js/issues/38)
      window.addEventListener("hashchange", function(event) {
        window.scrollTo(0, window.pageYOffset - (headerHeight + 25));
      });
  
      // responsive menu
      $('.distill-site-header').each(function(i, val) {
        var topnav = $(this);
        var toggle = topnav.find('.nav-toggle');
        toggle.on('click', function() {
          topnav.toggleClass('responsive');
        });
      });
  
      // nav dropdowns
      $('.nav-dropbtn').click(function(e) {
        $(this).next('.nav-dropdown-content').toggleClass('nav-dropdown-active');
        $(this).parent().siblings('.nav-dropdown')
           .children('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $("body").click(function(e){
        $('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $(".nav-dropdown").click(function(e){
        e.stopPropagation();
      });
    });
  </script>
  
  <style type="text/css">
  
  /* Theme (user-documented overrideables for nav appearance) */
  
  .distill-site-nav {
    color: rgba(255, 255, 255, 0.8);
    background-color: #455a64;
    font-size: 15px;
    font-weight: 300;
  }
  
  .distill-site-nav a {
    color: inherit;
    text-decoration: none;
  }
  
  .distill-site-nav a:hover {
    color: white;
  }
  
  @media print {
    .distill-site-nav {
      display: none;
    }
  }
  
  .distill-site-header {
  
  }
  
  .distill-site-footer {
  
  }
  
  
  /* Site Header */
  
  .distill-site-header {
    width: 100%;
    box-sizing: border-box;
    z-index: 3;
  }
  
  .distill-site-header .nav-left {
    display: inline-block;
    margin-left: 8px;
  }
  
  @media screen and (max-width: 768px) {
    .distill-site-header .nav-left {
      margin-left: 0;
    }
  }
  
  
  .distill-site-header .nav-right {
    float: right;
    margin-right: 8px;
  }
  
  .distill-site-header a,
  .distill-site-header .title {
    display: inline-block;
    text-align: center;
    padding: 14px 10px 14px 10px;
  }
  
  .distill-site-header .title {
    font-size: 18px;
  }
  
  .distill-site-header .logo {
    padding: 0;
  }
  
  .distill-site-header .logo img {
    display: none;
    max-height: 20px;
    width: auto;
    margin-bottom: -4px;
  }
  
  .distill-site-header .nav-image img {
    max-height: 18px;
    width: auto;
    display: inline-block;
    margin-bottom: -3px;
  }
  
  
  
  @media screen and (min-width: 1000px) {
    .distill-site-header .logo img {
      display: inline-block;
    }
    .distill-site-header .nav-left {
      margin-left: 20px;
    }
    .distill-site-header .nav-right {
      margin-right: 20px;
    }
    .distill-site-header .title {
      padding-left: 12px;
    }
  }
  
  
  .distill-site-header .nav-toggle {
    display: none;
  }
  
  .nav-dropdown {
    display: inline-block;
    position: relative;
  }
  
  .nav-dropdown .nav-dropbtn {
    border: none;
    outline: none;
    color: rgba(255, 255, 255, 0.8);
    padding: 16px 10px;
    background-color: transparent;
    font-family: inherit;
    font-size: inherit;
    font-weight: inherit;
    margin: 0;
    margin-top: 1px;
    z-index: 2;
  }
  
  .nav-dropdown-content {
    display: none;
    position: absolute;
    background-color: white;
    min-width: 200px;
    border: 1px solid rgba(0,0,0,0.15);
    border-radius: 4px;
    box-shadow: 0px 8px 16px 0px rgba(0,0,0,0.1);
    z-index: 1;
    margin-top: 2px;
    white-space: nowrap;
    padding-top: 4px;
    padding-bottom: 4px;
  }
  
  .nav-dropdown-content hr {
    margin-top: 4px;
    margin-bottom: 4px;
    border: none;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }
  
  .nav-dropdown-active {
    display: block;
  }
  
  .nav-dropdown-content a, .nav-dropdown-content .nav-dropdown-header {
    color: black;
    padding: 6px 24px;
    text-decoration: none;
    display: block;
    text-align: left;
  }
  
  .nav-dropdown-content .nav-dropdown-header {
    display: block;
    padding: 5px 24px;
    padding-bottom: 0;
    text-transform: uppercase;
    font-size: 14px;
    color: #999999;
    white-space: nowrap;
  }
  
  .nav-dropdown:hover .nav-dropbtn {
    color: white;
  }
  
  .nav-dropdown-content a:hover {
    background-color: #ddd;
    color: black;
  }
  
  .nav-right .nav-dropdown-content {
    margin-left: -45%;
    right: 0;
  }
  
  @media screen and (max-width: 768px) {
    .distill-site-header a, .distill-site-header .nav-dropdown  {display: none;}
    .distill-site-header a.nav-toggle {
      float: right;
      display: block;
    }
    .distill-site-header .title {
      margin-left: 0;
    }
    .distill-site-header .nav-right {
      margin-right: 0;
    }
    .distill-site-header {
      overflow: hidden;
    }
    .nav-right .nav-dropdown-content {
      margin-left: 0;
    }
  }
  
  
  @media screen and (max-width: 768px) {
    .distill-site-header.responsive {position: relative;}
    .distill-site-header.responsive a.nav-toggle {
      position: absolute;
      right: 0;
      top: 0;
    }
    .distill-site-header.responsive a,
    .distill-site-header.responsive .nav-dropdown {
      display: block;
      text-align: left;
    }
    .distill-site-header.responsive .nav-left,
    .distill-site-header.responsive .nav-right {
      width: 100%;
    }
    .distill-site-header.responsive .nav-dropdown {float: none;}
    .distill-site-header.responsive .nav-dropdown-content {position: relative;}
    .distill-site-header.responsive .nav-dropdown .nav-dropbtn {
      display: block;
      width: 100%;
      text-align: left;
    }
  }
  
  /* Site Footer */
  
  .distill-site-footer {
    width: 100%;
    overflow: hidden;
    box-sizing: border-box;
    z-index: 3;
    margin-top: 30px;
    padding-top: 30px;
    padding-bottom: 30px;
    text-align: center;
  }
  
  /* Headroom */
  
  d-title {
    padding-top: 6rem;
  }
  
  @media print {
    d-title {
      padding-top: 4rem;
    }
  }
  
  .headroom {
    z-index: 1000;
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
  }
  
  .headroom--transition {
    transition: all .4s ease-in-out;
  }
  
  .headroom--unpinned {
    top: -100px;
  }
  
  .headroom--pinned {
    top: 0;
  }
  
  </style>
  
  <link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet"/>
  <link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet"/>
  <script src="site_libs/headroom-0.9.4/headroom.min.js"></script>
  <!--/radix_placeholder_navigation_in_header-->
  <!--radix_placeholder_distill-->
  
  <style type="text/css">
  
  body {
    background-color: white;
  }
  
  .pandoc-table {
    width: 100%;
  }
  
  .pandoc-table>caption {
    margin-bottom: 10px;
  }
  
  .pandoc-table th:not([align]) {
    text-align: left;
  }
  
  .pagedtable-footer {
    font-size: 15px;
  }
  
  .html-widget {
    margin-bottom: 2.0em;
  }
  
  .l-screen-inset {
    padding-right: 16px;
  }
  
  .l-screen .caption {
    margin-left: 10px;
  }
  
  .shaded {
    background: rgb(247, 247, 247);
    padding-top: 20px;
    padding-bottom: 20px;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }
  
  .shaded .html-widget {
    margin-bottom: 0;
    border: 1px solid rgba(0, 0, 0, 0.1);
  }
  
  .shaded .shaded-content {
    background: white;
  }
  
  .text-output {
    margin-top: 0;
    line-height: 1.5em;
  }
  
  .hidden {
    display: none !important;
  }
  
  d-article {
    padding-bottom: 30px;
  }
  
  d-appendix {
    padding-top: 30px;
  }
  
  d-article>p>img {
    width: 100%;
  }
  
  d-article iframe {
    border: 1px solid rgba(0, 0, 0, 0.1);
    margin-bottom: 2.0em;
    width: 100%;
  }
  
  figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }
  
  /* CSS for table of contents */
  
  .d-toc {
    color: rgba(0,0,0,0.8);
    font-size: 0.8em;
    line-height: 1em;
  }
  
  .d-toc-header {
    font-size: 0.6rem;
    font-weight: 400;
    color: rgba(0, 0, 0, 0.5);
    text-transform: uppercase;
    margin-top: 0;
    margin-bottom: 1.3em;
  }
  
  .d-toc a {
    border-bottom: none;
  }
  
  .d-toc ul {
    padding-left: 0;
  }
  
  .d-toc li>ul {
    padding-top: 0.8em;
    padding-left: 16px;
    margin-bottom: 0.6em;
  }
  
  .d-toc ul,
  .d-toc li {
    list-style-type: none;
  }
  
  .d-toc li {
    margin-bottom: 0.9em;
  }
  
  .d-toc-separator {
    margin-top: 20px;
    margin-bottom: 2em;
  }
  
  .d-article-with-toc {
    border-top: none;
    padding-top: 0;
  }
  
  
  
  /* Tweak code blocks (note that this CSS is repeated above in an injection
     into the d-code shadow dom) */
  
  d-code {
    overflow-x: auto !important;
  }
  
  pre.d-code code.d-code {
    padding-left: 10px;
    font-size: 12px;
    border-left: 2px solid rgba(0,0,0,0.1);
  }
  
  pre.text-output {
  
    font-size: 12px;
    color: black;
    background: none;
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;
  
    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;
  
    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }
  
  @media(min-width: 768px) {
  
  d-code {
    overflow-x: visible !important;
  }
  
  pre.d-code code.d-code  {
      padding-left: 18px;
      font-size: 14px;
  }
  pre.text-output {
    font-size: 14px;
  }
  }
  
  /* Figure */
  
  .figure {
    position: relative;
    margin-bottom: 2.5em;
    margin-top: 1.5em;
  }
  
  .figure img {
    width: 100%;
  }
  
  .figure .caption {
    color: rgba(0, 0, 0, 0.6);
    font-size: 12px;
    line-height: 1.5em;
  }
  
  .figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }
  
  .figure .caption a {
    color: rgba(0, 0, 0, 0.6);
  }
  
  .figure .caption b,
  .figure .caption strong, {
    font-weight: 600;
    color: rgba(0, 0, 0, 1.0);
  }
  
  
  
  /* Tweak 1000px media break to show more text */
  
  @media(min-width: 1000px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 80px [middle-start] 50px [text-start kicker-end] 65px 65px 65px 65px 65px 65px 65px 65px [text-end gutter-start] 65px [middle-end] 65px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 16px;
    }
  
    .grid {
      grid-column-gap: 16px;
    }
  
    d-article {
      font-size: 1.06rem;
      line-height: 1.7em;
    }
    figure .caption, .figure .caption, figure figcaption {
      font-size: 13px;
    }
  }
  
  @media(min-width: 1180px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 60px [middle-start] 60px [text-start kicker-end] 60px 60px 60px 60px 60px 60px 60px 60px [text-end gutter-start] 60px [middle-end] 60px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 32px;
    }
  
    .grid {
      grid-column-gap: 32px;
    }
  }
  
  
  /* Get the citation styles for the appendix (not auto-injected on render since
     we do our own rendering of the citation appendix) */
  
  d-appendix .citation-appendix,
  .d-appendix .citation-appendix {
    font-size: 11px;
    line-height: 15px;
    border-left: 1px solid rgba(0, 0, 0, 0.1);
    padding-left: 18px;
    border: 1px solid rgba(0,0,0,0.1);
    background: rgba(0, 0, 0, 0.02);
    padding: 10px 18px;
    border-radius: 3px;
    color: rgba(150, 150, 150, 1);
    overflow: hidden;
    margin-top: -12px;
    white-space: pre-wrap;
    word-wrap: break-word;
  }
  
  
  /* Social footer */
  
  .social_footer {
    margin-top: 30px;
    margin-bottom: 0;
    color: rgba(0,0,0,0.67);
  }
  
  .disqus-comments {
    margin-right: 30px;
  }
  
  .disqus-comment-count {
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
    cursor: pointer;
  }
  
  #disqus_thread {
    margin-top: 30px;
  }
  
  .article-sharing a {
    border-bottom: none;
    margin-right: 8px;
  }
  
  .article-sharing a:hover {
    border-bottom: none;
  }
  
  .sidebar-section.subscribe {
    font-size: 12px;
    line-height: 1.6em;
  }
  
  .subscribe p {
    margin-bottom: 0.5em;
  }
  
  
  .article-footer .subscribe {
    font-size: 15px;
    margin-top: 45px;
  }
  
  
  .sidebar-section.custom {
    font-size: 12px;
    line-height: 1.6em;
  }
  
  .custom p {
    margin-bottom: 0.5em;
  }
  
  
  /* Improve display for browsers without grid (IE/Edge <= 15) */
  
  .downlevel {
    line-height: 1.6em;
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
    margin: 0;
  }
  
  .downlevel .d-title {
    padding-top: 6rem;
    padding-bottom: 1.5rem;
  }
  
  .downlevel .d-title h1 {
    font-size: 50px;
    font-weight: 700;
    line-height: 1.1em;
    margin: 0 0 0.5rem;
  }
  
  .downlevel .d-title p {
    font-weight: 300;
    font-size: 1.2rem;
    line-height: 1.55em;
    margin-top: 0;
  }
  
  .downlevel .d-byline {
    padding-top: 0.8em;
    padding-bottom: 0.8em;
    font-size: 0.8rem;
    line-height: 1.8em;
  }
  
  .downlevel .section-separator {
    border: none;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
  }
  
  .downlevel .d-article {
    font-size: 1.06rem;
    line-height: 1.7em;
    padding-top: 1rem;
    padding-bottom: 2rem;
  }
  
  
  .downlevel .d-appendix {
    padding-left: 0;
    padding-right: 0;
    max-width: none;
    font-size: 0.8em;
    line-height: 1.7em;
    margin-bottom: 0;
    color: rgba(0,0,0,0.5);
    padding-top: 40px;
    padding-bottom: 48px;
  }
  
  .downlevel .footnotes ol {
    padding-left: 13px;
  }
  
  .downlevel .base-grid,
  .downlevel .distill-header,
  .downlevel .d-title,
  .downlevel .d-abstract,
  .downlevel .d-article,
  .downlevel .d-appendix,
  .downlevel .distill-appendix,
  .downlevel .d-byline,
  .downlevel .d-footnote-list,
  .downlevel .d-citation-list,
  .downlevel .distill-footer,
  .downlevel .appendix-bottom,
  .downlevel .posts-container {
    padding-left: 40px;
    padding-right: 40px;
  }
  
  @media(min-width: 768px) {
    .downlevel .base-grid,
    .downlevel .distill-header,
    .downlevel .d-title,
    .downlevel .d-abstract,
    .downlevel .d-article,
    .downlevel .d-appendix,
    .downlevel .distill-appendix,
    .downlevel .d-byline,
    .downlevel .d-footnote-list,
    .downlevel .d-citation-list,
    .downlevel .distill-footer,
    .downlevel .appendix-bottom,
    .downlevel .posts-container {
    padding-left: 150px;
    padding-right: 150px;
    max-width: 900px;
  }
  }
  
  .downlevel pre code {
    display: block;
    border-left: 2px solid rgba(0, 0, 0, .1);
    padding: 0 0 0 20px;
    font-size: 14px;
  }
  
  .downlevel code, .downlevel pre {
    color: black;
    background: none;
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;
  
    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;
  
    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }
  
  </style>
  
  <script type="application/javascript">
  
  function is_downlevel_browser() {
    if (bowser.isUnsupportedBrowser({ msie: "12", msedge: "16"},
                                   window.navigator.userAgent)) {
      return true;
    } else {
      return window.load_distill_framework === undefined;
    }
  }
  
  // show body when load is complete
  function on_load_complete() {
  
    // set body to visible
    document.body.style.visibility = 'visible';
  
    // force redraw for leaflet widgets
    if (window.HTMLWidgets) {
      var maps = window.HTMLWidgets.findAll(".leaflet");
      $.each(maps, function(i, el) {
        var map = this.getMap();
        map.invalidateSize();
        map.eachLayer(function(layer) {
          if (layer instanceof L.TileLayer)
            layer.redraw();
        });
      });
    }
  
    // trigger 'shown' so htmlwidgets resize
    $('d-article').trigger('shown');
  }
  
  function init_distill() {
  
    init_common();
  
    // create front matter
    var front_matter = $('<d-front-matter></d-front-matter>');
    $('#distill-front-matter').wrap(front_matter);
  
    // create d-title
    $('.d-title').changeElementType('d-title');
  
    // create d-byline
    var byline = $('<d-byline></d-byline>');
    $('.d-byline').replaceWith(byline);
  
    // create d-article
    var article = $('<d-article></d-article>');
    $('.d-article').wrap(article).children().unwrap();
  
    // move posts container into article
    $('.posts-container').appendTo($('d-article'));
  
    // create d-appendix
    $('.d-appendix').changeElementType('d-appendix');
  
    // create d-bibliography
    var bibliography = $('<d-bibliography></d-bibliography>');
    $('#distill-bibliography').wrap(bibliography);
  
    // flag indicating that we have appendix items
    var appendix = $('.appendix-bottom').children('h3').length > 0;
  
    // replace citations with <d-cite>
    $('.citation').each(function(i, val) {
      appendix = true;
      var cites = $(this).attr('data-cites').split(" ");
      var dt_cite = $('<d-cite></d-cite>');
      dt_cite.attr('key', cites.join());
      $(this).replaceWith(dt_cite);
    });
    // remove refs
    $('#refs').remove();
  
    // replace footnotes with <d-footnote>
    $('.footnote-ref').each(function(i, val) {
      appendix = true;
      var href = $(this).attr('href');
      var id = href.replace('#', '');
      var fn = $('#' + id);
      var fn_p = $('#' + id + '>p');
      fn_p.find('.footnote-back').remove();
      var text = fn_p.html();
      var dtfn = $('<d-footnote></d-footnote>');
      dtfn.html(text);
      $(this).replaceWith(dtfn);
    });
    // remove footnotes
    $('.footnotes').remove();
  
    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      var id = $(this).attr('id');
      $('.d-toc a[href="#' + id + '"]').parent().remove();
      appendix = true;
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('d-appendix'));
    });
  
    // show d-appendix if we have appendix content
    $("d-appendix").css('display', appendix ? 'grid' : 'none');
  
    // replace code blocks with d-code
    $('pre>code').each(function(i, val) {
      var code = $(this);
      var pre = code.parent();
      var clz = "";
      var language = pre.attr('class');
      if (language) {
        // map unknown languages to "clike" (without this they just dissapear)
        if ($.inArray(language, ["bash", "clike", "css", "go", "html",
                                 "javascript", "js", "julia", "lua", "markdown",
                                 "markup", "mathml", "python", "svg", "xml"]) == -1)
          language = "clike";
        language = ' language="' + language + '"';
        var dt_code = $('<d-code block' + language + clz + '></d-code>');
        dt_code.text(code.text());
        pre.replaceWith(dt_code);
      } else {
        code.addClass('text-output').unwrap().changeElementType('pre');
      }
    });
  
    // localize layout chunks to just output
    $('.layout-chunk').each(function(i, val) {
  
      // capture layout
      var layout = $(this).attr('data-layout');
  
      // apply layout to markdown level block elements
      var elements = $(this).children().not('d-code, pre.text-output, script');
      elements.each(function(i, el) {
        var layout_div = $('<div class="' + layout + '"></div>');
        if (layout_div.hasClass('shaded')) {
          var shaded_content = $('<div class="shaded-content"></div>');
          $(this).wrap(shaded_content);
          $(this).parent().wrap(layout_div);
        } else {
          $(this).wrap(layout_div);
        }
      });
  
  
      // unwrap the layout-chunk div
      $(this).children().unwrap();
    });
  
    // load distill framework
    load_distill_framework();
  
    // wait for window.distillRunlevel == 4 to do post processing
    function distill_post_process() {
  
      if (!window.distillRunlevel || window.distillRunlevel < 4)
        return;
  
      // hide author/affiliations entirely if we have no authors
      var front_matter = JSON.parse($("#distill-front-matter").html());
      var have_authors = front_matter.authors && front_matter.authors.length > 0;
      if (!have_authors)
        $('d-byline').addClass('hidden');
  
      // table of contents
      if (have_authors) // adjust border if we are in authors
        $('.d-toc').parent().addClass('d-article-with-toc');
  
      // strip links that point to #
      $('.authors-affiliations').find('a[href="#"]').removeAttr('href');
  
      // hide elements of author/affiliations grid that have no value
      function hide_byline_column(caption) {
        $('d-byline').find('h3:contains("' + caption + '")').parent().css('visibility', 'hidden');
      }
  
      // affiliations
      var have_affiliations = false;
      for (var i = 0; i<front_matter.authors.length; ++i) {
        var author = front_matter.authors[i];
        if (author.affiliation !== "&nbsp;") {
          have_affiliations = true;
          break;
        }
      }
      if (!have_affiliations)
        $('d-byline').find('h3:contains("Affiliations")').css('visibility', 'hidden');
  
      // published date
      if (!front_matter.publishedDate)
        hide_byline_column("Published");
  
      // document object identifier
      var doi = $('d-byline').find('h3:contains("DOI")');
      var doi_p = doi.next().empty();
      if (!front_matter.doi) {
        // if we have a citation and valid citationText then link to that
        if ($('#citation').length > 0 && front_matter.citationText) {
          doi.html('Citation');
          $('<a href="#citation"></a>')
            .text(front_matter.citationText)
            .appendTo(doi_p);
        } else {
          hide_byline_column("DOI");
        }
      } else {
        $('<a></a>')
           .attr('href', "https://doi.org/" + front_matter.doi)
           .html(front_matter.doi)
           .appendTo(doi_p);
      }
  
       // change plural form of authors/affiliations
      if (front_matter.authors.length === 1) {
        var grid = $('.authors-affiliations');
        grid.children('h3:contains("Authors")').text('Author');
        grid.children('h3:contains("Affiliations")').text('Affiliation');
      }
  
      // inject pre code styles (can't do this with a global stylesheet b/c a shadow root is used)
      $('d-code').each(function(i, val) {
        var style = document.createElement('style');
        style.innerHTML = 'pre code { padding-left: 10px; font-size: 12px; border-left: 2px solid rgba(0,0,0,0.1); } ' +
                          '@media(min-width: 768px) { pre code { padding-left: 18px; font-size: 14px; } }';
        if (this.shadowRoot)
          this.shadowRoot.appendChild(style);
      });
  
      // move appendix-bottom entries to the bottom
      $('.appendix-bottom').appendTo('d-appendix').children().unwrap();
      $('.appendix-bottom').remove();
  
      // clear polling timer
      clearInterval(tid);
  
      // show body now that everything is ready
      on_load_complete();
    }
  
    var tid = setInterval(distill_post_process, 50);
    distill_post_process();
  
  }
  
  function init_downlevel() {
  
    init_common();
  
     // insert hr after d-title
    $('.d-title').after($('<hr class="section-separator"/>'));
  
    // check if we have authors
    var front_matter = JSON.parse($("#distill-front-matter").html());
    var have_authors = front_matter.authors && front_matter.authors.length > 0;
  
    // manage byline/border
    if (!have_authors)
      $('.d-byline').remove();
    $('.d-byline').after($('<hr class="section-separator"/>'));
    $('.d-byline a').remove();
  
    // remove toc
    $('.d-toc-header').remove();
    $('.d-toc').remove();
    $('.d-toc-separator').remove();
  
    // move appendix elements
    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('.d-appendix'));
    });
  
  
    // inject headers into references and footnotes
    var refs_header = $('<h3></h3>');
    refs_header.text('References');
    $('#refs').prepend(refs_header);
  
    var footnotes_header = $('<h3></h3');
    footnotes_header.text('Footnotes');
    $('.footnotes').children('hr').first().replaceWith(footnotes_header);
  
    // move appendix-bottom entries to the bottom
    $('.appendix-bottom').appendTo('.d-appendix').children().unwrap();
    $('.appendix-bottom').remove();
  
    // remove appendix if it's empty
    if ($('.d-appendix').children().length === 0)
      $('.d-appendix').remove();
  
    // prepend separator above appendix
    $('.d-appendix').before($('<hr class="section-separator" style="clear: both"/>'));
  
    // trim code
    $('pre>code').each(function(i, val) {
      $(this).html($.trim($(this).html()));
    });
  
    // move posts-container right before article
    $('.posts-container').insertBefore($('.d-article'));
  
    $('body').addClass('downlevel');
  
    on_load_complete();
  }
  
  
  function init_common() {
  
    // jquery plugin to change element types
    (function($) {
      $.fn.changeElementType = function(newType) {
        var attrs = {};
  
        $.each(this[0].attributes, function(idx, attr) {
          attrs[attr.nodeName] = attr.nodeValue;
        });
  
        this.replaceWith(function() {
          return $("<" + newType + "/>", attrs).append($(this).contents());
        });
      };
    })(jQuery);
  
    // prevent underline for linked images
    $('a > img').parent().css({'border-bottom' : 'none'});
  
    // mark non-body figures created by knitr chunks as 100% width
    $('.layout-chunk').each(function(i, val) {
      var figures = $(this).find('img, .html-widget');
      if ($(this).attr('data-layout') !== "l-body") {
        figures.css('width', '100%');
      } else {
        figures.css('max-width', '100%');
        figures.filter("[width]").each(function(i, val) {
          var fig = $(this);
          fig.css('width', fig.attr('width') + 'px');
        });
  
      }
    });
  
    // auto-append index.html to post-preview links in file: protocol
    // and in rstudio ide preview
    $('.post-preview').each(function(i, val) {
      if (window.location.protocol === "file:")
        $(this).attr('href', $(this).attr('href') + "index.html");
    });
  
    // get rid of index.html references in header
    if (window.location.protocol !== "file:") {
      $('.distill-site-header a[href]').each(function(i,val) {
        $(this).attr('href', $(this).attr('href').replace("index.html", "./"));
      });
    }
  
    // add class to pandoc style tables
    $('tr.header').parent('thead').parent('table').addClass('pandoc-table');
    $('.kable-table').children('table').addClass('pandoc-table');
  
    // add figcaption style to table captions
    $('caption').parent('table').addClass("figcaption");
  
    // initialize posts list
    if (window.init_posts_list)
      window.init_posts_list();
  
    // implmement disqus comment link
    $('.disqus-comment-count').click(function() {
      window.headroom_prevent_pin = true;
      $('#disqus_thread').toggleClass('hidden');
      if (!$('#disqus_thread').hasClass('hidden')) {
        var offset = $(this).offset();
        $(window).resize();
        $('html, body').animate({
          scrollTop: offset.top - 35
        });
      }
    });
  }
  
  document.addEventListener('DOMContentLoaded', function() {
    if (is_downlevel_browser())
      init_downlevel();
    else
      window.addEventListener('WebComponentsReady', init_distill);
  });
  
  </script>
  
  <!--/radix_placeholder_distill-->
  <script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
  <script src="site_libs/bowser-1.9.3/bowser.min.js"></script>
  <script src="site_libs/webcomponents-2.0.0/webcomponents.js"></script>
  <script src="site_libs/distill-2.2.21/template.v2.js"></script>
  <!--radix_placeholder_site_in_header-->
  <!--/radix_placeholder_site_in_header-->


</head>

<body>

<!--radix_placeholder_front_matter-->

<script id="distill-front-matter" type="text/json">
{"title":"Multi-sample comparisons of the Hayman/Pasricha (C084) ZIPT mini-bulk data set","authors":[{"author":"Peter Hickey","authorURL":"https://peterhickey.org","affiliation":"WEHI SCORE","affiliationURL":"https://www.wehi.edu.au/people/shalin-naik/3310/score"}],"publishedDate":"2020-08-27T00:00:00.000+10:00","citationText":"Hickey, 2020"}
</script>

<!--/radix_placeholder_front_matter-->
<!--radix_placeholder_navigation_before_body-->
<header class="header header--fixed" role="banner">
<nav class="distill-site-nav distill-site-header">
<div class="nav-left">
<a href="index.html" class="title">C084</a>
<a href="index.html">Introduction</a>
<a href="C084_Hayman_Pasricha.preprocess.html">Preprocessing</a>
<a href="C084_Hayman_Pasricha.multi-sample_comparisons.html">Differential analyses</a>
<div class="nav-dropdown">
<button class="nav-dropbtn">
Supplementary files
 
<span class="down-arrow">&#x25BE;</span>
</button>
<div class="nav-dropdown-content">
<a href="../output/scPipe/">scPipe reports</a>
<a href="../output/DEGs/">Differentially expressed genes</a>
<a href="../output/glimma-plots/">Glimma plots</a>
</div>
</div>
</div>
<div class="nav-right">
<a href="https://github.com/WEHISCORE/C084_Hayman_Pasricha">
<i class="fa fa-github" aria-hidden="true"></i>
</a>
<a href="javascript:void(0);" class="nav-toggle">&#9776;</a>
</div>
</nav>
</header>
<!--/radix_placeholder_navigation_before_body-->
<!--radix_placeholder_site_before_body-->
<!--/radix_placeholder_site_before_body-->

<div class="d-title">
<h1>Multi-sample comparisons of the Hayman/Pasricha (C084) ZIPT mini-bulk data set</h1>

</div>

<div class="d-byline">
  Peter Hickey <a href="https://peterhickey.org" class="uri">https://peterhickey.org</a> (WEHI SCORE)<a href="https://www.wehi.edu.au/people/shalin-naik/3310/score" class="uri">https://www.wehi.edu.au/people/shalin-naik/3310/score</a>
  
<br/>2020-08-27
</div>

<div class="d-article">
<h3 class="d-toc-header">Table of Contents</h3>
<nav class="d-toc" id="TOC">
<ul>
<li><a href="#preparing-the-data">Preparing the data</a></li>
<li><a href="#data-pre-processing">Data pre-processing</a><ul>
<li><a href="#transformations-from-the-raw-scale">Transformations from the raw-scale</a></li>
<li><a href="#removing-genes-that-are-lowly-expressed">Removing genes that are lowly expressed</a></li>
<li><a href="#normalising-gene-expression-distributions">Normalising gene expression distributions</a></li>
<li><a href="#unsupervised-clustering-of-samples">Unsupervised clustering of samples</a></li>
</ul></li>
<li><a href="#differential-expression-analysis">Differential expression analysis</a><ul>
<li><a href="#creating-a-design-matrix-and-contrasts">Creating a design matrix and contrasts</a></li>
<li><a href="#removing-heteroscedascity-from-count-data">Removing heteroscedascity from count data</a></li>
<li><a href="#fitting-linear-models-for-comparisons-of-interest">Fitting linear models for comparisons of interest</a></li>
</ul></li>
<li><a href="#results">Results</a><ul>
<li><a href="#figure-1">Figure 1</a><ul>
<li><a href="#endline-pairwise-analyses-between-treatments">Endline pairwise analyses between treatments</a></li>
</ul></li>
<li><a href="#figure-2">Figure 2</a><ul>
<li><a href="#endline-vs.-baseline">Endline vs. baseline</a></li>
</ul></li>
<li><a href="#figure-3">Figure 3</a><ul>
<li><a href="#gene-expression-by-height-for-age-at-endline">Gene expression by Height for Age at endline</a></li>
<li><a href="#gene-expression-by-weight-for-height-at-endline">Gene expression by Weight for Height at endline</a></li>
<li><a href="#gene-expression-by-weight-for-age-at-endline">Gene expression by Weight for Age at endline</a></li>
<li><a href="#gene-expression-by-age">Gene expression by age</a></li>
<li><a href="#gene-expression-by-sex">Gene expression by sex</a></li>
</ul></li>
<li><a href="#figure-4">Figure 4</a><ul>
<li><a href="#gene-expression-by-haemoglobin-at-endline">Gene expression by Haemoglobin at endline</a></li>
<li><a href="#correlation-between-recovering-from-anaemia-and-being-in-a-treatment-group">Correlation between recovering from anaemia and being in a treatment group</a></li>
</ul></li>
</ul></li>
<li><a href="#additional-information">Additional information</a></li>
<li><a href="#session-info">Session info</a></li>
</ul>
</nav>
<hr class="d-toc-separator"/>
<div class="layout-chunk" data-layout="l-body">

</div>
<h1 id="preparing-the-data">Preparing the data</h1>
<p>We start from the preprocessed <em>SingleCellExperiment</em> object created in <a href="C084_Hayman_Pasricha.preprocess.html">‘Preprocessing the Hayman/Pasricha (C084) ZIPT mini-bulk data sett’</a>.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>We convert the data to a <em>DGEList</em> object in preparation for analysis with <em><a href="https://bioconductor.org/packages/3.11/edgeR">edgeR</a></em>.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<h1 id="data-pre-processing">Data pre-processing</h1>
<h2 id="transformations-from-the-raw-scale">Transformations from the raw-scale</h2>
<p>For differential expression and related analyses, gene expression is rarely considered at the level of raw counts since libraries sequenced at a greater depth will result in higher counts. Rather, it is common practice to transform raw counts onto a scale that accounts for such library size differences. Popular transformations include counts per million (CPM), log2-counts per million (log-CPM), reads per kilobase of transcript per million (RPKM), and fragments per kilobase of transcript per million (FPKM).</p>
<p>In our analyses, CPM and log-CPM transformations are used regularly although they do not account for gene length differences as RPKM and FPKM values do. Whilst RPKM and FPKM values can just as well be used, CPM and log-CPM values can be calculated using a counts matrix alone and will suffice for the type of comparisons we are interested in<a href="#fn1" class="footnote-ref" id="fnref1" role="doc-noteref"><sup>1</sup></a>. Assuming that there are no differences in isoform usage between conditions, differential expression analyses look at gene expression changes between conditions rather than comparing expression across multiple genes or drawing conclusions on absolute levels of expression. In other words, gene lengths remain constant for comparisons of interest and any observed differences are a result of changes in condition rather than changes in gene length.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>A CPM value of 1 for a gene equates to having 0.05 counts in the sample with the lowest sequencing depth (LC475_I24, library size approx. 0.05 million) or 0.44 counts in the sample with the greatest sequencing depth (LC475_E19, library size approx. 0.44 million).</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>The log-CPM values will be used for exploratory plots. The log-CPM values use an offset to the CPM values before converting to the log2-scale. By default, the offset is <span class="math inline">\(2/L\)</span> where 2 is the ‘prior count’ and <span class="math inline">\(L\)</span> is the average library size in millions, so the log-CPM values are related to the CPM values by <span class="math inline">\(log_{2}(CPM + 2/L)\)</span>. This calculation ensures that any two read counts with identical CPM values will also have identical log-CPM values. The prior count avoids taking the logarithm of zero, and also reduces spurious variability for genes with very low counts by shrinking all the inter-sample log-fold-changes towards zero, something that is helpful for exploratory plotting.</p>
<p>For this dataset, the average library size is about 0.17 million, so <span class="math inline">\(L \approx\)</span> 0.17 and the minimum log-CPM value for each sample becomes <span class="math inline">\(log_{2}(CPM + 2/L) \approx\)</span> 3.56 In other words, a count of zero for this data maps to a log-CPM value of 3.56 after adding the prior count or offset.</p>
<h2 id="removing-genes-that-are-lowly-expressed">Removing genes that are lowly expressed</h2>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>All datasets will include a mix of genes that are expressed and those that are not expressed. Whilst it is of interest to examine genes that are expressed in one condition but not in another, some genes are unexpressed throughout all samples. We have already removed all genes in this dataset that have zero counts across all 388 samples.</p>
<p>Plotting the distribution log-CPM values shows that a sizeable proportion of genes within each sample are either unexpressed or lowly-expressed with log-CPM values that are small (Figure <a href="#fig:lcpm-density">1</a>).</p>
<p>Genes that do not have a worthwhile number of reads in any sample should be filtered out of the downstream analyses. There are several reasons for this. From a biological point of view, genes that not expressed at a biologically meaningful level in any condition are not of interest and are therefore best ignored. From a statistical point of view, removing low count genes allows the mean-variance relationship in the data to be estimated with greater reliability and also reduces the number of statistical tests that need to be carried out in downstream analyses looking at differential expression.</p>
<p>The <code>filterByExpr()</code> function in the <em><a href="https://bioconductor.org/packages/3.11/edgeR">edgeR</a></em> package provides an automatic way to filter genes, while keeping as many genes as possible with worthwhile counts.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>By default, the function keeps genes with about 10 read counts or more in a minimum number of samples, where the number of samples is chosen according to the minimum group sample size^The actual filtering uses CPM values rather than counts in order to avoid giving preference to samples with large library sizes.]. For this dataset, the median library size is about 0.16) million and 10 / 0.16, big.mark = “,”) approx. 62.2, so the <code>filterByExpr()</code> function keeps genes that have a CPM of 62.2 or more in at least 56 samples. A biologically interesting gene should be expressed in at least 56 samples because all the groups have at least 56 replicates within each plate. The cutoffs used depend on the sequencing depth and on the experimental design. If the library sizes had been larger then a lower CPM cutoff would have been chosen, because larger library sizes provide better resolution to explore more genes at lower expression levels. Alternatively, smaller library sizes decrease our ability to explore marginal genes and hence would have led to a higher CPM cutoff.</p>
<p>Using this criterion, the number of genes is reduced to 5,009, about 15% of the number that we started with (panel B of the Figure <a href="#fig:lcpm-density">1</a>).</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:lcpm-density"></span>
<img src="C084_Hayman_Pasricha.multi-sample_comparisons_files/lcpm-density-1.png" alt="The density of log-CPM values for raw pre-filtered data (A) and post-filtered data (B) are shown for 36 randomly chosen samples. Dotted vertical lines mark the log-CPM threshold used in the filtering step." width="624" />
<p class="caption">
Figure 1: The density of log-CPM values for raw pre-filtered data (A) and post-filtered data (B) are shown for 36 randomly chosen samples. Dotted vertical lines mark the log-CPM threshold used in the filtering step.
</p>
</div>
</div>
<h2 id="normalising-gene-expression-distributions">Normalising gene expression distributions</h2>
<p>During the sample preparation or sequencing process, external factors that are not of biological interest can affect the expression of individual samples. For example, samples processed in the first batch of an experiment can have higher expression overall when compared to samples processed in a second batch. It is assumed that all samples should have a similar range and distribution of expression values. Normalisation is required to ensure that the expression distributions of each sample are similar across the entire experiment.</p>
<p>Any plot showing the per sample expression distributions, such as a density or boxplot, is useful in determining whether any samples are dissimilar to others. Distributions of log-CPM values appear to be similar throughout all samples within this dataset (panel B of Figure <a href="#fig:lcpm-density">1</a>).</p>
<p>Nonetheless, normalisation by the method of trimmed mean of M-values (TMM) <span class="citation" data-cites="robinson2010scaling">(Robinson and Oshlack <a href="#ref-robinson2010scaling" role="doc-biblioref">2010</a>)</span> is performed using the <code>calcNormFactors()</code> function in <em><a href="https://bioconductor.org/packages/3.11/edgeR">edgeR</a></em>. The normalisation factors calculated here are used as a scaling factor for the library sizes.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>For this dataset the effect of TMM-normalisation is mild, as evident in the magnitude of the scaling factors, which are all relatively close to 1 (median = 1.01, IQR = 0.09, 5% and 95% quantiles = 0.85 and 1.15).</p>
<p>Figure <a href="#fig:lcpm-boxplot">2</a> shows the expression distribution of 36 randomly chosen samples for unnormalised and normalised data, where distributions are noticeably different pre-normalisation and are similar post-normalisation.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:lcpm-boxplot"></span>
<img src="C084_Hayman_Pasricha.multi-sample_comparisons_files/lcpm-boxplot-1.png" alt="Boxplots of log-CPM values showing expression distributions for unnormalised data (A) and normalised data (B) for 36 randomly chosen samples." width="624" />
<p class="caption">
Figure 2: Boxplots of log-CPM values showing expression distributions for unnormalised data (A) and normalised data (B) for 36 randomly chosen samples.
</p>
</div>
</div>
<h2 id="unsupervised-clustering-of-samples">Unsupervised clustering of samples</h2>
<p>In our opinion, one of the most important exploratory plots to examine for gene expression analyses is the multi-dimensional scaling (MDS) plot, or similar. The plot shows similarities and dissimilarities between samples in an unsupervised manner so that one can have an idea of the extent to which differential expression can be detected before carrying out formal tests. Ideally, samples would cluster well within the primary condition of interest, and any sample straying far from its group could be identified and followed up for sources of error or extra variation. If present, technical replicates should lie very close to one another.</p>
<p>Such a plot can be made in <em><a href="https://bioconductor.org/packages/3.11/limma">limma</a></em> using the <code>plotMDS()</code> function. The first dimension represents the leading-fold-change that best separates samples and explains the largest proportion of variation in the data, with subsequent dimensions having a smaller effect and being orthogonal to the ones before it. When experimental design involves multiple factors, it is recommended that each factor is examined over several dimensions. If samples cluster by a given factor in any of these dimensions, it suggests that the factor contributes to expression differences and is worth including in the linear modelling. On the other hand, factors that show little or no effect may be left out of downstream analysis.</p>
<p>In this dataset, samples can be seen to cluster strongly by experimental batch over dimension 1. This dominates any variation due to timepoint or treatment. Keeping in mind that the first dimension explains the largest proportion of variation in the data, notice that the range of values over the dimensions become smaller as we move to higher dimensions.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:mds"></span>
<img src="C084_Hayman_Pasricha.multi-sample_comparisons_files/mds-1.png" alt="MDS plots of log-CPM values over dimensions 1 and 2 with samples coloured by batch and over dimensions 3 and 4 with samples coloured by treatment and timepoint. Distances on the plot correspond to the leading fold-change, which is the average (root-mean-square) log2-fold-change for the 500 genes most divergent between each pair of samples by default." width="624" />
<p class="caption">
Figure 3: MDS plots of log-CPM values over dimensions 1 and 2 with samples coloured by batch and over dimensions 3 and 4 with samples coloured by treatment and timepoint. Distances on the plot correspond to the leading fold-change, which is the average (root-mean-square) log2-fold-change for the 500 genes most divergent between each pair of samples by default.
</p>
</div>
</div>
<p>Datasets where samples do not cluster by experimental group may show little or no evidence of differential expression in the downstream analysis.</p>
<h1 id="differential-expression-analysis">Differential expression analysis</h1>
<h2 id="creating-a-design-matrix-and-contrasts">Creating a design matrix and contrasts</h2>
<p>In this study, the primary interest is to see which genes are expressed at different levels between the different treatments and timepoints. In our analysis, linear models are fitted to the data with the assumption that the underlying data is normally distributed. To get started, a design matrix is set up with the <code>group</code> (i.e. the combination of <code>timepoint</code> and <code>treatment</code>), <code>id</code> (i.e. subject ID), and <code>plate_number</code> (i.e. batch) information.</p>
<aside>
Of secondary interest is to see which genes have their expression associated with additional coveriates (e.g., weight for age at endline). We will use different design matrices and contrasts to answer these questions.
</aside>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Contrasts for pairwise comparisons between <code>treatment</code> and <code>timepoint</code>, within each <code>id</code> and blocking on <code>plate_number</code>, are set up in <em><a href="https://bioconductor.org/packages/3.11/limma">limma</a></em> using the <code>makeContrasts()</code> function.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>A key strength of <em><a href="https://bioconductor.org/packages/3.11/limma">limma</a></em>’s linear modelling approach, is the ability accommodate arbitrary experimental complexity. Simple designs through to more complicated factorial designs and models with interaction terms can be handled relatively easily.</p>
<aside>
We will make use of more complicated models in some of the analyses of secondary outcomes.
</aside>
<h2 id="removing-heteroscedascity-from-count-data">Removing heteroscedascity from count data</h2>
<p>It has been shown that for RNA-seq count data, the variance is not independent of the mean <span class="citation" data-cites="law2014voom">(Law et al. <a href="#ref-law2014voom" role="doc-biblioref">2014</a>)</span> – this is true of raw counts or when transformed to log-CPM values. Methods that model counts using a Negative Binomial distribution assume a quadratic mean-variance relationship. In <em><a href="https://bioconductor.org/packages/3.11/limma">limma</a></em>, linear modelling is carried out on the log-CPM values which are assumed to be normally distributed and the mean-variance relationship is accommodated using precision weights calculated by the <code>voom()</code> function.</p>
<p><code>voom()</code> converts raw counts to log-CPM values by automatically extracting library sizes and normalisation factors.</p>
<p>The mean-variance relationship of log-CPM values for this dataset is shown in the left-hand panel of Figure <a href="#fig:voom">4</a>. Typically, the voom-plot shows a decreasing trend between the means and variances resulting from a combination of technical variation in the sequencing experiment and biological variation amongst the replicate samples from different cell populations. Experiments with high biological variation usually result in flatter trends, where variance values plateau at high expression values. Experiments with low biological variation tend to result in sharp decreasing trends.</p>
<p>Moreover, the voom-plot provides a visual check on the level of filtering performed upstream. If filtering of lowly-expressed genes is insufficient, a drop in variance levels can be observed at the low end of the expression scale due to very small counts. If this is observed, one should return to the earlier filtering step and increase the expression threshold applied to the dataset.</p>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
Coefficients not estimable: id356 id387 </code></pre>
<pre><code>
Coefficients not estimable: id356 id387 </code></pre>
<div class="figure"><span id="fig:voom"></span>
<img src="C084_Hayman_Pasricha.multi-sample_comparisons_files/voom-1.png" alt="Means (x-axis) and variances (y-axis) of each gene are plotted to show the dependence between the two before voom is applied to the data (left panel) and how the trend is removed after voom precision weights are applied to the data (right panel). The plot on the left is created within the `voom()` function which extracts residual variances from fitting linear models to log-CPM transformed data. Variances are then rescaled to quarter-root variances (or square-root of standard deviations) and plotted against the average log2 count for each gene. The plot on the right is created using `plotSA()` which plots log2 residual standard deviations against mean log-CPM values. In both plots, each black dot represents a gene. On the left plot, the red curve shows the estimated mean-variance trend used to compute the voom weights. On the right plot, the average log2 residual standard deviation estimated by the empirical Bayes algorithm is marked by a horizontal blue line." width="624" />
<p class="caption">
Figure 4: Means (x-axis) and variances (y-axis) of each gene are plotted to show the dependence between the two before voom is applied to the data (left panel) and how the trend is removed after voom precision weights are applied to the data (right panel). The plot on the left is created within the <code>voom()</code> function which extracts residual variances from fitting linear models to log-CPM transformed data. Variances are then rescaled to quarter-root variances (or square-root of standard deviations) and plotted against the average log2 count for each gene. The plot on the right is created using <code>plotSA()</code> which plots log2 residual standard deviations against mean log-CPM values. In both plots, each black dot represents a gene. On the left plot, the red curve shows the estimated mean-variance trend used to compute the voom weights. On the right plot, the average log2 residual standard deviation estimated by the empirical Bayes algorithm is marked by a horizontal blue line.
</p>
</div>
</div>
<h2 id="fitting-linear-models-for-comparisons-of-interest">Fitting linear models for comparisons of interest</h2>
<p>Linear modelling in <em><a href="https://bioconductor.org/packages/3.11/limma">limma</a></em> fits a separate model to the expression values for each gene. Next, empirical Bayes moderation is carried out by borrowing information across all the genes to obtain more precise estimates of gene-wise variability <span class="citation" data-cites="smyth2004linear">(Smyth <a href="#ref-smyth2004linear" role="doc-biblioref">2004</a>)</span>. The model’s residual variances are plotted against average expression values in Figure <a href="#fig:voom">4</a>. It can be seen from this plot that the variance is no longer dependent on the mean expression level.</p>
<h1 id="results">Results</h1>
<h2 id="figure-1">Figure 1</h2>
<h3 id="endline-pairwise-analyses-between-treatments">Endline pairwise analyses between treatments</h3>
<p>For a quick look at differential expression levels, the number of significantly up- and down-regulated genes can be summarised in a table. Significance is defined using an adjusted p-value cutoff that is set at 5% by default.</p>
<div class="layout-chunk" data-layout="l-body">
<table>
<caption><span id="tab:unnamed-chunk-10">Table 1: </span>Number of DEGs per comparison. Each row corresponds to a comparison and each column corresponds to the number of downregulated genes (<code>Down</code>), the number of non-differentially expressed genes (<code>NotSig</code>), the number of upregulated genes (<code>Up</code>)</caption>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: right;">Down</th>
<th style="text-align: right;">NotSig</th>
<th style="text-align: right;">Up</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">A.endline_vs_B.endline</td>
<td style="text-align: right;">14</td>
<td style="text-align: right;">4983</td>
<td style="text-align: right;">12</td>
</tr>
<tr class="even">
<td style="text-align: left;">A.endline_vs_C.endline</td>
<td style="text-align: right;">19</td>
<td style="text-align: right;">4981</td>
<td style="text-align: right;">9</td>
</tr>
<tr class="odd">
<td style="text-align: left;">B.endline_vs_C.endline</td>
<td style="text-align: right;">25</td>
<td style="text-align: right;">4934</td>
<td style="text-align: right;">50</td>
</tr>
</tbody>
</table>
</div>
<p><strong>TODO</strong> Describe other outputs</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<h4 id="adjusted-by-sex">Adjusted by sex</h4>
<p><strong>TODO</strong>: Decide on strategy of fixed/random effects to achieve this aim.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<h2 id="figure-2">Figure 2</h2>
<h3 id="endline-vs.-baseline">Endline vs. baseline</h3>
<p>For a quick look at differential expression levels, the number of significantly up- and down-regulated genes can be summarised in a table. Significance is defined using an adjusted p-value cutoff that is set at 5% by default.</p>
<div class="layout-chunk" data-layout="l-body">
<table>
<caption><span id="tab:unnamed-chunk-13">Table 2: </span>Number of DEGs per comparison. Each row corresponds to a comparison and each column corresponds to the number of downregulated genes (<code>Down</code>), the number of non-differentially expressed genes (<code>NotSig</code>), the number of upregulated genes (<code>Up</code>)</caption>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: right;">Down</th>
<th style="text-align: right;">NotSig</th>
<th style="text-align: right;">Up</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">A.endline_vs_A.baseline</td>
<td style="text-align: right;">409</td>
<td style="text-align: right;">4057</td>
<td style="text-align: right;">543</td>
</tr>
<tr class="even">
<td style="text-align: left;">B.endline_vs_B.baseline</td>
<td style="text-align: right;">60</td>
<td style="text-align: right;">4734</td>
<td style="text-align: right;">215</td>
</tr>
<tr class="odd">
<td style="text-align: left;">C.endline_vs_C.baseline</td>
<td style="text-align: right;">138</td>
<td style="text-align: right;">4565</td>
<td style="text-align: right;">306</td>
</tr>
</tbody>
</table>
</div>
<p><strong>TODO</strong> Describe other outputs</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<h4 id="check-for-differences-in-age-at-enrolment">Check for differences in age at enrolment</h4>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Figure <a href="#fig:age0-density">5</a> shows the distribution of age at enrolment (<code>age0</code>) by treatment arm. There is no significant difference in <code>age0</code> by treatment arm (ANOVA, <span class="math inline">\(P =\)</span> 0.998) nor is there an association when <code>age0</code> is binned as <span class="math inline">\((8.5-10]\)</span> or <span class="math inline">\((10-11.5]\)</span> (<span class="math inline">\(\chi^2\)</span>-test <span class="math inline">\(P =\)</span> 0.87).</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:age0-density"></span>
<img src="C084_Hayman_Pasricha.multi-sample_comparisons_files/age0-density-1.png" alt="Density estimates of `age0` in each treatment arm. The density curves were scaled such that the area under each curve corresponds to the total number of subjects in each treatment arm." width="624" />
<p class="caption">
Figure 5: Density estimates of <code>age0</code> in each treatment arm. The density curves were scaled such that the area under each curve corresponds to the total number of subjects in each treatment arm.
</p>
</div>
</div>
<h2 id="figure-3">Figure 3</h2>
<h3 id="gene-expression-by-height-for-age-at-endline">Gene expression by Height for Age at endline</h3>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Figure <a href="#fig:HAZ24-density">6</a> shows the distribution of height for age at endline (<code>HAZ24</code>) by treatment arm. There is 1 samples without a <code>HAZ24</code> measurement (id: 348).</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:HAZ24-density"></span>
<img src="C084_Hayman_Pasricha.multi-sample_comparisons_files/HAZ24-density-1.png" alt="Density estimates of `HAZ24` in each treatment arm. The density curves were scaled such that the area under each curve corresponds to the total number of subjects in each treatment arm. Samples to the left of the vertical dashed line are `Stunted24`." width="624" />
<p class="caption">
Figure 6: Density estimates of <code>HAZ24</code> in each treatment arm. The density curves were scaled such that the area under each curve corresponds to the total number of subjects in each treatment arm. Samples to the left of the vertical dashed line are <code>Stunted24</code>.
</p>
</div>
</div>
<p>Stunting at endline (<code>Stunted24</code>) is defined as <code>HAZ24</code> <span class="math inline">\(&lt; -2\)</span>. The table below classifies subjects by <code>treatment</code> and <code>Stunted24</code>.</p>
<div class="layout-chunk" data-layout="l-body">
<table>
<thead>
<tr class="header">
<th style="text-align: left;">Stunted24/treatment</th>
<th style="text-align: right;">A</th>
<th style="text-align: right;">B</th>
<th style="text-align: right;">C</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">FALSE</td>
<td style="text-align: right;">17</td>
<td style="text-align: right;">26</td>
<td style="text-align: right;">22</td>
</tr>
<tr class="even">
<td style="text-align: left;">TRUE</td>
<td style="text-align: right;">12</td>
<td style="text-align: right;">13</td>
<td style="text-align: right;">11</td>
</tr>
<tr class="odd">
<td style="text-align: left;">NA</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
</tr>
</tbody>
</table>
</div>
<h4 id="haz24"><code>HAZ24</code></h4>
<p>We subset the data to just the endline <code>timepoint</code> and update the design matrix to include <code>HAZ24</code> in addition to <code>treatment</code>, <code>id</code> (i.e. subject ID), and <code>plate_number</code> (i.e. batch) information.</p>
<p>Contrasts for association of gene expression with <code>HAZ24</code>, accounting for <code>group</code>, <code>id</code>, and <code>plate_number</code>, are set up in <em><a href="https://bioconductor.org/packages/3.11/limma">limma</a></em> using the <code>makeContrasts()</code> function.</p>
<div class="layout-chunk" data-layout="l-body">
<p><img src="C084_Hayman_Pasricha.multi-sample_comparisons_files/unnamed-chunk-18-1.png" width="624" /></p>
</div>
<p>For a quick look at differential expression levels, the number of significantly up- and down-regulated genes can be summarised in a table. Significance is defined using an adjusted p-value cutoff that is set at 5% by default.</p>
<div class="layout-chunk" data-layout="l-body">
<table>
<caption><span id="tab:unnamed-chunk-19">Table 3: </span>Number of DEGs per comparison. Each row corresponds to a comparison and each column corresponds to the number of downregulated genes (<code>Down</code>), the number of non-differentially expressed genes (<code>NotSig</code>), the number of upregulated genes (<code>Up</code>)</caption>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: right;">Down</th>
<th style="text-align: right;">NotSig</th>
<th style="text-align: right;">Up</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">HAZ24</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">5009</td>
<td style="text-align: right;">0</td>
</tr>
</tbody>
</table>
</div>
<p><strong>TODO</strong> Describe other outputs</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<h4 id="stunted24"><code>Stunted24</code></h4>
<p>We subset the data to just the endline <code>timepoint</code> and update the design matrix to include <code>Stunted24</code> in addition to <code>treatment</code>, <code>id</code> (i.e. subject ID), and <code>plate_number</code> (i.e. batch) information.</p>
<p>Contrasts for association of gene expression with <code>Stunted24</code>, accounting for <code>group</code>, <code>id</code>, and <code>plate_number</code>, are set up in <em><a href="https://bioconductor.org/packages/3.11/limma">limma</a></em> using the <code>makeContrasts()</code> function.</p>
<div class="layout-chunk" data-layout="l-body">
<p><img src="C084_Hayman_Pasricha.multi-sample_comparisons_files/unnamed-chunk-21-1.png" width="624" /></p>
</div>
<p>For a quick look at differential expression levels, the number of significantly up- and down-regulated genes can be summarised in a table. Significance is defined using an adjusted p-value cutoff that is set at 5% by default.</p>
<div class="layout-chunk" data-layout="l-body">
<table>
<caption><span id="tab:unnamed-chunk-22">Table 4: </span>Number of DEGs per comparison. Each row corresponds to a comparison and each column corresponds to the number of downregulated genes (<code>Down</code>), the number of non-differentially expressed genes (<code>NotSig</code>), the number of upregulated genes (<code>Up</code>)</caption>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: right;">Down</th>
<th style="text-align: right;">NotSig</th>
<th style="text-align: right;">Up</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Stunted24</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">5009</td>
<td style="text-align: right;">0</td>
</tr>
</tbody>
</table>
</div>
<p><strong>TODO</strong> Describe other outputs</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<h3 id="gene-expression-by-weight-for-height-at-endline">Gene expression by Weight for Height at endline</h3>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Figure <a href="#fig:WHZ24-density">7</a> shows the distribution of weight for height at endline (<code>WHZ24</code>) by treatment arm. There is 1 samples without a <code>WHZ24</code> measurement (id: 348).</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:WHZ24-density"></span>
<img src="C084_Hayman_Pasricha.multi-sample_comparisons_files/WHZ24-density-1.png" alt="Density estimates of `WHZ24` in each treatment arm. The density curves were scaled such that the area under each curve corresponds to the total number of subjects in each treatment arm. Samples to the left of the vertical dashed line are `Wasted24`." width="624" />
<p class="caption">
Figure 7: Density estimates of <code>WHZ24</code> in each treatment arm. The density curves were scaled such that the area under each curve corresponds to the total number of subjects in each treatment arm. Samples to the left of the vertical dashed line are <code>Wasted24</code>.
</p>
</div>
</div>
<p>Wasting at endline (<code>Wasted24</code>) is defined as <code>WHZ24</code> <span class="math inline">\(&lt; -2\)</span>. The table below classifies subjects by <code>treatment</code> and <code>Wasted24</code>.</p>
<div class="layout-chunk" data-layout="l-body">
<table>
<thead>
<tr class="header">
<th style="text-align: left;">Wasted24/treatment</th>
<th style="text-align: right;">A</th>
<th style="text-align: right;">B</th>
<th style="text-align: right;">C</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">FALSE</td>
<td style="text-align: right;">24</td>
<td style="text-align: right;">35</td>
<td style="text-align: right;">30</td>
</tr>
<tr class="even">
<td style="text-align: left;">TRUE</td>
<td style="text-align: right;">5</td>
<td style="text-align: right;">4</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="odd">
<td style="text-align: left;">NA</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
</tr>
</tbody>
</table>
</div>
<h4 id="whz24"><code>WHZ24</code></h4>
<p>We subset the data to just the endline <code>timepoint</code> and update the design matrix to include <code>WHZ24</code> in addition to <code>treatment</code>, <code>id</code> (i.e. subject ID), and <code>plate_number</code> (i.e. batch) information.</p>
<p>Contrasts for association of gene expression with <code>WHZ24</code>, accounting for <code>group</code>, <code>id</code>, and <code>plate_number</code>, are set up in <em><a href="https://bioconductor.org/packages/3.11/limma">limma</a></em> using the <code>makeContrasts()</code> function.</p>
<div class="layout-chunk" data-layout="l-body">
<p><img src="C084_Hayman_Pasricha.multi-sample_comparisons_files/unnamed-chunk-26-1.png" width="624" /></p>
</div>
<p>For a quick look at differential expression levels, the number of significantly up- and down-regulated genes can be summarised in a table. Significance is defined using an adjusted p-value cutoff that is set at 5% by default.</p>
<div class="layout-chunk" data-layout="l-body">
<table>
<caption><span id="tab:unnamed-chunk-27">Table 5: </span>Number of DEGs per comparison. Each row corresponds to a comparison and each column corresponds to the number of downregulated genes (<code>Down</code>), the number of non-differentially expressed genes (<code>NotSig</code>), the number of upregulated genes (<code>Up</code>)</caption>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: right;">Down</th>
<th style="text-align: right;">NotSig</th>
<th style="text-align: right;">Up</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">WHZ24</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">5009</td>
<td style="text-align: right;">0</td>
</tr>
</tbody>
</table>
</div>
<p><strong>TODO</strong> Describe other outputs</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<h4 id="wasted24"><code>Wasted24</code></h4>
<p>We subset the data to just the endline <code>timepoint</code> and update the design matrix to include <code>Wasted24</code> in addition to <code>treatment</code>, <code>id</code> (i.e. subject ID), and <code>plate_number</code> (i.e. batch) information.</p>
<p>Contrasts for association of gene expression with <code>Wasted24</code>, accounting for <code>group</code>, <code>id</code>, and <code>plate_number</code>, are set up in <em><a href="https://bioconductor.org/packages/3.11/limma">limma</a></em> using the <code>makeContrasts()</code> function.</p>
<div class="layout-chunk" data-layout="l-body">
<p><img src="C084_Hayman_Pasricha.multi-sample_comparisons_files/unnamed-chunk-29-1.png" width="624" /></p>
</div>
<p>For a quick look at differential expression levels, the number of significantly up- and down-regulated genes can be summarised in a table. Significance is defined using an adjusted p-value cutoff that is set at 5% by default.</p>
<div class="layout-chunk" data-layout="l-body">
<table>
<caption><span id="tab:unnamed-chunk-30">Table 6: </span>Number of DEGs per comparison. Each row corresponds to a comparison and each column corresponds to the number of downregulated genes (<code>Down</code>), the number of non-differentially expressed genes (<code>NotSig</code>), the number of upregulated genes (<code>Up</code>)</caption>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: right;">Down</th>
<th style="text-align: right;">NotSig</th>
<th style="text-align: right;">Up</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Wasted24</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">5003</td>
<td style="text-align: right;">4</td>
</tr>
</tbody>
</table>
</div>
<p><strong>TODO</strong> Describe other outputs</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<h3 id="gene-expression-by-weight-for-age-at-endline">Gene expression by Weight for Age at endline</h3>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Figure <a href="#fig:WAZ24-density">8</a> shows the distribution of weight for age at endline (<code>WAZ24</code>) by treatment arm. There is 1 samples without a <code>WAZ24</code> measurement (id: 348).</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:WAZ24-density"></span>
<img src="C084_Hayman_Pasricha.multi-sample_comparisons_files/WAZ24-density-1.png" alt="Density estimates of `WAZ24` in each treatment arm. The density curves were scaled such that the area under each curve corresponds to the total number of subjects in each treatment arm. Samples to the left of the vertical dashed line are `Underweight24`." width="624" />
<p class="caption">
Figure 8: Density estimates of <code>WAZ24</code> in each treatment arm. The density curves were scaled such that the area under each curve corresponds to the total number of subjects in each treatment arm. Samples to the left of the vertical dashed line are <code>Underweight24</code>.
</p>
</div>
</div>
<p>Underweight at endline (<code>Underweight24</code>) is defined as <code>WAZ24</code> <span class="math inline">\(&lt; -2\)</span>. The table below classifies subjects by <code>treatment</code> and <code>Underweight24</code>.</p>
<div class="layout-chunk" data-layout="l-body">
<table>
<thead>
<tr class="header">
<th style="text-align: left;">Underweight24/treatment</th>
<th style="text-align: right;">A</th>
<th style="text-align: right;">B</th>
<th style="text-align: right;">C</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">FALSE</td>
<td style="text-align: right;">18</td>
<td style="text-align: right;">31</td>
<td style="text-align: right;">27</td>
</tr>
<tr class="even">
<td style="text-align: left;">TRUE</td>
<td style="text-align: right;">11</td>
<td style="text-align: right;">8</td>
<td style="text-align: right;">6</td>
</tr>
<tr class="odd">
<td style="text-align: left;">NA</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
</tr>
</tbody>
</table>
</div>
<h4 id="waz24"><code>WAZ24</code></h4>
<p>We subset the data to just the endline <code>timepoint</code> and update the design matrix to include <code>WAZ24</code> in addition to <code>treatment</code>, <code>id</code> (i.e. subject ID), and <code>plate_number</code> (i.e. batch) information.</p>
<p>Contrasts for association of gene expression with <code>WAZ24</code>, accounting for <code>group</code>, <code>id</code>, and <code>plate_number</code>, are set up in <em><a href="https://bioconductor.org/packages/3.11/limma">limma</a></em> using the <code>makeContrasts()</code> function.</p>
<div class="layout-chunk" data-layout="l-body">
<p><img src="C084_Hayman_Pasricha.multi-sample_comparisons_files/unnamed-chunk-34-1.png" width="624" /></p>
</div>
<p>For a quick look at differential expression levels, the number of significantly up- and down-regulated genes can be summarised in a table. Significance is defined using an adjusted p-value cutoff that is set at 5% by default.</p>
<div class="layout-chunk" data-layout="l-body">
<table>
<caption><span id="tab:unnamed-chunk-35">Table 7: </span>Number of DEGs per comparison. Each row corresponds to a comparison and each column corresponds to the number of downregulated genes (<code>Down</code>), the number of non-differentially expressed genes (<code>NotSig</code>), the number of upregulated genes (<code>Up</code>)</caption>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: right;">Down</th>
<th style="text-align: right;">NotSig</th>
<th style="text-align: right;">Up</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">WAZ24</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">5009</td>
<td style="text-align: right;">0</td>
</tr>
</tbody>
</table>
</div>
<p><strong>TODO</strong> Describe other outputs</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<h4 id="underweight24"><code>Underweight24</code></h4>
<p>We subset the data to just the endline <code>timepoint</code> and update the design matrix to include <code>Underweight24</code> in addition to <code>treatment</code>, <code>id</code> (i.e. subject ID), and <code>plate_number</code> (i.e. batch) information.</p>
<p>Contrasts for association of gene expression with <code>Underweight24</code>, accounting for <code>group</code>, <code>id</code>, and <code>plate_number</code>, are set up in <em><a href="https://bioconductor.org/packages/3.11/limma">limma</a></em> using the <code>makeContrasts()</code> function.</p>
<div class="layout-chunk" data-layout="l-body">
<p><img src="C084_Hayman_Pasricha.multi-sample_comparisons_files/unnamed-chunk-37-1.png" width="624" /></p>
</div>
<p>For a quick look at differential expression levels, the number of significantly up- and down-regulated genes can be summarised in a table. Significance is defined using an adjusted p-value cutoff that is set at 5% by default.</p>
<div class="layout-chunk" data-layout="l-body">
<table>
<caption><span id="tab:unnamed-chunk-38">Table 8: </span>Number of DEGs per comparison. Each row corresponds to a comparison and each column corresponds to the number of downregulated genes (<code>Down</code>), the number of non-differentially expressed genes (<code>NotSig</code>), the number of upregulated genes (<code>Up</code>)</caption>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: right;">Down</th>
<th style="text-align: right;">NotSig</th>
<th style="text-align: right;">Up</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Underweight24</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">5009</td>
<td style="text-align: right;">0</td>
</tr>
</tbody>
</table>
</div>
<p><strong>TODO</strong> Describe other outputs</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<h3 id="gene-expression-by-age">Gene expression by age</h3>
<p><strong>TODO</strong>: It’s not clear what comparison is being requested from Tom’s email.</p>
<p><strong>TODO</strong>: It’s not clear which age column to use (EnrAgeMo, age0, age12, or age24).</p>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
factor(0)
102 Levels: 102 103 104 106 107 108 109 110 111 112 113 114 ... 387</code></pre>
<pre><code>
[1] 348
102 Levels: 102 103 104 106 107 108 109 110 111 112 113 114 ... 387</code></pre>
<pre><code>
[1] 348
102 Levels: 102 103 104 106 107 108 109 110 111 112 113 114 ... 387</code></pre>
<pre><code>
factor(0)
102 Levels: 102 103 104 106 107 108 109 110 111 112 113 114 ... 387</code></pre>
<p><img src="C084_Hayman_Pasricha.multi-sample_comparisons_files/unnamed-chunk-40-1.png" width="624" /></p>
</div>
<h3 id="gene-expression-by-sex">Gene expression by sex</h3>
<p><strong>TODO</strong>: It’s not clear what comparison is being requested from Tom’s email.</p>
<h2 id="figure-4">Figure 4</h2>
<h3 id="gene-expression-by-haemoglobin-at-endline">Gene expression by Haemoglobin at endline</h3>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Figure <a href="#fig:hb24-density">9</a> shows the distribution of weight for age at endline (<code>hb24</code>) by treatment arm. There is 2 samples without a <code>hb24</code> measurement (id: 147 and 348).</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:hb24-density"></span>
<img src="C084_Hayman_Pasricha.multi-sample_comparisons_files/hb24-density-1.png" alt="Density estimates of `hb24` in each treatment arm. The density curves were scaled such that the area under each curve corresponds to the total number of subjects in each treatment arm. Samples to the left of the vertical dashed line are `Anaemic24`." width="624" />
<p class="caption">
Figure 9: Density estimates of <code>hb24</code> in each treatment arm. The density curves were scaled such that the area under each curve corresponds to the total number of subjects in each treatment arm. Samples to the left of the vertical dashed line are <code>Anaemic24</code>.
</p>
</div>
</div>
<p>Anaemic at endline (<code>Anaemic24</code>) is defined as <code>hb24</code> <span class="math inline">\(&lt; -2\)</span>. The table below classifies subjects by <code>treatment</code> and <code>Anaemic24</code>.</p>
<div class="layout-chunk" data-layout="l-body">
<table>
<thead>
<tr class="header">
<th style="text-align: left;">Anaemic24/treatment</th>
<th style="text-align: right;">A</th>
<th style="text-align: right;">B</th>
<th style="text-align: right;">C</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">FALSE</td>
<td style="text-align: right;">9</td>
<td style="text-align: right;">9</td>
<td style="text-align: right;">6</td>
</tr>
<tr class="even">
<td style="text-align: left;">TRUE</td>
<td style="text-align: right;">20</td>
<td style="text-align: right;">29</td>
<td style="text-align: right;">27</td>
</tr>
<tr class="odd">
<td style="text-align: left;">NA</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">0</td>
</tr>
</tbody>
</table>
</div>
<h4 id="hb24"><code>hb24</code></h4>
<p>We subset the data to just the endline <code>timepoint</code> and update the design matrix to include <code>hb24</code> in addition to <code>treatment</code>, <code>id</code> (i.e. subject ID), and <code>plate_number</code> (i.e. batch) information.</p>
<p>Contrasts for association of gene expression with <code>hb24</code>, accounting for <code>group</code>, <code>id</code>, and <code>plate_number</code>, are set up in <em><a href="https://bioconductor.org/packages/3.11/limma">limma</a></em> using the <code>makeContrasts()</code> function.</p>
<div class="layout-chunk" data-layout="l-body">
<p><img src="C084_Hayman_Pasricha.multi-sample_comparisons_files/unnamed-chunk-43-1.png" width="624" /></p>
</div>
<p>For a quick look at differential expression levels, the number of significantly up- and down-regulated genes can be summarised in a table. Significance is defined using an adjusted p-value cutoff that is set at 5% by default.</p>
<div class="layout-chunk" data-layout="l-body">
<table>
<caption><span id="tab:unnamed-chunk-44">Table 9: </span>Number of DEGs per comparison. Each row corresponds to a comparison and each column corresponds to the number of downregulated genes (<code>Down</code>), the number of non-differentially expressed genes (<code>NotSig</code>), the number of upregulated genes (<code>Up</code>)</caption>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: right;">Down</th>
<th style="text-align: right;">NotSig</th>
<th style="text-align: right;">Up</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">hb24</td>
<td style="text-align: right;">112</td>
<td style="text-align: right;">4849</td>
<td style="text-align: right;">48</td>
</tr>
</tbody>
</table>
</div>
<p><strong>TODO</strong> Describe other outputs</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<h4 id="anaemic24"><code>Anaemic24</code></h4>
<p>We subset the data to just the endline <code>timepoint</code> and update the design matrix to include <code>Anaemic24</code> in addition to <code>treatment</code>, <code>id</code> (i.e. subject ID), and <code>plate_number</code> (i.e. batch) information.</p>
<p>Contrasts for association of gene expression with <code>Anaemic24</code>, accounting for <code>group</code>, <code>id</code>, and <code>plate_number</code>, are set up in <em><a href="https://bioconductor.org/packages/3.11/limma">limma</a></em> using the <code>makeContrasts()</code> function.</p>
<div class="layout-chunk" data-layout="l-body">
<p><img src="C084_Hayman_Pasricha.multi-sample_comparisons_files/unnamed-chunk-46-1.png" width="624" /></p>
</div>
<p>For a quick look at differential expression levels, the number of significantly up- and down-regulated genes can be summarised in a table. Significance is defined using an adjusted p-value cutoff that is set at 5% by default.</p>
<div class="layout-chunk" data-layout="l-body">
<table>
<caption><span id="tab:unnamed-chunk-47">Table 10: </span>Number of DEGs per comparison. Each row corresponds to a comparison and each column corresponds to the number of downregulated genes (<code>Down</code>), the number of non-differentially expressed genes (<code>NotSig</code>), the number of upregulated genes (<code>Up</code>)</caption>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: right;">Down</th>
<th style="text-align: right;">NotSig</th>
<th style="text-align: right;">Up</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Anaemic24</td>
<td style="text-align: right;">7</td>
<td style="text-align: right;">4984</td>
<td style="text-align: right;">18</td>
</tr>
</tbody>
</table>
</div>
<p><strong>TODO</strong> Describe other outputs</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<h3 id="correlation-between-recovering-from-anaemia-and-being-in-a-treatment-group">Correlation between recovering from anaemia and being in a treatment group</h3>
<p><strong>TODO</strong>: It’s not clear what comparison is being requested from Tom’s email.</p>
<div class="layout-chunk" data-layout="l-body">
<p><img src="C084_Hayman_Pasricha.multi-sample_comparisons_files/unnamed-chunk-49-1.png" width="624" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<p><img src="C084_Hayman_Pasricha.multi-sample_comparisons_files/unnamed-chunk-50-1.png" width="624" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Anaemia rates between treatments at baseline.</p>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
   
    FALSE TRUE
  A     6   23
  B     8   30
  C     9   24</code></pre>
<pre><code>
    Pearson&#39;s Chi-squared test

data:  table(tmp2$treatment, tmp2$Anaemic0)
X-squared = 0.50895, df = 2, p-value = 0.7753</code></pre>
</div>
<p>Anaemic rates between treatments at endline</p>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
   
    FALSE TRUE
  A     9   20
  B     9   29
  C     6   27</code></pre>
<pre><code>
    Pearson&#39;s Chi-squared test

data:  table(tmp2$treatment, tmp2$Anaemic24)
X-squared = 1.4013, df = 2, p-value = 0.4963</code></pre>
</div>
<p>Anaemic rates between treatments at baseline and endline</p>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
   treatment Anaemic0 Anaemic24  n
1          A    FALSE     FALSE  4
2          A    FALSE      TRUE  2
3          A     TRUE     FALSE  5
4          A     TRUE      TRUE 18
5          B    FALSE     FALSE  5
6          B    FALSE      TRUE  3
7          B     TRUE     FALSE  4
8          B     TRUE      TRUE 26
9          C    FALSE     FALSE  3
10         C    FALSE      TRUE  6
11         C     TRUE     FALSE  3
12         C     TRUE      TRUE 21</code></pre>
</div>
<p>Treatment A</p>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
       
        FALSE TRUE
  FALSE     4    2
  TRUE      5   18</code></pre>
<pre><code>
    Fisher&#39;s Exact Test for Count Data

data:  z[, , &quot;A&quot;]
p-value = 0.05588
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
  0.7148579 93.9033412
sample estimates:
odds ratio 
  6.609192 </code></pre>
</div>
<p>Treatment B</p>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
       
        FALSE TRUE
  FALSE     5    3
  TRUE      4   26</code></pre>
<pre><code>
    Fisher&#39;s Exact Test for Count Data

data:  z[, , &quot;B&quot;]
p-value = 0.01013
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
  1.350045 92.748871
sample estimates:
odds ratio 
  9.878826 </code></pre>
</div>
<p>Treatment C</p>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
       
        FALSE TRUE
  FALSE     3    6
  TRUE      3   21</code></pre>
<pre><code>
    Fisher&#39;s Exact Test for Count Data

data:  z[, , &quot;C&quot;]
p-value = 0.3092
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
  0.3548773 32.2771518
sample estimates:
odds ratio 
  3.345343 </code></pre>
</div>
<p>Transitions between treatments</p>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
   
    FALSE-&gt;FALSE FALSE-&gt;TRUE TRUE-&gt;FALSE TRUE-&gt;TRUE
  A            4           2           5         18
  B            5           3           4         26
  C            3           6           3         21</code></pre>
<pre><code>
    Pearson&#39;s Chi-squared test

data:  zz
X-squared = 3.7589, df = 6, p-value = 0.7093</code></pre>
</div>
<p>Children that recovered from anaemia</p>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
    Pearson&#39;s Chi-squared test

data:  zz[, c(&quot;TRUE-&gt;FALSE&quot;, &quot;TRUE-&gt;TRUE&quot;)]
X-squared = 0.95138, df = 2, p-value = 0.6215</code></pre>
</div>
<p>Children that developed anaemia</p>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
    Pearson&#39;s Chi-squared test

data:  zz[, c(&quot;FALSE-&gt;FALSE&quot;, &quot;FALSE-&gt;TRUE&quot;)]
X-squared = 2.1272, df = 2, p-value = 0.3452</code></pre>
</div>
<p>Classifying as ‘recovered’ or ‘not recovered’ (including those that never go anaemia).</p>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
   
    not recovered recovered
  A            24         5
  B            34         4
  C            30         3</code></pre>
<pre><code>
    Pearson&#39;s Chi-squared test

data:  zzz
X-squared = 1.0971, df = 2, p-value = 0.5778</code></pre>
</div>
<h1 id="additional-information" class="appendix">Additional information</h1>
<p>The following are available on request:</p>
<ul>
<li>Full CSV tables of any data presented.</li>
<li>PDF/PNG files of any static plots.</li>
</ul>
<h1 id="session-info" class="appendix">Session info</h1>
<summary>The analysis and this document were prepared using the following software (click triangle to expand)</summary>
<details>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
─ Session info ─────────────────────────────────────────────────────
 setting  value                       
 version  R version 4.0.0 (2020-04-24)
 os       Ubuntu 18.04.5 LTS          
 system   x86_64, linux-gnu           
 ui       X11                         
 language en_AU:en                    
 collate  en_AU.UTF-8                 
 ctype    en_AU.UTF-8                 
 tz       Australia/Melbourne         
 date     2020-08-27                  

─ Packages ─────────────────────────────────────────────────────────
 ! package              * version  date       lib source        
 P AnnotationDbi          1.50.3   2020-07-25 [?] Bioconductor  
 P assertthat             0.2.1    2019-03-21 [?] CRAN (R 4.0.0)
 P backports              1.1.8    2020-06-17 [?] CRAN (R 4.0.0)
 P Biobase              * 2.48.0   2020-04-27 [?] Bioconductor  
 P BiocGenerics         * 0.34.0   2020-04-27 [?] Bioconductor  
 P BiocManager            1.30.10  2019-11-16 [?] CRAN (R 4.0.0)
 P BiocParallel         * 1.22.0   2020-04-27 [?] Bioconductor  
 P BiocStyle              2.16.0   2020-04-27 [?] Bioconductor  
 P bit                    4.0.4    2020-08-04 [?] CRAN (R 4.0.0)
 P bit64                  4.0.2    2020-07-30 [?] CRAN (R 4.0.0)
 P bitops                 1.0-6    2013-08-17 [?] CRAN (R 4.0.0)
 P blob                   1.2.1    2020-01-20 [?] CRAN (R 4.0.0)
 P cli                    2.0.2    2020-02-28 [?] CRAN (R 4.0.0)
 P colorspace             1.4-1    2019-03-18 [?] CRAN (R 4.0.0)
 P cowplot              * 1.0.0    2019-07-11 [?] CRAN (R 4.0.0)
 P crayon                 1.3.4    2017-09-16 [?] CRAN (R 4.0.0)
 P DBI                    1.1.0    2019-12-15 [?] CRAN (R 4.0.0)
 P DelayedArray         * 0.14.1   2020-07-14 [?] Bioconductor  
 P digest                 0.6.25   2020-02-23 [?] CRAN (R 4.0.0)
 P distill                0.8      2020-06-04 [?] CRAN (R 4.0.0)
 P dplyr                  1.0.1    2020-07-31 [?] CRAN (R 4.0.0)
 P edgeR                * 3.30.3   2020-06-02 [?] Bioconductor  
 P ellipsis               0.3.1    2020-05-15 [?] CRAN (R 4.0.0)
 P evaluate               0.14     2019-05-28 [?] CRAN (R 4.0.0)
 P fansi                  0.4.1    2020-01-08 [?] CRAN (R 4.0.0)
 P farver                 2.0.3    2020-01-16 [?] CRAN (R 4.0.0)
 P generics               0.0.2    2018-11-29 [?] CRAN (R 4.0.0)
 P GenomeInfoDb         * 1.24.2   2020-06-15 [?] Bioconductor  
 P GenomeInfoDbData       1.2.3    2020-05-21 [?] Bioconductor  
 P GenomicRanges        * 1.40.0   2020-04-27 [?] Bioconductor  
 P gghighlight          * 0.3.0    2020-03-29 [?] CRAN (R 4.0.0)
 P ggplot2              * 3.3.2    2020-06-19 [?] CRAN (R 4.0.0)
 P Glimma                 1.16.0   2020-04-27 [?] Bioconductor  
 P glue                   1.4.1    2020-05-13 [?] CRAN (R 4.0.0)
 P GO.db                  3.11.4   2020-07-22 [?] Bioconductor  
 P gtable                 0.3.0    2019-03-25 [?] CRAN (R 4.0.0)
 P here                 * 0.1      2017-05-28 [?] CRAN (R 4.0.0)
 P highr                  0.8      2019-03-20 [?] CRAN (R 4.0.0)
 P htmltools              0.5.0    2020-06-16 [?] CRAN (R 4.0.0)
 P IRanges              * 2.22.2   2020-05-21 [?] Bioconductor  
 P janitor                2.0.1    2020-04-12 [?] CRAN (R 4.0.0)
 P jsonlite               1.7.0    2020-06-25 [?] CRAN (R 4.0.0)
 P knitr                  1.29     2020-06-23 [?] CRAN (R 4.0.0)
 P labeling               0.3      2014-08-23 [?] CRAN (R 4.0.0)
 P lattice                0.20-41  2020-04-02 [5] CRAN (R 4.0.0)
 P lifecycle              0.2.0    2020-03-06 [?] CRAN (R 4.0.0)
 P limma                * 3.44.3   2020-06-12 [?] Bioconductor  
 P locfit                 1.5-9.4  2020-03-25 [?] CRAN (R 4.0.0)
 P lubridate              1.7.9    2020-06-08 [?] CRAN (R 4.0.0)
 P magrittr               1.5      2014-11-22 [?] CRAN (R 4.0.0)
 P Matrix                 1.2-18   2019-11-27 [5] CRAN (R 4.0.0)
 P matrixStats          * 0.56.0   2020-03-13 [?] CRAN (R 4.0.0)
 P memoise                1.1.0    2017-04-21 [?] CRAN (R 4.0.0)
 P msigdbr              * 7.1.1    2020-05-14 [?] CRAN (R 4.0.0)
 P munsell                0.5.0    2018-06-12 [?] CRAN (R 4.0.0)
 P org.Hs.eg.db           3.11.4   2020-06-15 [?] Bioconductor  
 P patchwork            * 1.0.1    2020-06-22 [?] CRAN (R 4.0.0)
 P pheatmap             * 1.0.12   2019-01-04 [?] CRAN (R 4.0.0)
 P pillar                 1.4.6    2020-07-10 [?] CRAN (R 4.0.0)
 P pkgconfig              2.0.3    2019-09-22 [?] CRAN (R 4.0.0)
 P purrr                  0.3.4    2020-04-17 [?] CRAN (R 4.0.0)
 P R6                     2.4.1    2019-11-12 [?] CRAN (R 4.0.0)
 P RColorBrewer           1.1-2    2014-12-07 [?] CRAN (R 4.0.0)
 P Rcpp                   1.0.5    2020-07-06 [?] CRAN (R 4.0.0)
 P RCurl                  1.98-1.2 2020-04-18 [?] CRAN (R 4.0.0)
 P rlang                  0.4.7    2020-07-09 [?] CRAN (R 4.0.0)
 P rmarkdown              2.3      2020-06-18 [?] CRAN (R 4.0.0)
 P rprojroot              1.3-2    2018-01-03 [?] CRAN (R 4.0.0)
 P RSQLite                2.2.0    2020-01-07 [?] CRAN (R 4.0.0)
 P S4Vectors            * 0.26.1   2020-05-16 [?] Bioconductor  
 P scales               * 1.1.1    2020-05-11 [?] CRAN (R 4.0.0)
 P sessioninfo            1.1.1    2018-11-05 [?] CRAN (R 4.0.0)
 P SingleCellExperiment * 1.10.1   2020-04-28 [?] Bioconductor  
 P snakecase              0.11.0   2019-05-25 [?] CRAN (R 4.0.0)
 P stringi                1.4.6    2020-02-17 [?] CRAN (R 4.0.0)
 P stringr                1.4.0    2019-02-10 [?] CRAN (R 4.0.0)
 P SummarizedExperiment * 1.18.2   2020-07-09 [?] Bioconductor  
 P tibble                 3.0.3    2020-07-10 [?] CRAN (R 4.0.0)
 P tidyr                  1.1.1    2020-07-31 [?] CRAN (R 4.0.0)
 P tidyselect             1.1.0    2020-05-11 [?] CRAN (R 4.0.0)
 P vctrs                  0.3.2    2020-07-15 [?] CRAN (R 4.0.0)
 P withr                  2.2.0    2020-04-20 [?] CRAN (R 4.0.0)
 P xfun                   0.16     2020-07-24 [?] CRAN (R 4.0.0)
 P XVector                0.28.0   2020-04-27 [?] Bioconductor  
 P yaml                   2.2.1    2020-02-01 [?] CRAN (R 4.0.0)
 P zlibbioc               1.34.0   2020-04-27 [?] Bioconductor  

[1] /home/peter/SCORE/C084_Hayman_Pasricha/renv/library/R-4.0/x86_64-pc-linux-gnu
[2] /tmp/RtmpVhm0n2/renv-system-library
[3] /usr/local/lib/R/site-library
[4] /usr/lib/R/site-library
[5] /usr/lib/R/library

 P ── Loaded and on-disk path mismatch.</code></pre>
</div>
</details>
<div class="layout-chunk" data-layout="l-body">

</div>
<div id="refs" class="references" role="doc-bibliography">
<div id="ref-law2014voom">
<p>Law, C. W., Y. Chen, W. Shi, and G. K. Smyth. 2014. “Voom: Precision Weights Unlock Linear Model Analysis Tools for Rna-Seq Read Counts.” <em>Genome Biol.</em> 15 (2): R29. <a href="https://www.ncbi.nlm.nih.gov/pubmed/24485249">https://www.ncbi.nlm.nih.gov/pubmed/24485249</a>.</p>
</div>
<div id="ref-robinson2010scaling">
<p>Robinson, M. D., and A. Oshlack. 2010. “A Scaling Normalization Method for Differential Expression Analysis of Rna-Seq Data.” <em>Genome Biol.</em> 11 (3): R25. <a href="https://www.ncbi.nlm.nih.gov/pubmed/20196867">https://www.ncbi.nlm.nih.gov/pubmed/20196867</a>.</p>
</div>
<div id="ref-smyth2004linear">
<p>Smyth, Gordon K. 2004. “Linear Models and Empirical Bayes Methods for Assessing Differential Expression in Microarray Experiments.” <em>Statistical Applications in Genetics and Molecular Biology</em> 3 (1).</p>
</div>
</div>
<section class="footnotes" role="doc-endnotes">
<hr />
<ol>
<li id="fn1" role="doc-endnote"><p>Furthermore, the 3’-tag sequencing of SCORE’s mini-bulk protocol basically negates the need to account for gene length differences.<a href="#fnref1" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
</ol>
</section>
<!--radix_placeholder_article_footer-->
<!--/radix_placeholder_article_footer-->
</div>

<div class="d-appendix">
</div>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

<!--radix_placeholder_site_after_body-->
<!--/radix_placeholder_site_after_body-->
<!--radix_placeholder_appendices-->
<div class="appendix-bottom"></div>
<script id="distill-bibliography" type="text/bibtex">
@Article{stegle2015computational,
    Author="Stegle, O.  and Teichmann, S. A.  and Marioni, J. C. ",
    Title="Computational and analytical challenges in single-cell transcriptomics",
    Journal="Nat. Rev. Genet.",
    Year="2015",
    Volume="16",
    Number="3",
    Pages="133--145",
    Month="Mar",
    URL={https://www.ncbi.nlm.nih.gov/pubmed/25628217}
}

@Article{islam2014quantitative,
    Author="Islam, S.  and Zeisel, A.  and Joost, S.  and La Manno, G.  and Zajac, P.  and Kasper, M.  and Lonnerberg, P.  and Linnarsson, S. ",
    Title="Quantitative single-cell RNA-seq with unique molecular identifiers",
    Journal="Nat. Methods",
    Year="2014",
    Volume="11",
    Number="2",
    Pages="163--166",
    Month="Feb"
}

@Article{huber2015orchestrating,
    Author="Huber, W.  and Carey, V. J.  and Gentleman, R.  and Anders, S.  and Carlson, M.  and Carvalho, B. S.  and Bravo, H. C.  and Davis, S.  and Gatto, L.  and Girke, T.  and Gottardo, R.  and Hahne, F.  and Hansen, K. D.  and Irizarry, R. A.  and Lawrence, M.  and Love, M. I.  and MacDonald, J.  and Obenchain, V.  and Oles, A. K.  and Pag\`es, H.  and Reyes, A.  and Shannon, P.  and Smyth, G. K.  and Tenenbaum, D.  and Waldron, L.  and Morgan, M. ",
    Title="{{O}rchestrating high-throughput genomic analysis with {B}ioconductor}",
    Journal="Nat. Methods",
    Year="2015",
    Volume="12",
    Number="2",
    Pages="115--121",
    Month="Feb"
}

@Article{brennecke2013accounting,
    Author="Brennecke, P.  and Anders, S.  and Kim, J. K.  and Ko\l{}odziejczyk, A. A.  and Zhang, X.  and Proserpio, V.  and Baying, B.  and Benes, V.  and Teichmann, S. A.  and Marioni, J. C.  and Heisler, M. G. ",
    Title="{{A}ccounting for technical noise in single-cell {R}{N}{A}-seq experiments}",
    Journal="Nat. Methods",
    Year="2013",
    Volume="10",
    Number="11",
    Pages="1093--1095",
    Month="Nov"
}

@Article{marinov2014singlecell,
    Author="Marinov, G. K.  and Williams, B. A.  and McCue, K.  and Schroth, G. P.  and Gertz, J.  and Myers, R. M.  and Wold, B. J. ",
    Title="{{F}rom single-cell to cell-pool transcriptomes: stochasticity in gene expression and {R}{N}{A} splicing}",
    Journal="Genome Res.",
    Year="2014",
    Volume="24",
    Number="3",
    Pages="496--510",
    Month="Mar"
}

@Article{vallejos2015basics,
    Author="Vallejos, C. A.  and Marioni, J. C.  and Richardson, S. ",
    Title="{{B}{A}{S}i{C}{S}: Bayesian analysis of single-cell sequencing data}",
    Journal="PLoS Comput. Biol.",
    Year="2015",
    Volume="11",
    Number="6",
    Pages="e1004333",
    Month="Jun"
}

@Article{trapnell2014dynamics,
    Author="Trapnell, C.  and Cacchiarelli, D.  and Grimsby, J.  and Pokharel, P.  and Li, S.  and Morse, M.  and Lennon, N. J.  and Livak, K. J.  and Mikkelsen, T. S.  and Rinn, J. L. ",
    Title="{{T}he dynamics and regulators of cell fate decisions are revealed by pseudotemporal ordering of single cells}",
    Journal="Nat. Biotechnol.",
    Year="2014",
    Volume="32",
    Number="4",
    Pages="381--386",
    Month="Apr"
}

@Article{kolod2015singlecell,
    Author="Ko\l{}odziejczyk, A. A.  and Kim, J. K.  and Tsang, J. C.  and Ilicic, T.  and Henriksson, J.  and Natarajan, K. N.  and Tuck, A. C.  and Gao, X.  and Buhler, M.  and Liu, P.  and Marioni, J. C.  and Teichmann, S. A. ",
    Title="{{S}ingle cell {R}{N}{A}-sequencing of pluripotent states unlocks modular transcriptional variation}",
    Journal="Cell Stem Cell",
    Year="2015",
    Volume="17",
    Number="4",
    Pages="471--485",
    Month="Oct"
}

@Article{anders2010differential,
    Author="Anders, S.  and Huber, W. ",
    Title="Differential expression analysis for sequence count data",
    Journal="Genome Biol.",
    Year="2010",
    Volume="11",
    Number="10",
    Pages="R106",
    URL={https://www.ncbi.nlm.nih.gov/pubmed/20979621}
}

@Article{robinson2010scaling,
    Author="Robinson, M. D.  and Oshlack, A. ",
    Title="A scaling normalization method for differential expression analysis of RNA-seq data",
    Journal="Genome Biol.",
    Year="2010",
    Volume="11",
    Number="3",
    Pages="R25",
    URL={https://www.ncbi.nlm.nih.gov/pubmed/20196867}
}

@Article{islam2011characterization,
    Author="Islam, S.  and Kjallquist, U.  and Moliner, A.  and Zajac, P.  and Fan, J. B.  and Lonnerberg, P.  and Linnarsson, S. ",
    Title="{{C}haracterization of the single-cell transcriptional landscape by highly multiplex {R}{N}{A}-seq}",
    Journal="Genome Res.",
    Year="2011",
    Volume="21",
    Number="7",
    Pages="1160--1167",
    Month="Jul"
}

@Article{wilson2015combined,
    Author="Wilson, N. K.  and Kent, D. G.  and Buettner, F.  and Shehata, M.  and Macaulay, I. C.  and Calero-Nieto, F. J.  and Sanchez Castillo, M.  and Oedekoven, C. A.  and Diamanti, E.  and Schulte, R.  and Ponting, C. P.  and Voet, T.  and Caldas, C.  and Stingl, J.  and Green, A. R.  and Theis, F. J.  and Gottgens, B. ",
    Title="{{C}ombined single-cell functional and gene expression analysis resolves heterogeneity within stem cell populations}",
    Journal="Cell Stem Cell",
    Year="2015",
    Volume="16",
    Number="6",
    Pages="712--724",
    Month="Jun"
}

@Article{picelli2014fulllength,
    Author="Picelli, S.  and Faridani, O. R.  and Bjorklund, A. K.  and Winberg, G.  and Sagasser, S.  and Sandberg, R. ",
    Title="{{F}ull-length {R}{N}{A}-seq from single cells using {S}mart-seq2}",
    Journal="Nat Protoc",
    Year="2014",
    Volume="9",
    Number="1",
    Pages="171--181",
    Month="Jan"
}

@Article{pollen2014lowcoverage,
    Author="Pollen, A. A.  and Nowakowski, T. J.  and Shuga, J.  and Wang, X.  and Leyrat, A. A.  and Lui, J. H.  and Li, N.  and Szpankowski, L.  and Fowler, B.  and Chen, P.  and Ramalingam, N.  and Sun, G.  and Thu, M.  and Norris, M.  and Lebofsky, R.  and Toppani, D.  and Kemp, D. W.  and Wong, M.  and Clerkson, B.  and Jones, B. N.  and Wu, S.  and Knutsson, L.  and Alvarado, B.  and Wang, J.  and Weaver, L. S.  and May, A. P.  and Jones, R. C.  and Unger, M. A.  and Kriegstein, A. R.  and West, J. A. ",
    Title="{{L}ow-coverage single-cell m{R}{N}{A} sequencing reveals cellular heterogeneity and activated signaling pathways in developing cerebral cortex}",
    Journal="Nat. Biotechnol.",
    Year="2014",
    Volume="32",
    Number="10",
    Pages="1053--1058",
    Month="Oct"
}

@Article{love2015rnaseq,
    Author="Love, M. I.  and Anders, S.  and Kim, V.  and Huber, W. ",
    Title="{{R}{N}{A}-{S}eq workflow: gene-level exploratory analysis and differential expression}",
    Journal="F1000Res",
    Year="2015",
    Volume="4",
    Pages="1070"
}

@Article{liao2014featurecounts,
    Author="Liao, Y.  and Smyth, G. K.  and Shi, W. ",
    Title="{feature{C}ounts: an efficient general purpose program for assigning sequence reads to genomic features}",
    Journal="Bioinformatics",
    Year="2014",
    Volume="30",
    Number="7",
    Pages="923--930",
    Month="Apr"
}

@Article{liao2013subread,
    Author="Liao, Y.  and Smyth, G. K.  and Shi, W. ",
    Title="{{T}he {S}ubread aligner: fast, accurate and scalable read mapping by seed-and-vote}",
    Journal="Nucleic Acids Res.",
    Year="2013",
    Volume="41",
    Number="10",
    Pages="e108",
    Month="May"
}

@Article{law2014voom,
    Author="Law, C. W.  and Chen, Y.  and Shi, W.  and Smyth, G. K. ",
    Title="voom: Precision weights unlock linear model analysis tools for RNA-seq read counts",
    Journal="Genome Biol.",
    Year="2014",
    Volume="15",
    Number="2",
    Pages="R29",
    URL={https://www.ncbi.nlm.nih.gov/pubmed/24485249}
}

@Article{love2014moderated,
    Author="Love, M. I.  and Huber, W.  and Anders, S. ",
    Title="Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2",
    Journal="Genome Biol.",
    Year="2014",
    Volume="15",
    Number="12",
    Pages="550",
    URL={https://www.ncbi.nlm.nih.gov/pubmed/25516281}
}

@article{van2008visualizing,
    title={Visualizing data using t-SNE},
    author={Van der Maaten, L. and Hinton, G.},
    journal={J. Mach. Learn. Res.},
    volume={9},
    number={2579-2605},
    pages={85},
    year={2008},
    URL={http://www.jmlr.org/papers/v9/vandermaaten08a.html}
}

@Article{scialdone2015computational,
    Author="Scialdone, A.  and Natarajan, K. N.  and Saraiva, L. R.  and Proserpio, V.  and Teichmann, S. A.  and Stegle, O.  and Marioni, J. C.  and Buettner, F. ",
    Title="Computational assignment of cell-cycle stage from single-cell transcriptome data",
    Journal="Methods",
    Year="2015",
    Volume="85",
    Pages="54--61",
    Month="Sep",
    URL = {https://www.ncbi.nlm.nih.gov/pubmed/26142758},
}

@Article{kim2015characterizing,
    Author="Kim, J. K.  and Ko\l{}odziejczyk, A. A.  and Illicic, T.  and Teichmann, S. A.  and Marioni, J. C. ",
    Title="{{C}haracterizing noise structure in single-cell {R}{N}{A}-seq distinguishes genuine from technical stochastic allelic expression}",
    Journal="Nat. Commun.",
    Year="2015",
    Volume="6",
    Pages="8687"
}

@Article{mccarthy2012differential,
    Author="McCarthy, D. J.  and Chen, Y.  and Smyth, G. K. ",
    Title="{{D}ifferential expression analysis of multifactor {R}{N}{A}-{S}eq experiments with respect to biological variation}",
    Journal="Nucleic Acids Res.",
    Year="2012",
    Volume="40",
    Number="10",
    Pages="4288--4297",
    Month="May"
}

@Article{angel1991role,
    Author="Angel, P.  and Karin, M. ",
    Title="{{T}he role of {J}un, {F}os and the {A}{P}-1 complex in cell-proliferation and transformation}",
    Journal="Biochim. Biophys. Acta",
    Year="1991",
    Volume="1072",
    Number="2-3",
    Pages="129--157",
    Month="Dec"
}

@Article{langfelder2008defining,
    Author="Langfelder, P.  and Zhang, B.  and Horvath, S. ",
    Title="{{D}efining clusters from a hierarchical cluster tree: the {D}ynamic {T}ree {C}ut package for {R}}",
    Journal="Bioinformatics",
    Year="2008",
    Volume="24",
    Number="5",
    Pages="719--720",
    Month="Mar"
}

@Article{ilicic2016classification,
    Author="Ilicic, T.  and Kim, J. K.  and Ko\l{}odziejczyk, A. A.  and Bagger, F. O.  and McCarthy, D. J.  and Marioni, J. C.  and Teichmann, S. A. ",
    Title="{{C}lassification of low quality cells from single-cell {R}{N}{A}-seq data}",
    Journal="Genome Biol.",
    Year="2016",
    Volume="17",
    Number="1",
    Pages="29"
}

@Article{heng2008immunological,
    Author="Heng, T. S.  and Painter, M. W.  and Elpek, K.  and Lukacs-Kornek, V.  and Mauermann, N.  and Turley, S. J.  and Koller, D.  and Kim, F. S.  and Wagers, A. J.  and Asinovski, N.  and Davis, S.  and Fassett, M.  and Feuerer, M.  and Gray, D. H.  and Haxhinasto, S.  and Hill, J. A.  and Hyatt, G.  and Laplace, C.  and Leatherbee, K.  and Mathis, D.  and Benoist, C.  and Jianu, R.  and Laidlaw, D. H.  and Best, J. A.  and Knell, J.  and Goldrath, A. W.  and Jarjoura, J.  and Sun, J. C.  and Zhu, Y.  and Lanier, L. L.  and Ergun, A.  and Li, Z.  and Collins, J. J.  and Shinton, S. A.  and Hardy, R. R.  and Friedline, R.  and Sylvia, K.  and Kang, J. ",
    Title="{{T}he {I}mmunological {G}enome {P}roject: networks of gene expression in immune cells}",
    Journal="Nat. Immunol.",
    Year="2008",
    Volume="9",
    Number="10",
    Pages="1091--1094",
    Month="Oct"
}

@Article{zeisel2015brain,
    Author="Zeisel, A.  and Munoz-Manchado, A. B.  and Codeluppi, S.  and Lonnerberg, P.  and La Manno, G.  and Jureus, A.  and Marques, S.  and Munguba, H.  and He, L.  and Betsholtz, C.  and Rolny, C.  and Castelo-Branco, G.  and Hjerling-Leffler, J.  and Linnarsson, S. ",
    Title="{{B}rain structure. {C}ell types in the mouse cortex and hippocampus revealed by single-cell {R}{N}{A}-seq}",
    Journal="Science",
    Year="2015",
    Volume="347",
    Number="6226",
    Pages="1138--1142",
    Month="Mar"
}

@Article{robinson2010edgeR,
    Author="Robinson, M. D.  and McCarthy, D. J.  and Smyth, G. K. ",
    Title="edgeR: a Bioconductor package for differential expression analysis of digital gene expression data",
    Journal="Bioinformatics",
    Year="2010",
    Volume="26",
    Number="1",
    Pages="139--140",
    Month="Jan"
}

@Article{ritchie2015limma,
    Author="Ritchie, M. E.  and Phipson, B.  and Wu, D.  and Hu, Y.  and Law, C. W.  and Shi, W.  and Smyth, G. K. ",
    Title="limma powers differential expression analyses for RNA-sequencing and microarray studies",
    Journal="Nucleic Acids Res.",
    Year="2015",
    Volume="43",
    Number="7",
    Pages="e47",
    Month="Apr"
}

@Article{julia2015sincell,
    Author="Julia, M.  and Telenti, A.  and Rausell, A. ",
    Title="{{S}incell: an {R}/{B}ioconductor package for statistical assessment of cell-state hierarchies from single-cell {R}{N}{A}-seq}",
    Journal="Bioinformatics",
    Year="2015",
    Volume="31",
    Number="20",
    Pages="3380--3382",
    Month="Oct"
}

@Article{buettner2015computational,
    Author="Buettner, F.  and Natarajan, K. N.  and Casale, F. P.  and Proserpio, V.  and Scialdone, A.  and Theis, F. J.  and Teichmann, S. A.  and Marioni, J. C.  and Stegle, O. ",
    Title="{{C}omputational analysis of cell-to-cell heterogeneity in single-cell {R}{N}{A}-sequencing data reveals hidden subpopulations of cells}",
    Journal="Nat. Biotechnol.",
    Year="2015",
    Volume="33",
    Number="2",
    Pages="155--160",
    Month="Feb"
}

@Article{mahata2014singlecell,
    Author="Mahata, B.  and Zhang, X.  and Ko\l{}odziejczyk, A. A.  and Proserpio, V.  and Haim-Vilmovsky, L.  and Taylor, A. E.  and Hebenstreit, D.  and Dingler, F. A.  and Moignard, V.  and Gottgens, B.  and Arlt, W.  and McKenzie, A. N.  and Teichmann, S. A. ",
    Title="{{S}ingle-cell {R}{N}{A} sequencing reveals {T} helper cells synthesizing steroids de novo to contribute to immune homeostasis}",
    Journal="Cell Rep.",
    Year="2014",
    Volume="7",
    Number="4",
    Pages="1130--1142",
    Month="May"
}

@Article{phipson2010permutation,
    Author="Phipson, B.  and Smyth, G. K. ",
    Title="{{P}ermutation {P}-values should never be zero: calculating exact {P}-values when permutations are randomly drawn}",
    Journal="Stat. Appl. Genet. Mol. Biol.",
    Year="2010",
    Volume="9",
    Pages="Article 39"
}

@incollection{chen2014differential,
    title="Differential expression analysis of complex RNA-seq experiments using edgeR",
    author={Chen, Y. and Lun, A. T. L. and Smyth, G. K.},
    booktitle={Statistical analysis of next generation sequencing data},
    editor={Datta, S. and Nettleton, D.},
    pages={51--74},
    year={2014},
    publisher={Springer}
}

@article{hicks2015widespread,
    author = {Hicks, S. C. and Teng, M. and Irizarry, R. A.},
    title = "On the widespread and critical impact of systematic bias and batch effects in single-cell RNA-Seq data",
    year = {2015},
    doi = {10.1101/025528},
    publisher = {Cold Spring Harbor Labs Journals},
    URL = {http://biorxiv.org/content/early/2015/08/25/025528},
    eprint = {http://biorxiv.org/content/early/2015/08/25/025528.full.pdf},
    journal = {bioRxiv}
}

@Article{leng2015oscope,
    Author="Leng, N.  and Chu, L. F.  and Barry, C.  and Li, Y.  and Choi, J.  and Li, X.  and Jiang, P.  and Stewart, R. M.  and Thomson, J. A.  and Kendziorski, C. ",
    Title="{{O}scope identifies oscillatory genes in unsynchronized single-cell {R}{N}{A}-seq experiments}",
    Journal="Nat. Methods",
    Year="2015",
    Volume="12",
    Number="10",
    Pages="947--950",
    Month="Oct"
}

@Article{fan2016characterizing,
    Author="Fan, J.  and Salathia, N.  and Liu, R.  and Kaeser, G. E.  and Yung, Y. C.  and Herman, J. L.  and Kaper, F.  and Fan, J. B.  and Zhang, K.  and Chun, J.  and Kharchenko, P. V. ",
    Title="{{C}haracterizing transcriptional heterogeneity through pathway and gene set overdispersion analysis}",
    Journal="Nat. Methods",
    Year="2016",
    Volume="13",
    Number="3",
    Pages="241--244",
    Month="Mar"
}

@Article{lund2012detecting,
    Author="Lund, S. P.  and Nettleton, D.  and McCarthy, D. J.  and Smyth, G. K. ",
    Title="{{D}etecting differential expression in {R}{N}{A}-sequence data using quasi-likelihood with shrunken dispersion estimates}",
    Journal="Stat Appl Genet Mol Biol",
    Year="2012",
    Volume="11",
    Number="5",
    Pages=" "
}

@Article{angerer2016destiny,
    Author="Angerer, P.  and Haghverdi, L.  and Buttner, M.  and Theis, F. J.  and Marr, C.  and Buettner, F. ",
    Title="{destiny: diffusion maps for large-scale single-cell data in {R}}",
    Journal="Bioinformatics",
    Year="2016",
    Volume="32",
    Number="8",
    Pages="1241--1243",
}

@Article{lun2016pooling,
    Author="Lun, A. T.  and Bach, K.  and Marioni, J. C. ",
    Title="Pooling across cells to normalize single-cell RNA sequencing data with many zero counts}",
    Journal="Genome Biol.",
    Year="2016",
    Volume="17",
    Pages="75",
    Month="Apr",
    URL={https://www.ncbi.nlm.nih.gov/pubmed/27122128}
}

@Article{zhu2006gata3,
    Author="Zhu, J. and Yamane, H. and Cote-Sierra, J. and Guo, L. and Paul, W. E.",
    Title="{{G}{A}{T}{A}-3 promotes {T}h2 responses through three different mechanisms: induction of {T}h2 cytokine production, selective growth of {T}h2 cells and inhibition of {T}h1 cell-specific factors}",
    Journal="Cell Res.",
    Year="2006",
    Volume="16",
    Number="1",
    Pages="3--10",
    Month="Jan"
}

@article{berger1996bioequivalence,
    author = "Berger, R. L. and Hsu, J. C.",
    doi = "10.1214/ss/1032280304",
    fjournal = "Statistical Science",
    journal = "Statist. Sci.",
    month = "11",
    number = "4",
    pages = "283--319",
    publisher = "The Institute of Mathematical Statistics",
    title = "Bioequivalence trials, intersection-union tests and equivalence confidence sets",
    url = "http://dx.doi.org/10.1214/ss/1032280304",
    volume = "11",
    year = "1996"
}

@Article{santos2015cyclebase,
    Author="Santos, A.  and Wernersson, R.  and Jensen, L. J. ",
    Title="{{C}yclebase 3.0: a multi-organism database on cell-cycle regulation and phenotypes}",
    Journal="Nucleic Acids Res.",
    Year="2015",
    Volume="43",
    Number="Database issue",
    Pages="D1140--1144",
    Month="Jan"
}

@article {tung2016batch,
    Author="Tung, P. Y.  and Blischak, J. D.  and Hsiao, C. J.  and Knowles, D. A.  and Burnett, J. E.  and Pritchard, J. K.  and Gilad, Y. ",
    Title="Batch effects and the effective design of single-cell gene expression studies",
    Journal="Sci. Rep.",
    Year="2017",
    Volume="7",
    Pages="39921",
    Month="Jan"
}

@Article{bray2016near,
    Author="Bray, N. L.  and Pimentel, H.  and Melsted, P.  and Pachter, L. ",
    Title="{{N}ear-optimal probabilistic {R}{N}{A}-seq quantification}",
    Journal="Nat. Biotechnol.",
    Year="2016",
    Volume="34",
    Number="5",
    Pages="525--527",
    Month="May"
}

@article{patro2015accurate,
    author = {Patro, Rob and Duggal, Geet and Kingsford, Carl},
    title = {Accurate, fast, and model-aware transcript expression quantification with Salmon},
    year = {2015},
    doi = {10.1101/021592},
    publisher = {Cold Spring Harbor Labs Journals},
    journal = {bioRxiv}
}

@Article{ji2016tscan,
    Author="Ji, Z.  and Ji, H. ",
    Title="{{T}{S}{C}{A}{N}: {P}seudo-time reconstruction and evaluation in single-cell {R}{N}{A}-seq analysis}",
    Journal="Nucleic Acids Res.",
    Year="2016",
    Volume="44",
    Number="13",
    Pages="e117",
    Month="Jul"
}

@Article{klein2015droplet,
    Author="Klein, A. M.  and Mazutis, L.  and Akartuna, I.  and Tallapragada, N.  and Veres, A.  and Li, V.  and Peshkin, L.  and Weitz, D. A.  and Kirschner, M. W. ",
    Title="{{D}roplet barcoding for single-cell transcriptomics applied to embryonic stem cells}",
    Journal="Cell",
    Year="2015",
    Volume="161",
    Number="5",
    Pages="1187--1201",
    Month="May"
}

@Article{macosko2015highly,
    Author="Macosko, E. Z.  and Basu, A.  and Satija, R.  and Nemesh, J.  and Shekhar, K.  and Goldman, M.  and Tirosh, I.  and Bialas, A. R.  and Kamitaki, N.  and Martersteck, E. M.  and Trombetta, J. J.  and Weitz, D. A.  and Sanes, J. R.  and Shalek, A. K.  and Regev, A.  and McCarroll, S. A. ",
    Title="{{H}ighly parallel genome-wide expression profiling of individual cells using nanoliter droplets}",
    Journal="Cell",
    Year="2015",
    Volume="161",
    Number="5",
    Pages="1202--1214",
    Month="May"
}

@Article{bourgon2010independent,
    Author="Bourgon, R.  and Gentleman, R.  and Huber, W. ",
    Title="Independent filtering increases detection power for high-throughput experiments",
    Journal="Proc. Natl. Acad. Sci. U.S.A.",
    Year="2010",
    Volume="107",
    Number="21",
    Pages="9546--9551",
    Month="May",
    URL={https://www.ncbi.nlm.nih.gov/pubmed/20460310}
}

@Article{finak2015mast,
    Author="Finak, G.  and McDavid, A.  and Yajima, M.  and Deng, J.  and Gersuk, V.  and Shalek, A. K.  and Slichter, C. K.  and Miller, H. W.  and McElrath, M. J.  and Prlic, M.  and Linsley, P. S.  and Gottardo, R. ",
    Title="{{M}{A}{S}{T}: a flexible statistical framework for assessing transcriptional changes and characterizing heterogeneity in single-cell {R}{N}{A} sequencing data}",
    Journal="Genome Biol.",
    Year="2015",
    Volume="16",
    Pages="278"
}

@Article{conboy2007cell,
    Author="Conboy, C. M.  and Spyrou, C.  and Thorne, N. P.  and Wade, E. J.  and Barbosa-Morais, N. L.  and Wilson, M. D.  and Bhattacharjee, A.  and Young, R. A.  and Tavare, S.  and Lees, J. A.  and Odom, D. T. ",
    Title="Cell cycle genes are the evolutionarily conserved targets of the E2F4 transcription factor",
    Journal="PLoS ONE",
    Year="2007",
    Volume="2",
    Number="10",
    Pages="e1061",
    URL={https://www.ncbi.nlm.nih.gov/pubmed/17957245}
}

@Article{bertoli2013control,
    Author="Bertoli, C.  and Skotheim, J. M.  and de Bruin, R. A. ",
    Title="Control of cell cycle transcription during G1 and S phases",
    Journal="Nat. Rev. Mol. Cell Biol.",
    Year="2013",
    Volume="14",
    Number="8",
    Pages="518--528",
    Month="Aug",
    URL={{C}ontrol of cell cycle transcription during {G}1 and {S} phases}
}

@Article{chen2016from,
    Author="Chen, Y.  and Lun, A. T.  and Smyth, G. K. ",
    Title="{{F}rom reads to genes to pathways: differential expression analysis of {R}{N}{A}-{S}eq experiments using {R}subread and the edge{R} quasi-likelihood pipeline}",
    Journal="F1000Res",
    Year="2016",
    Volume="5",
    Pages="1438"
}

@Article{coller2006new,
    Author="Coller, H. A.  and Sang, L.  and Roberts, J. M. ",
    Title="{{A} new description of cellular quiescence}",
    Journal="PLoS Biol.",
    Year="2006",
    Volume="4",
    Number="3",
    Pages="e83",
    Month="Mar"
}

@Article{mccarthy2017scater,
    Author="McCarthy, D. J.  and Campbell, K. R.  and Lun, A. T.  and Wills, Q. F. ",
    Title="{{S}cater: pre-processing, quality control, normalization and visualization of single-cell {R}{N}{A}-seq data in {R}}",
    Journal="Bioinformatics",
    Year="2017",
    Volume="33",
    Number="8",
    Pages="1179--1186",
    Month="Apr"
}

@article{simes1986improved,
    author = {Simes, R. J.},
    title = {An improved Bonferroni procedure for multiple tests of significance},
    volume = {73},
    number = {3},
    pages = {751--754},
    year = {1986},
    journal = {Biometrika}
}

@Article{xu2015identification,
    Author="Xu, C.  and Su, Z. ",
    Title="Identification of cell types from single-cell transcriptomes using a novel clustering method",
    Journal="Bioinformatics",
    Year="2015",
    Volume="31",
    Number="12",
    Pages="1974--1980",
    Month="Jun"
}

@Article{lun2017assessing,
    Author="Lun, A. T. L.  and Calero-Nieto, F. J.  and Haim-Vilmovsky, L.  and Gottgens, B.  and Marioni, J. C. ",
    Title="Assessing the reliability of spike-in normalization for analyses of single-cell RNA sequencing data",
    Journal="Genome Res.",
    Year="2017",
    Volume="27",
    Number="11",
    Pages="1795--1806",
    Month="Nov"
}

@Article{yang2016comparative,
    Author="Yang, Z.  and Algesheimer, R.  and Tessone, C. J. ",
    Title="{A comparative analysis of community detection algorithms on artificial networks}",
    Journal="Sci. Rep.",
    Year="2016",
    Volume="6",
    Pages="30750",
    Month="Aug"
}

@Article{smith2017umitools,
    Author="Smith, T.  and Heger, A.  and Sudbery, I. ",
    Title="{{U}{M}{I}-tools: modeling sequencing errors in {U}nique {M}olecular {I}dentifiers to improve quantification accuracy}",
    Journal="Genome Res.",
    Year="2017",
    Volume="27",
    Number="3",
    Pages="491--499",
    Month="Mar"
}

@Article{tasic2016adult,
    Author="Tasic, B.  and Menon, V.  and Nguyen, T. N.  and Kim, T. K.  and Jarsky, T.  and Yao, Z.  and Levi, B.  and Gray, L. T.  and Sorensen, S. A.  and Dolbeare, T.  and Bertagnolli, D.  and Goldy, J.  and Shapovalova, N.  and Parry, S.  and Lee, C.  and Smith, K.  and Bernard, A.  and Madisen, L.  and Sunkin, S. M.  and Hawrylycz, M.  and Koch, C.  and Zeng, H. ",
    Title="{{A}dult mouse cortical cell taxonomy revealed by single cell transcriptomics}",
    Journal="Nat. Neurosci.",
    Year="2016",
    Volume="19",
    Number="2",
    Pages="335--346",
    Month="Feb"
}

@Article{wajapeyee2010senescence,
    Author="Wajapeyee, N.  and Wang, S. Z.  and Serra, R. W.  and Solomon, P. D.  and Nagarajan, A.  and Zhu, X.  and Green, M. R. ",
    Title="{{S}enescence induction in human fibroblasts and hematopoietic progenitors by leukemogenic fusion proteins}",
    Journal="Blood",
    Year="2010",
    Volume="115",
    Number="24",
    Pages="5057--5060",
    Month="Jun"
}

@Article{zeng2012largescale,
    Author="Zeng, H.  and Shen, E. H.  and Hohmann, J. G.  and Oh, S. W.  and Bernard, A.  and Royall, J. J.  and Glattfelder, K. J.  and Sunkin, S. M.  and Morris, J. A.  and Guillozet-Bongaarts, A. L.  and Smith, K. A.  and Ebbert, A. J.  and Swanson, B.  and Kuan, L.  and Page, D. T.  and Overly, C. C.  and Lein, E. S.  and Hawrylycz, M. J.  and Hof, P. R.  and Hyde, T. M.  and Kleinman, J. E.  and Jones, A. R. ",
    Title="{{L}arge-scale cellular-resolution gene profiling in human neocortex reveals species-specific molecular signatures}",
    Journal="Cell",
    Year="2012",
    Volume="149",
    Number="2",
    Pages="483--496",
    Month="Apr"
}

@Article{grun2016denovo,
    Author="Grun, D.  and Muraro, M. J.  and Boisset, J. C.  and Wiebrands, K.  and Lyubimova, A.  and Dharmadhikari, G.  and van den Born, M.  and van Es, J.  and Jansen, E.  and Clevers, H.  and de Koning, E. J. P.  and van Oudenaarden, A. ",
    Title="{{D}e {N}ovo {P}rediction of {S}tem {C}ell {I}dentity using {S}ingle-{C}ell {T}ranscriptome {D}ata}",
    Journal="Cell Stem Cell",
    Year="2016",
    Volume="19",
    Number="2",
    Pages="266--277",
    Month="08"
}

@article{haghverdi2018batch,
    Author="Haghverdi, L.  and Lun, A. T. L.  and Morgan, M. D.  and Marioni, J. C. ",
    Title="Batch effects in single-cell RNA-sequencing data are corrected by matching mutual nearest neighbors",
    Journal="Nat. Biotechnol.",
    Year="2018",
    Month="Apr"
}

@Article{muraro2016singlecell,
    Author="Muraro, M. J.  and Dharmadhikari, G.  and Grun, D.  and Groen, N.  and Dielen, T.  and Jansen, E.  and van Gurp, L.  and Engelse, M. A.  and Carlotti, F.  and de Koning, E. J.  and van Oudenaarden, A. ",
    Title="{{A} {S}ingle-{C}ell {T}ranscriptome {A}tlas of the {H}uman {P}ancreas}",
    Journal="Cell Syst",
    Year="2016",
    Volume="3",
    Number="4",
    Pages="385--394",
    Month="Oct"
}

@Article{segerstolpe2016singlecell,
    Author="Segerstolpe, A.  and Palasantza, A.  and Eliasson, P.  and Andersson, E. M.  and Andreasson, A. C.  and Sun, X.  and Picelli, S.  and Sabirsh, A.  and Clausen, M.  and Bjursell, M. K.  and Smith, D. M.  and Kasper, M.  and Ammala, C.  and Sandberg, R. ",
    Title="{{S}ingle-{C}ell {T}ranscriptome {P}rofiling of {H}uman {P}ancreatic {I}slets in {H}ealth and {T}ype 2 {D}iabetes}",
    Journal="Cell Metab.",
    Year="2016",
    Volume="24",
    Number="4",
    Pages="593--607",
    Month="Oct"
}

@Article{horn1965rationale,
    author="Horn, J. L.",
    title="A rationale and test for the number of factors in factor analysis",
    journal="Psychometrika",
    year="1965",
    month="Jun",
    day="01",
    volume="30",
    number="2",
    pages="179--185",
    issn="1860-0980",
    doi="10.1007/BF02289447",
    url="https://doi.org/10.1007/BF02289447"
}

@Article{buja1992remarks,
    Author="Buja, A.  and Eyuboglu, N. ",
    Title="{{R}emarks on {P}arallel {A}nalysis}",
    Journal="Multivariate Behav Res",
    Year="1992",
    Volume="27",
    Number="4",
    Pages="509--540",
    Month="Oct"
}

@Article{zheng2017massively,
    Author="Zheng, G. X.  and Terry, J. M.  and Belgrader, P.  and Ryvkin, P.  and Bent, Z. W.  and Wilson, R.  and Ziraldo, S. B.  and Wheeler, T. D.  and McDermott, G. P.  and Zhu, J.  and Gregory, M. T.  and Shuga, J.  and Montesclaros, L.  and Underwood, J. G.  and Masquelier, D. A.  and Nishimura, S. Y.  and Schnall-Levin, M.  and Wyatt, P. W.  and Hindson, C. M.  and Bharadwaj, R.  and Wong, A.  and Ness, K. D.  and Beppu, L. W.  and Deeg, H. J.  and McFarland, C.  and Loeb, K. R.  and Valente, W. J.  and Ericson, N. G.  and Stevens, E. A.  and Radich, J. P.  and Mikkelsen, T. S.  and Hindson, B. J.  and Bielas, J. H. ",
    Title="{{M}assively parallel digital transcriptional profiling of single cells}",
    Journal="Nat Commun",
    Year="2017",
    Volume="8",
    Pages="14049",
    Month="Jan"
}

@article{lun2018distinguishing,
    author = {Lun, A. and Riesenfeld, S. and Andrews, T. and Dao, T. P. and Gomes, T. and {{participants in the 1\textsuperscript{st} Human Cell Atlas Jamboree}} and Marioni, J.},
    title = {Distinguishing cells from empty droplets in droplet-based single-cell RNA sequencing data},
    year = {2018},
    doi = {10.1101/234872},
    publisher = {Cold Spring Harbor Laboratory},
    URL = {https://www.biorxiv.org/content/early/2018/04/04/234872},
    eprint = {https://www.biorxiv.org/content/early/2018/04/04/234872.full.pdf},
    journal = {bioRxiv}
}

@article {wolf2017graph,
    author = {Wolf, F. Alexander and Hamey, Fiona and Plass, Mireya and Solana, Jordi and Dahlin, Joakim S. and Gottgens, Berthold and Rajewsky, Nikolaus and Simon, Lukas and Theis, Fabian J.},
    title = {Graph abstraction reconciles clustering with trajectory inference through a topology preserving map of single cells},
    year = {2017},
    doi = {10.1101/208819},
    publisher = {Cold Spring Harbor Laboratory},
    URL = {https://www.biorxiv.org/content/early/2017/10/25/208819},
    eprint = {https://www.biorxiv.org/content/early/2017/10/25/208819.full.pdf},
    journal = {bioRxiv}
}

@Article{lawlor2017singlecell,
    Author="Lawlor, N.  and George, J.  and Bolisetty, M.  and Kursawe, R.  and Sun, L.  and Sivakamasundari, V.  and Kycia, I.  and Robson, P.  and Stitzel, M. L. ",
    Title="{{S}ingle-cell transcriptomes identify human islet cell signatures and reveal cell-type-specific expression changes in type 2 diabetes}",
    Journal="Genome Res.",
    Year="2017",
    Volume="27",
    Number="2",
    Pages="208--222",
    Month="02"
}

@Article{guimaraes2016patterns,
   Author="Guimaraes, J. C.  and Zavolan, M. ",
   Title="{{P}atterns of ribosomal protein expression specify normal and malignant human cells}",
   Journal="Genome Biol.",
   Year="2016",
   Volume="17",
   Number="1",
   Pages="236",
   Month="11"
}

@Article{glare2002betaactin,
   Author="Glare, E. M.  and Divjak, M.  and Bailey, M. J.  and Walters, E. H. ",
   Title="{beta-{A}ctin and {G}{A}{P}{D}{H} housekeeping gene expression in asthmatic airways is variable and not suitable for normalising m{R}{N}{A} levels}",
   Journal="Thorax",
   Year="2002",
   Volume="57",
   Number="9",
   Pages="765--770",
   Month="Sep"
}

@Article{nazari2015gapdh,
   Author="Nazari, F.  and Parham, A.  and Maleki, A. F. ",
   Title="{{G}{A}{P}{D}{H}, \beta-actin and \beta-microglobulin, as three common reference genes, are not reliable for gene expression studies in equine adipose- and marrow-derived mesenchymal stem cells}",
   Journal="J Anim Sci Technol",
   Year="2015",
   Volume="57",
   Pages="18"
}

@Article{soneson2018bias,
    Author="Soneson, C.  and Robinson, M. D. ",
    Title="Bias, robustness and scalability in single-cell differential expression analysis",
    Journal="Nat. Methods",
    URL = {https://www.ncbi.nlm.nih.gov/pubmed/29481549},
    Year="2018",
    Volume="15",
    Number="4",
    Pages="255--261",
    Month="Apr"
}

@Article{soneson2015differential,
    Author="Soneson, C.  and Love, M. I.  and Robinson, M. D. ",
    Title="{{D}ifferential analyses for {R}{N}{A}-seq: transcript-level estimates improve gene-level inferences}",
    Journal="F1000Res",
    Year="2015",
    Volume="4",
    Pages="1521"
}

@article{stoeckius2017hashing,
    author = {Stoeckius, M. and Zheng, S. and Houck-Loomis, B. and Hao, S. and Yeung, B. and Smibert, P. and Satija, R.},
    title = {Cell "hashing" with barcoded antibodies enables multiplexing and doublet detection for single cell genomics},
    year = {2017},
    doi = {10.1101/237693},
    publisher = {Cold Spring Harbor Laboratory},
    URL = {https://www.biorxiv.org/content/early/2017/12/21/237693},
    eprint = {https://www.biorxiv.org/content/early/2017/12/21/237693.full.pdf},
    journal = {bioRxiv}
}

@Article{kang2018multiplexed,
    Author="Kang, H. M.  and Subramaniam, M.  and Targ, S.  and Nguyen, M.  and Maliskova, L.  and McCarthy, E.  and Wan, E.  and Wong, S.  and Byrnes, L.  and Lanata, C. M.  and Gate, R. E.  and Mostafavi, S.  and Marson, A.  and Zaitlen, N.  and Criswell, L. A.  and Ye, C. J. ",
    Title="{{M}ultiplexed droplet single-cell {R}{N}{A}-sequencing using natural genetic variation}",
    Journal="Nat. Biotechnol.",
    Year="2018",
    Volume="36",
    Number="1",
    Pages="89--94",
    Month="Jan"
}

@Article{dahlin2018single,
    Author="Dahlin, J. S.  and Hamey, F. K.  and Pijuan-Sala, B.  and Shepherd, M.  and Lau, W. W. Y.  and Nestorowa, S.  and Weinreb, C.  and Wolock, S.  and Hannah, R.  and Diamanti, E.  and Kent, D. G.  and Gottgens, B.  and Wilson, N. K. ",
    Title="{{A} single-cell hematopoietic landscape resolves 8 lineage trajectories and defects in {K}it mutant mice}",
    Journal="Blood",
    Year="2018",
    Volume="131",
    Number="21",
    Pages="e1-e11",
    Month="May"
}

@Article{bach2017differentiation,
    Author="Bach, K.  and Pensa, S.  and Grzelak, M.  and Hadfield, J.  and Adams, D. J.  and Marioni, J. C.  and Khaled, W. T. ",
    Title="{{D}ifferentiation dynamics of mammary epithelial cells revealed by single-cell {R}{N}{A} sequencing}",
    Journal="Nat Commun",
    Year="2017",
    Volume="8",
    Number="1",
    Pages="2128",
    Month="Dec"
}

@article{tian2018scpipe,
  title={scPipe: A flexible R/Bioconductor preprocessing pipeline for single-cell RNA-sequencing data},
  author={Tian, Luyi and Su, Shian and Dong, Xueyi and Amann-Zalcenstein, Daniela and Biben, Christine and Seidi, Azadeh and Hilton, Douglas J and Naik, Shalin H and Ritchie, Matthew E},
  journal={PLoS computational biology},
  volume={14},
  number={8},
  pages={e1006361},
  year={2018},
  publisher={Public Library of Science},
  URL={https://www.ncbi.nlm.nih.gov/pubmed/30096152}
}

@article{kiselev2017sc3,
  title={SC3: consensus clustering of single-cell RNA-seq data},
  author={Kiselev, Vladimir Yu and Kirschner, Kristina and Schaub, Michael T and Andrews, Tallulah and Yiu, Andrew and Chandra, Tamir and Natarajan, Kedar N and Reik, Wolf and Barahona, Mauricio and Green, Anthony R and others},
  journal={Nature methods},
  volume={14},
  number={5},
  pages={483},
  year={2017},
  publisher={Nature Publishing Group},
  URL={https://www.ncbi.nlm.nih.gov/pubmed/28346451}
}

@ARTICLE{McInnes2018-hy,
  title         = "UMAP: Uniform Manifold Approximation and Projection for
                   Dimension Reduction",
  author        = "McInnes, Leland and Healy, John and Melville, James",
  abstract      = "UMAP (Uniform Manifold Approximation and Projection) is a
                   novel manifold learning technique for dimension reduction.
                   UMAP is constructed from a theoretical framework based in
                   Riemannian geometry and algebraic topology. The result is a
                   practical scalable algorithm that applies to real world
                   data. The UMAP algorithm is competitive with t-SNE for
                   visualization quality, and arguably preserves more of the
                   global structure with superior run time performance.
                   Furthermore, UMAP has no computational restrictions on
                   embedding dimension, making it viable as a general purpose
                   dimension reduction technique for machine learning.",
  month         =  feb,
  year          =  2018,
  url           = "http://arxiv.org/abs/1802.03426",
  archivePrefix = "arXiv",
  eprint        = "1802.03426",
  primaryClass  = "stat.ML",
  arxivid       = "1802.03426"
}

@article{lun2017overcoming,
  title={Overcoming confounding plate effects in differential expression analyses of single-cell RNA-seq data},
  author={Lun, Aaron TL and Marioni, John C},
  journal={Biostatistics},
  volume={18},
  number={3},
  pages={451--464},
  year={2017},
  publisher={Oxford University Press}
}

@Article{lun2017testing,
    Author="Lun, A. T. L.  and Richard, A. C.  and Marioni, J. C. ",
    Title="Testing for differential abundance in mass cytometry data",
    Journal="Nat. Methods",
    Year="2017",
    Volume="14",
    Number="7",
    Pages="707--709",
    Month="Jul"
}

@Article{chen2016reads,
    Author="Chen, Y.  and Lun, A. T.  and Smyth, G. K. ",
    Title="{{F}rom reads to genes to pathways: differential expression analysis of {R}{N}{A}-{S}eq experiments using {R}subread and the edge{R} quasi-likelihood pipeline}",
    Journal="F1000Res",
    Year="2016",
    Volume="5",
    Pages="1438"
}

@Article{phipson2016robust,
    Author="Phipson, B. and Lee, S. and Majewski, I. J. and Alexander, W. S. and Smyth, G. K.",
    year = 2016,
    title = "Robust hyperparameter estimation protects against hypervariable genes and improves power to detect differential expression",
    Journal="Ann. Appl. Stat.",
    Volume=10,
    Number=2,
    Pages="946-963"
}

@Article{aran2019reference,
   Author="Aran, D.  and Looney, A. P.  and Liu, L.  and Wu, E.  and Fong, V.  and Hsu, A.  and Chak, S.  and Naikawadi, R. P.  and Wolters, P. J.  and Abate, A. R.  and Butte, A. J.  and Bhattacharya, M. ",
   Title="Reference-based analysis of lung single-cell sequencing reveals a transitional profibrotic macrophage",
   Journal="Nat. Immunol.",
   Year="2019",
   Volume="20",
   Number="2",
   Pages="163--172",
   Month="02"
}

@article{Benayoun2019,
author = {Benayoun, B{\'{e}}r{\'{e}}nice A. and Pollina, Elizabeth A. and Singh, Param Priya and Mahmoudi, Salah and Harel, Itamar and Casey, Kerriann M. and Dulken, Ben W. and Kundaje, Anshul and Brunet, Anne},
doi = {10.1101/gr.240093.118},
journal = {Genome Research},
volume = {29},
pages = {697-709},
title = {Remodeling of epigenome and transcriptome landscapes with aging in mice reveals widespread induction of inflammatory responses},
year = {2019}
}

@article{ImmGenRef,
    author = {Heng, Tracy S.P. and Painter, Michio W. and Elpek, Kutlu and Lukacs-Kornek, Veronika and Mauermann, Nora and Turley, Shannon J. and Koller, Daphne and Kim, Francis S. and Wagers, Amy J. and Asinovski, Natasha and Davis, Scott and Fassett, Marlys and Feuerer, Markus and Gray, Daniel H.D. and Haxhinasto, Sokol and Hill, Jonathan A. and Hyatt, Gordon and Laplace, Catherine and Leatherbee, Kristen and Mathis, Diane and Benoist, Christophe and Jianu, Radu and Laidlaw, David H. and Best, J. Adam and Knell, Jamie and Goldrath, Ananda W. and Jarjoura, Jessica and Sun, Joseph C. and Zhu, Yanan and Lanier, Lewis L. and Ergun, Ayla and Li, Zheng and Collins, James J. and Shinton, Susan A. and Hardy, Richard R. and Friedline, Randall and Sylvia, Katelyn and Kang, Joonsoo},
    journal = {Nature Immunology},
    volume = {9},
    issue = {10},
    pages = {1091-4},
    doi = {10.1038/ni1008-1091},
    title = {The immunological genome project: Networks of gene expression in immune cells},
    year = {2008}
}

@article{choi2018haemopedia,
  title={Haemopedia RNA-seq: a database of gene expression during haematopoiesis in mice and humans},
  author={Choi, Jarny and Baldwin, Tracey M and Wong, Mae and Bolden, Jessica E and Fairfax, Kirsten A and Lucas, Erin C and Cole, Rebecca and Biben, Christine and Morgan, Clare and Ramsay, Kerry A and others},
  journal={Nucleic acids research},
  volume={47},
  number={D1},
  pages={D780--D785},
  year={2018},
  publisher={Oxford University Press}
}

@Article{mccarthy2009treat,
    Author="McCarthy, D. J.  and Smyth, G. K. ",
    Title="{Testing significance relative to a fold-change threshold is a TREAT}",
    Journal="Bioinformatics",
    Year="2009",
    Volume="25",
    Number="6",
    Pages="765--771",
    Month="Mar"
}

@article{hpcaRef,
author = {Mabbott, Neil A. and Baillie, J. K. and Brown, Helen and Freeman, Tom C. and Hume, David A.},
doi = {10.1186/1471-2164-14-632},
journal = {BMC Genomics},
volume = {14},
title = {{An expression atlas of human primary cells: Inference of gene function from coexpression networks}},
year = {2013}
}

@article{blueprintRef,
    author = {Martens, Joost H A and Stunnenberg, Hendrik G.},
    doi = {10.3324/haematol.2013.094243},
    journal = {Haematologica},
    title = {{BLUEPRINT: Mapping human blood cell epigenomes}},
    year = {2013},
    volume = {98},
    pages = {1487-1489}
}

@article{encodeRef,
author = {{The ENCODE Project Consortium}},
doi = {10.1038/nature11247},
issn = {14764687},
journal = {Nature},
title = {{An integrated encyclopedia of DNA elements in the human genome}},
year = {2012}
}

@article{lun2016step,
  title={A step-by-step workflow for low-level analysis of single-cell {RNA-seq} data},
  author={Lun, A. T. L. and McCarthy, D. J. and Marioni, J. C.},
  journal ={F1000Res.},
  volume={5},
  month={aug},
  year={2016}
}

@Article{richard2018tcell,
   Author="Richard, A. C.  and Lun, A. T. L.  and Lau, W. W. Y.  and Gottgens, B.  and Marioni, J. C.  and Griffiths, G. M. ",
   Title="T cell cytolytic capacity is independent of initial stimulation strength",
   Journal="Nat. Immunol.",
   Year="2018",
   Volume="19",
   Number="8",
   Pages="849--858",
   Month="08"
}

@Article{leek2012sva,
   Author="Leek, J. T.  and Johnson, W. E.  and Parker, H. S.  and Jaffe, A. E.  and Storey, J. D. ",
   Title="The sva package for removing batch effects and other unwanted variation in high-throughput experiments",
   Journal="Bioinformatics",
   Year="2012",
   Volume="28",
   Number="6",
   Pages="882--883",
   Month="Mar"
}

@Article{butler2018integrating,
   Author="Butler, A.  and Hoffman, P.  and Smibert, P.  and Papalexi, E.  and Satija, R. ",
   Title="Integrating single-cell transcriptomic data across different conditions, technologies, and species",
   Journal="Nat. Biotechnol.",
   Year="2018",
   Volume="36",
   Number="5",
   Pages="411--420",
   Month="06"
}

@Article{lin2019scmerge,
   Author="Lin, Y.  and Ghazanfar, S.  and Wang, K. Y. X.  and Gagnon-Bartsch, J. A.  and Lo, K. K.  and Su, X.  and Han, Z. G.  and Ormerod, J. T.  and Speed, T. P.  and Yang, P.  and Yang, J. Y. H. ",
   Title="scMerge leverages factor analysis, stable expression, and pseudoreplication to merge multiple single-cell RNA-seq datasets",
   Journal="Proc. Natl. Acad. Sci. U.S.A.",
   Year="2019",
   Volume="116",
   Number="20",
   Pages="9775--9784",
   Month="May"
}

@Article{scialdone2016resolving,
   Author="Scialdone, A.  and Tanaka, Y.  and Jawaid, W.  and Moignard, V.  and Wilson, N. K.  and Macaulay, I. C.  and Marioni, J. C.  and Gottgens, B. ",
   Title="Resolving early mesoderm diversification through single-cell expression profiling",
   Journal="Nature",
   Year="2016",
   Volume="535",
   Number="7611",
   Pages="289--293",
   Month="07"
}

@article{crowell2019discovery,
	author = {Crowell, H. L. and Soneson, C. and Germain, P.-L. and Calini, D. and Collin, L. and Raposo, C. and Malhotra, D. and Robinson, M. D.},
	title = {On the discovery of population-specific state transitions from multi-sample multi-condition single-cell RNA sequencing data},
	year = {2019},
	doi = {10.1101/713412},
	journal = {bioRxiv}
}

@article{wu2010roast,
  title={ROAST: rotation gene set tests for complex microarray experiments},
  author={Wu, Di and Lim, Elgene and Vaillant, Fran{\c{c}}ois and Asselin-Labat, Marie-Liesse and Visvader, Jane E and Smyth, Gordon K},
  journal={Bioinformatics},
  volume={26},
  number={17},
  pages={2176--2182},
  year={2010},
  publisher={Oxford University Press}
}

@article{gu2016depletion,
  title="Depletion of Abundant Sequences by Hybridization (DASH): using Cas9 to remove unwanted high-abundance species in sequencing libraries and molecular counting applications",
  author="Gu, Wei and Crawford, Emily D and O’Donovan, BD and Wilson, Michael R and Chow, Eric D and Retallack, Hanna and DeRisi, Joseph L",
  journal={Genome biology},
  volume={17},
  number={1},
  pages={41},
  year={2016},
  publisher={Springer}
}

@article{law2016rna,
  title="RNA-seq analysis is easy as 1-2-3 with limma, Glimma and edgeR",
  author="Law, Charity W and Alhamdoosh, Monther and Su, Shian and Dong, Xueyi and Tian, Luyi and Smyth, Gordon K and Ritchie, Matthew E",
  journal={F1000Research},
  volume={5},
  year={2016},
  publisher={Faculty of 1000 Ltd}
}

@article{smyth2004linear,
  title="Linear models and empirical bayes methods for assessing differential expression in microarray experiments",
  author="Smyth, Gordon K",
  journal={Statistical applications in genetics and molecular biology},
  volume={3},
  number={1},
  year={2004},
  publisher={De Gruyter}
}
</script>
<!--/radix_placeholder_appendices-->
<!--radix_placeholder_navigation_after_body-->
<!--/radix_placeholder_navigation_after_body-->

</body>

</html>
