---
title: "Multi-sample comparisons of the Hayman/Pasricha (C084) ZIPT mini-bulk data set"
description: |
author:
  - name: Peter Hickey
    url: https://peterhickey.org
    affiliation: WEHI SCORE
    affiliation_url: https://www.wehi.edu.au/people/shalin-naik/3310/score
date: "`r Sys.Date()`"
editor_options:
  chunk_output_type: console
bibliography: ref.bib
---

```{r setup}
library(SingleCellExperiment)
library(here)
library(edgeR)
library(BiocParallel)
library(scales)
library(ggplot2)
library(cowplot)
library(patchwork)
library(pheatmap)

# NOTE: Using >= 4 cores seizes up my laptop. Can use more on RStudio server.
options(
  "mc.cores" = 
    ifelse(Sys.info()[["nodename"]] == "rstudio-1.hpc.wehi.edu.au", 8L, 2L))

register(MulticoreParam(workers = getOption("mc.cores")))

source(here("code", "helper_functions.R"))

knitr::opts_chunk$set(
  fig.path = "C084_Hayman_Pasricha.multi-sample_comparisons_files/")
```

# Preparing the data

We start from the preprocessed *SingleCellExperiment* object created in ['Preprocessing the Hayman/Pasricha (C084) ZIPT mini-bulk data sett'](C084_Hayman_Pasricha.preprocess.html).

```{r}
sce <- readRDS(
  here("data", "SCEs", "C084_Hayman_Pasricha.preprocessed.SCE.rds"))

# Some useful colours
plate_number_colours <- setNames(
  unique(sce$plate_number_colours),
  unique(names(sce$plate_number_colours)))
treatment_colours <- setNames(
  unique(sce$treatment_colours),
  unique(names(sce$treatment_colours)))
timepoint_colours <- setNames(
  unique(sce$timepoint_colours),
  unique(names(sce$timepoint_colours)))
sex_colours <- setNames(
  unique(sce$sex_colours),
  unique(names(sce$sex_colours)))

# Some useful gene sets
mito_set <- rownames(sce)[any(rowData(sce)$ENSEMBL.SEQNAME == "MT")]

ribo_set <- grep("^RP(S|L)", rownames(sce), value = TRUE)
# NOTE: A more curated approach for identifying ribosomal protein genes 
#       (https://github.com/Bioconductor/OrchestratingSingleCellAnalysis-base/blob/ae201bf26e3e4fa82d9165d8abf4f4dc4b8e5a68/feature-selection.Rmd#L376-L380)
library(msigdbr)
c2_sets <- msigdbr(species = "Homo sapiens", category = "C2")
ribo_set <- union(
  ribo_set,
  c2_sets[c2_sets$gs_name == "KEGG_RIBOSOME", ]$gene_symbol)

e <- new.env()
load(url("http://bioinf.wehi.edu.au/software/GenderGenes/GenderGenes.RData"), e)
sex_set <- rownames(sce)[
  any(rowData(sce)$NCBI.ENTREZID %in% c(e$XiEgenes, e$msYgenes))]

pseudogene_set <- rownames(sce)[
  any(grepl("pseudogene", rowData(sce)$ENSEMBL.GENEBIOTYPE))]

hemo_set <- rownames(sce)[
  any(grepl("hemoglobin subunit", rowData(sce)$NCBI.GENENAME))]
```

We convert the data to a *DGEList* object in preparation for analysis with `r BiocStyle::Biocpkg("edgeR")`.

```{r}
x <- DGEList(
  counts = as.matrix(counts(sce)),
  samples = colData(sce),
  group = factor(paste0(sce$treatment, ".", sce$timepoint)),
  genes = flattenDF(rowData(sce)))
```

# Data pre-processing

## Transformations from the raw-scale

For differential expression and related analyses, gene expression is rarely considered at the level of raw counts since libraries sequenced at a greater depth will result in higher counts.
Rather, it is common practice to transform raw counts onto a scale that accounts for such library size differences.
Popular transformations include counts per million (CPM), log2-counts per million (log-CPM), reads per kilobase of transcript per million (RPKM), and fragments per kilobase of transcript per million (FPKM).

In our analyses, CPM and log-CPM transformations are used regularly although they do not account for gene length differences as RPKM and FPKM values do. 
Whilst RPKM and FPKM values can just as well be used, CPM and log-CPM values can be calculated using a counts matrix alone and will suffice for the type of comparisons we are interested in^[Furthermore, the 3'-tag sequencing of SCORE's mini-bulk protocol basically negates the need to account for gene length differences.].
Assuming that there are no differences in isoform usage between conditions, differential expression analyses look at gene expression changes between conditions rather than comparing expression across multiple genes or drawing conclusions on absolute levels of expression.
In other words, gene lengths remain constant for comparisons of interest and any observed differences are a result of changes in condition rather than changes in gene length.

```{r}
cpm <- cpm(x)
lcpm <- cpm(x, log = TRUE)
```

A CPM value of 1 for a gene equates to having `r number(min(x$samples$lib.size / 1e6), accuracy = 0.01)` counts in the sample with the lowest sequencing depth (`r colnames(x)[which.min(x$samples$sum)]`, library size approx. `r number(min(x$samples$lib.size / 1e6), accuracy = 0.01)` million) or `r number(max(x$samples$lib.size / 1e6), accuracy = 0.01)` counts in the sample with the greatest sequencing depth (`r colnames(x)[which.max(x$samples$sum)]`, library size approx. `r number(max(x$samples$lib.size / 1e6), accuracy = 0.01)` million).

```{r}
L <- mean(x$samples$lib.size) * 1e-6
M <- median(x$samples$lib.size) * 1e-6
```

The log-CPM values will be used for exploratory plots.
The log-CPM values use an offset to the CPM values before converting to the log2-scale.
By default, the offset is $2/L$ where 2 is the 'prior count' and $L$ is the average library size in millions, so the log-CPM values are related to the CPM values by $log_{2}(CPM + 2/L)$.
This calculation ensures that any two read counts with identical CPM values will also have identical log-CPM values.
The prior count avoids taking the logarithm of zero, and also reduces spurious variability for genes with very low counts by shrinking all the inter-sample log-fold-changes towards zero, something that is helpful for exploratory plotting.

For this dataset, the average library size is about `r number(L, accuracy = 0.01)` million, so $L \approx$ `r number(L, accuracy = 0.01)` and the minimum log-CPM value for each sample becomes $log_{2}(CPM + 2/L) \approx$ `r number(log2(2 / L), accuracy = 0.01)`
In other words, a count of zero for this data maps to a log-CPM value of `r number(log2(2 / L), accuracy = 0.01)` after adding the prior count or offset.

## Removing genes that are lowly expressed

```{r}
stopifnot(sum(matrixStats::rowAlls(x$counts == 0)) == 0)
```

All datasets will include a mix of genes that are expressed and those that are not expressed.
Whilst it is of interest to examine genes that are expressed in one condition but not in another, some genes are unexpressed throughout all samples.
We have already removed all genes in this dataset that have zero counts across all `r ncol(x)` samples.

Plotting the distribution log-CPM values shows that a sizeable proportion of genes within each sample are either unexpressed or lowly-expressed with log-CPM values that are small (Figure \@ref(fig:lcpm-density)).

Genes that do not have a worthwhile number of reads in any sample should be filtered out of the downstream analyses.
There are several reasons for this.
From a biological point of view, genes that not expressed at a biologically meaningful level in any condition are not of interest and are therefore best ignored.
From a statistical point of view, removing low count genes allows the mean-variance relationship in the data to be estimated with greater reliability and also reduces the number of statistical tests that need to be carried out in downstream analyses looking at differential expression.

The `filterByExpr()` function in the `r BiocStyle::Biocpkg("edgeR")` package provides an automatic way to filter genes, while keeping as many genes as possible with worthwhile counts.

```{r}
keep.exprs <- filterByExpr(x, group = x$samples$group)
x <- x[keep.exprs, , keep.lib.sizes = FALSE]
```

By default, the function keeps genes with about 10 read counts or more in a minimum number of samples, where the number of samples is chosen according to the minimum group sample size^The actual filtering uses CPM values rather than counts in order to avoid giving preference to samples with large library sizes.].
For this dataset, the median library size is about `r number(M, accuracy = 0.01)`) million and 10 / `r number(M, accuracy = 0.01)`, big.mark = ",") approx. `r number(10 / M, accuracy = 0.1)`, so the `filterByExpr()` function keeps genes that have a CPM of `r number(10 / M, accuracy = 0.1)` or more in at least `r min(table(x$samples$group))` samples.
A biologically interesting gene should be expressed in at least `r min(table(x$samples$group))` samples because all the groups have at least `r min(table(x$samples$group))` replicates within each plate.
The cutoffs used depend on the sequencing depth and on the experimental design. If the library sizes had been larger then a lower CPM cutoff would have been chosen, because larger library sizes provide better resolution to explore more genes at lower expression levels.
Alternatively, smaller library sizes decrease our ability to explore marginal genes and hence would have led to a higher CPM cutoff.

Using this criterion, the number of genes is reduced to `r number(sum(keep.exprs), big.mark = ",")`, about `r percent(sum(keep.exprs) / length(keep.exprs))` of the number that we started with (panel B of the Figure \@ref(fig:lcpm-density)). 

```{r lcpm-density, fig.cap = "The density of log-CPM values for raw pre-filtered data (A) and post-filtered data (B) are shown for 36 randomly chosen samples. Dotted vertical lines mark the log-CPM threshold used in the filtering step."}
lcpm.cutoff <- log2(10 / M + 2 / L)
set.seed(666)
nsamples <- 36
j <- sort(sample(ncol(x), nsamples))
samplenames <- colnames(x)[j]
col <- palette.colors(nsamples, "Polychrome 36")

par(mfrow = c(1, 2))
plot(
  density(lcpm[, j[1]]),
  col = col[1],
  lwd = 2,
  ylim = c(0, 2.6),
  las = 2,
  main = "",
  xlab = "")
title(main = "A. Raw data", xlab = "Log-cpm")
abline(v = lcpm.cutoff, lty = 3)
for (i in seq_along(j)[-1]) {
  den <- density(lcpm[, j[i]])
  lines(den$x, den$y, col = col[i], lwd = 2)
}
legend("topright", samplenames, text.col = col, bty = "n", cex = 0.5, ncol = 4)

lcpm <- cpm(x, log = TRUE)
plot(
  density(lcpm[, 1]),
  col = col[1],
  lwd = 2,
  ylim = c(0, 2.6),
  las = 2,
  main = "",
  xlab = "")
title(main = "B. Filtered data", xlab = "Log-cpm")
abline(v = lcpm.cutoff, lty = 3)
for (i in seq_along(j)[-1]) {
den <- density(lcpm[, j[i]])
  lines(den$x, den$y, col = col[i], lwd = 2)
}
legend("topright", samplenames, text.col = col, bty = "n", cex = 0.5, ncol = 4)
```

## Normalising gene expression distributions

During the sample preparation or sequencing process, external factors that are not of biological interest can affect the expression of individual samples.
For example, samples processed in the first batch of an experiment can have higher expression overall when compared to samples processed in a second batch. 
It is assumed that all samples should have a similar range and distribution of expression values.
Normalisation is required to ensure that the expression distributions of each sample are similar across the entire experiment.

Any plot showing the per sample expression distributions, such as a density or boxplot, is useful in determining whether any samples are dissimilar to others. 
Distributions of log-CPM values appear to be similar throughout all samples within this dataset (panel B of Figure \@ref(fig:lcpm-density)).

Nonetheless, normalisation by the method of trimmed mean of M-values (TMM) [@robinson2010scaling] is performed using the `calcNormFactors()` function in `r BiocStyle::Biocpkg("edgeR")`.
The normalisation factors calculated here are used as a scaling factor for the library sizes.

```{r}
x <- calcNormFactors(x, method = "TMM")
```

For this dataset the effect of TMM-normalisation is mild, as evident in the magnitude of the scaling factors, which are all relatively close to 1 (median = `r number(median(x$samples$norm.factors), 0.01)`, IQR = `r number(iqr(x$samples$norm.factors), 0.01)`, 5% and 95% quantiles = `r glue::glue_collapse(number(quantile(x$samples$norm.factors, c(0.05, 0.95)), accuracy = 0.01), sep = " and ")`).

Figure \@ref(fig:lcpm-boxplot) shows the expression distribution of 36 randomly chosen samples for unnormalised and normalised data, where distributions are noticeably different pre-normalisation and are similar post-normalisation.

```{r lcpm-boxplot, fig.cap = "Boxplots of log-CPM values showing expression distributions for unnormalised data (A) and normalised data (B) for 36 randomly chosen samples."}
set.seed(666)
nsamples <- 36
j <- sort(sample(ncol(x), nsamples))
samplenames <- colnames(x)[j]
col <- palette.colors(nsamples, "Polychrome 36")

par(mfrow = c(1, 2))
boxplot(lcpm[, j], las = 2, col = col, main = "", cex = 0.1, xaxt = "n")
title(main = "A. Unnormalised data", ylab = "Log-cpm")

lcpm <- cpm(x, log = TRUE)
boxplot(lcpm[, j], las = 2, col = col, main = "", cex = 0.1, xaxt = "n")
title(main = "B. Normalised data", ylab = "Log-cpm")
```

## Unsupervised clustering of samples

In our opinion, one of the most important exploratory plots to examine for gene expression analyses is the multi-dimensional scaling (MDS) plot, or similar.
The plot shows similarities and dissimilarities between samples in an unsupervised manner so that one can have an idea of the extent to which differential expression can be detected before carrying out formal tests.
Ideally, samples would cluster well within the primary condition of interest, and any sample straying far from its group could be identified and followed up for sources of error or extra variation.
If present, technical replicates should lie very close to one another.

Such a plot can be made in `r BiocStyle::Biocpkg("limma")` using the `plotMDS()` function.
The first dimension represents the leading-fold-change that best separates samples and explains the largest proportion of variation in the data, with subsequent dimensions having a smaller effect and being orthogonal to the ones before it. 
When experimental design involves multiple factors, it is recommended that each factor is examined over several dimensions.
If samples cluster by a given factor in any of these dimensions, it suggests that the factor contributes to expression differences and is worth including in the linear modelling.
On the other hand, factors that show little or no effect may be left out of downstream analysis.

In this dataset, samples can be seen to cluster strongly by experimental batch over dimension 1.
This dominates any variation due to timepoint or treatment.
Keeping in mind that the first dimension explains the largest proportion of variation in the data, notice that the range of values over the dimensions become smaller as we move to higher dimensions.

```{r mds, fig.cap = "MDS plots of log-CPM values over dimensions 1 and 2 with samples coloured by batch and over dimensions 3 and 4 with samples coloured by treatment and timepoint. Distances on the plot correspond to the leading fold-change, which is the average (root-mean-square) log2-fold-change for the 500 genes most divergent between each pair of samples by default.", fig.asp = 3 / 2}
lcpm <- cpm(x, log = TRUE)
col_batch <- plate_number_colours[x$samples$plate_number]
col_treatment <- treatment_colours[x$samples$treatment]
col_timepoint <- timepoint_colours[x$samples$timepoint]

par(mfrow = c(3, 2))
plotMDS(lcpm, pch = 16, col = col_batch)
legend(
  "bottom",
  col = plate_number_colours,
  legend = levels(x$samples$plate_number),
  pch = 16,
  title = "batch")
plotMDS(lcpm, pch = 16, col = col_batch, dim = c(3, 4))

plotMDS(lcpm, pch = 16, col = col_treatment)
legend(
  "bottom",
  col = treatment_colours,
  legend = levels(x$samples$treatment),
  pch = 16,
  title = "treatment")
plotMDS(lcpm, pch = 16, col = col_treatment, dim = c(3, 4))

plotMDS(lcpm, pch = 16, col = col_timepoint)
legend(
  "bottom",
  col = timepoint_colours,
  legend = levels(x$samples$timepoint),
  pch = 16,
  title = "timepoint")
plotMDS(lcpm, pch = 16, col = col_timepoint, dim = c(3, 4))
```

Datasets where samples do not cluster by experimental group may show little or no evidence of differential expression in the downstream analysis.

# Differential expression analysis

## Creating a design matrix and contrasts

In this study, the primary interest is to see which genes are expressed at different levels between the different treatments and timepoints.
In our analysis, linear models are fitted to the data with the assumption that the underlying data is normally distributed.
To get started, a design matrix is set up with the `group` (i.e. the combination of `timepoint` and `treatment`), `id` (i.e. subject ID), and `plate_number` (i.e. batch) information.

<aside>
Of secondary interest is to see which genes have their expression associated with additional coveriates (e.g., weight for age at endline). We will use different design matrices and contrasts to answer these questions.
</aside>

```{r}
design <- model.matrix(~0 + group + id + plate_number, x$samples)
colnames(design) <- sub("plate\\_number|group", "", colnames(design))
```

Contrasts for pairwise comparisons between `treatment` and `timepoint`, within each `id` and blocking on `plate_number`, are set up in `r BiocStyle::Biocpkg("limma")` using the `makeContrasts()` function.

```{r}
contr.matrix <- makeContrasts(
  A.endline_vs_A.baseline = A.endline - A.baseline,
  B.endline_vs_B.baseline = B.endline - B.baseline,
  C.endline_vs_C.baseline = C.endline - C.baseline,
  A.endline_vs_B.endline = A.endline - B.endline,
  A.endline_vs_C.endline = A.endline - C.endline,
  B.endline_vs_C.endline = B.endline - C.endline,
  levels = colnames(design))
```

A key strength of `r BiocStyle::Biocpkg("limma")`'s linear modelling approach, is the ability accommodate arbitrary experimental complexity.
Simple designs through to more complicated factorial designs and models with interaction terms can be handled relatively easily.

<aside>
We will make use of more complicated models in some of the analyses of secondary outcomes.
</aside>

## Removing heteroscedascity from count data

It has been shown that for RNA-seq count data, the variance is not independent of the mean [@law2014voom] â€“ this is true of raw counts or when transformed to log-CPM values.
Methods that model counts using a Negative Binomial distribution assume a quadratic mean-variance relationship.
In `r BiocStyle::Biocpkg("limma")`, linear modelling is carried out on the log-CPM values which are assumed to be normally distributed and the mean-variance relationship is accommodated using precision weights calculated by the `voom()` function.

`voom()` converts raw counts to log-CPM values by automatically extracting library sizes and normalisation factors.

The mean-variance relationship of log-CPM values for this dataset is shown in the left-hand panel of Figure \@ref(fig:voom).
Typically, the voom-plot shows a decreasing trend between the means and variances resulting from a combination of technical variation in the sequencing experiment and biological variation amongst the replicate samples from different cell populations.
Experiments with high biological variation usually result in flatter trends, where variance values plateau at high expression values.
Experiments with low biological variation tend to result in sharp decreasing trends.

Moreover, the voom-plot provides a visual check on the level of filtering performed upstream.
If filtering of lowly-expressed genes is insufficient, a drop in variance levels can be observed at the low end of the expression scale due to very small counts. 
If this is observed, one should return to the earlier filtering step and increase the expression threshold applied to the dataset.

```{r voom, fig.cap = "Means (x-axis) and variances (y-axis) of each gene are plotted to show the dependence between the two before voom is applied to the data (left panel) and how the trend is removed after voom precision weights are applied to the data (right panel). The plot on the left is created within the `voom()` function which extracts residual variances from fitting linear models to log-CPM transformed data. Variances are then rescaled to quarter-root variances (or square-root of standard deviations) and plotted against the average log2 count for each gene. The plot on the right is created using `plotSA()` which plots log2 residual standard deviations against mean log-CPM values. In both plots, each black dot represents a gene. On the left plot, the red curve shows the estimated mean-variance trend used to compute the voom weights. On the right plot, the average log2 residual standard deviation estimated by the empirical Bayes algorithm is marked by a horizontal blue line."}
par(mfrow = c(1, 2))
v <- voom(x, design, plot = TRUE)
vfit <- lmFit(v, design)
vfit <- contrasts.fit(vfit, contrasts = contr.matrix)
efit <- eBayes(vfit)
plotSA(efit, main = "Final model: Mean-variance trend")
```

## Fitting linear models for comparisons of interest

Linear modelling in `r BiocStyle::Biocpkg("limma")` fits a separate model to the expression values for each gene.
Next, empirical Bayes moderation is carried out by borrowing information across all the genes to obtain more precise estimates of gene-wise variability [@smyth2004linear].
The model's residual variances are plotted against average expression values in Figure \@ref(fig:voom).
It can be seen from this plot that the variance is no longer dependent on the mean expression level.

# Results

## Figure 1

### Endline pairwise analyses between treatments

For a quick look at differential expression levels, the number of significantly up- and down-regulated genes can be summarised in a table.
Significance is defined using an adjusted p-value cutoff that is set at 5% by default.

```{r}
efit_ <- efit[, c(
    "A.endline_vs_B.endline",
    "A.endline_vs_C.endline",
    "B.endline_vs_C.endline")]

dt <- decideTests(efit_)
knitr::kable(
  t(summary(dt)),
  caption = "Number of DEGs per comparison. Each row corresponds to a comparison and each column corresponds to the number of downregulated genes (`Down`), the number of non-differentially expressed genes (`NotSig`), the number of upregulated genes (`Up`)")
```

**TODO** Describe other outputs

```{r, message = FALSE}
# Create outputs
deg_dir <- here("output", "DEGs")
dir.create(deg_dir)

v_ <- v[, v$targets$timepoint == "endline"]
createDEGOutputs(
  outdir = deg_dir,
  efit = efit_,
  v = v_,
  prefix = "endline_treatment_comparisons",
  o = order(
    v_$targets$plate_number,
    v_$targets$treatment,
    v_$targets$timepoint,
    v_$targets$sex,
    v_$targets$id),
  groups = paste0(
    sub("LC47", "", v_$targets$plate_number),
    ".",
    v_$targets$treatment),
  sample.cols = v$targets$treatment_colours)
```

#### Adjusted by sex

**TODO**: Decide on strategy of fixed/random effects to achieve this aim.

```{r}

```

## Figure 2

### Endline vs. baseline

For a quick look at differential expression levels, the number of significantly up- and down-regulated genes can be summarised in a table.
Significance is defined using an adjusted p-value cutoff that is set at 5% by default.

```{r}
efit_ <- efit[, c(
    "A.endline_vs_A.baseline",
    "B.endline_vs_B.baseline",
    "C.endline_vs_C.baseline")]
dt <- decideTests(efit_)
knitr::kable(
  t(summary(dt)),
  caption = "Number of DEGs per comparison. Each row corresponds to a comparison and each column corresponds to the number of downregulated genes (`Down`), the number of non-differentially expressed genes (`NotSig`), the number of upregulated genes (`Up`)")
```

**TODO** Describe other outputs

```{r, message = FALSE}
# Create outputs
deg_dir <- here("output", "DEGs")
dir.create(deg_dir)
createDEGOutputs(
  outdir = deg_dir,
  efit = efit_,
  v = v,
  prefix = "timepoint_comparisons",
  o = order(
    v$targets$plate_number,
    v$targets$treatment,
    v$targets$timepoint,
    v$targets$sex,
    v$targets$id),
  groups = paste0(
    sub("LC47", "", v_$targets$plate_number),
    ".",
    v_$targets$treatment,
    ".",
    ifelse(v$targets$timepoint == "baseline", "b", "e")),
  sample.cols = v$targets$timepoint_colours)
```

#### Check for differences in age at enrolment

```{r}
tmp <- unique(x$samples[, c("id", "treatment", "age0")])
tmp$age0_cut <- cut(tmp$age0, breaks = c(8.5, 10, 11.5))
```

Figure \@ref(fig:age0-density) shows the distribution of age at enrolment (`age0`) by treatment arm.
There is no significant difference in `age0` by treatment arm (ANOVA, $P =$ `r number(summary(aov(tmp$age0 ~ tmp$treatment))[[1]][["Pr(>F)"]][1], accuracy = 0.001)`) nor is there an association when `age0` is binned as $(8.5-10]$ or $(10-11.5]$ ($\chi^2$-test $P =$ `r number(chisq.test(tmp$treatment, tmp$age0_cut)$p.value, accuracy = 0.01)`).

```{r age0-density, fig.cap = "Density estimates of `age0` in each treatment arm. The density curves were scaled such that the area under each curve corresponds to the total number of subjects in each treatment arm."}
ggplot(tmp, aes(x = age0, y = ..count.., fill = treatment)) + 
  geom_density(alpha = 1 / 2) +
  scale_fill_manual(values = treatment_colours) +
  theme_minimal_hgrid() +
  ylab("scaled density") +
  guides(fill = guide_legend(override.aes = list(linetype = 0))) +
  theme(axis.line.x = element_blank()) + 
  geom_vline(xintercept = 10, lty = 2)
```

## Figure 3

### Gene expression by Height for Age at endline

```{r}
tmp <- unique(x$samples[, c("id", "treatment", "HAZ24", "Stunted24")])
```

Figure \@ref(fig:HAZ24-density) shows the distribution of height for age at endline (`HAZ24`) by treatment arm.
There is `r sum(is.na(tmp$HAZ24))` samples without a `HAZ24` measurement (id: `r glue::glue_collapse(tmp$id[is.na(tmp$HAZ24)], sep = ", ", last = ", and ")`).

```{r HAZ24-density, fig.cap = "Density estimates of `HAZ24` in each treatment arm. The density curves were scaled such that the area under each curve corresponds to the total number of subjects in each treatment arm. Samples to the left of the vertical dashed line are `Stunted24`."}
ggplot(tmp, aes(x = HAZ24, y = ..count.., fill = treatment)) + 
  geom_density(alpha = 1 / 2) +
  scale_fill_manual(values = treatment_colours) +
  theme_minimal_hgrid() +
  ylab("scaled density") +
  guides(fill = guide_legend(override.aes = list(linetype = 0))) +
  theme(axis.line.x = element_blank()) + 
  geom_vline(xintercept = -2, lty = 2)
```

Stunting at endline (`Stunted24`) is defined as `HAZ24` $< -2$.
The table below classifies subjects by `treatment` and `Stunted24`.

```{r}
knitr::kable(
  janitor::tabyl(tmp, Stunted24, treatment) %>%
    janitor::adorn_title("combined"))
```

#### `HAZ24`

We subset the data to just the endline `timepoint` and update the design matrix to include `HAZ24` in addition to `treatment`, `id` (i.e. subject ID), and `plate_number` (i.e. batch) information.

Contrasts for association of gene expression with `HAZ24`, accounting for `group`, `id`, and `plate_number`, are set up in `r BiocStyle::Biocpkg("limma")` using the `makeContrasts()` function.

```{r, results = "hide"}
x_ <- x[, !is.na(x$samples$HAZ24) & x$samples$timepoint == "endline"]

design_ <- model.matrix(~0 + treatment + HAZ24 + plate_number, x_$samples)
colnames(design_) <- sub("plate\\_number", "", colnames(design_))

contr.matrix_ <- makeContrasts(HAZ24 = HAZ24, levels = colnames(design_))

par(mfrow = c(1, 2))
v_ <- voom(x_, design_, plot = TRUE)
vfit_ <- lmFit(v_, design_)
vfit_ <- contrasts.fit(vfit_, contrasts = contr.matrix_)
efit_ <- eBayes(vfit_)
plotSA(efit_, main = "Final model: Mean-variance trend")
```

For a quick look at differential expression levels, the number of significantly up- and down-regulated genes can be summarised in a table.
Significance is defined using an adjusted p-value cutoff that is set at 5% by default.

```{r}
dt <- decideTests(efit_)
knitr::kable(
  t(summary(dt)),
  caption = "Number of DEGs per comparison. Each row corresponds to a comparison and each column corresponds to the number of downregulated genes (`Down`), the number of non-differentially expressed genes (`NotSig`), the number of upregulated genes (`Up`)")
```

**TODO** Describe other outputs

```{r, message = FALSE}
# Create outputs
deg_dir <- here("output", "DEGs")
dir.create(deg_dir)
createDEGOutputs(
  outdir = deg_dir,
  efit = efit_,
  v = v_,
  prefix = "HAZ24",
  o = order(
    v_$targets$plate_number,
    v_$targets$treatment,
    v_$targets$timepoint,
    v_$targets$HAZ24,
    v_$targets$Stunted24,
    v_$targets$id),
  groups = paste0(
    sub("LC47", "", v_$targets$plate_number),
    ".",
    v_$targets$treatment,
    ".",
    ifelse(v_$targets$Stunted24, "T", "F")), 
  sample.cols = v_$targets$Stunted24)
```

#### `Stunted24`

We subset the data to just the endline `timepoint` and update the design matrix to include `Stunted24` in addition to `treatment`, `id` (i.e. subject ID), and `plate_number` (i.e. batch) information.

Contrasts for association of gene expression with `Stunted24`, accounting for `group`, `id`, and `plate_number`, are set up in `r BiocStyle::Biocpkg("limma")` using the `makeContrasts()` function.

```{r, results = "hide"}
x_ <- x[, !is.na(x$samples$Stunted24) & x$samples$timepoint == "endline"]

design_ <- model.matrix(~0 + treatment + Stunted24 + plate_number, x_$samples)
colnames(design_) <- sub("plate\\_number", "", colnames(design_))

contr.matrix_ <- makeContrasts(
  Stunted24 = Stunted24TRUE,
  levels = colnames(design_))

par(mfrow = c(1, 2))
v_ <- voom(x_, design_, plot = TRUE)
vfit_ <- lmFit(v_, design_)
vfit_ <- contrasts.fit(vfit_, contrasts = contr.matrix_)
efit_ <- eBayes(vfit_)
plotSA(efit_, main = "Final model: Mean-variance trend")
```

For a quick look at differential expression levels, the number of significantly up- and down-regulated genes can be summarised in a table.
Significance is defined using an adjusted p-value cutoff that is set at 5% by default.

```{r}
dt <- decideTests(efit_)
knitr::kable(
  t(summary(dt)),
  caption = "Number of DEGs per comparison. Each row corresponds to a comparison and each column corresponds to the number of downregulated genes (`Down`), the number of non-differentially expressed genes (`NotSig`), the number of upregulated genes (`Up`)")
```

**TODO** Describe other outputs

```{r, message = FALSE}
# Create outputs
deg_dir <- here("output", "DEGs")
dir.create(deg_dir)
createDEGOutputs(
  outdir = deg_dir,
  efit = efit_,
  v = v_,
  prefix = "Stunted24",
  o = order(
    v_$targets$plate_number,
    v_$targets$treatment,
    v_$targets$timepoint,
    v_$targets$HAZ24,
    v_$targets$Stunted24,
    v_$targets$id),
  groups = paste0(
    sub("LC47", "", v_$targets$plate_number),
    ".",
    v_$targets$treatment,
    ".",
    ifelse(v_$targets$Stunted24, "T", "F")),
  sample.cols = v_$targets$Stunted24)
```

### Gene expression by Weight for Height at endline

```{r}
tmp <- unique(x$samples[, c("id", "treatment", "WHZ24", "Wasted24")])
```

Figure \@ref(fig:WHZ24-density) shows the distribution of weight for height at endline (`WHZ24`) by treatment arm.
There is `r sum(is.na(tmp$WHZ24))` samples without a `WHZ24` measurement (id: `r glue::glue_collapse(tmp$id[is.na(tmp$WHZ24)], sep = ", ", last = ", and ")`).

```{r WHZ24-density, fig.cap = "Density estimates of `WHZ24` in each treatment arm. The density curves were scaled such that the area under each curve corresponds to the total number of subjects in each treatment arm. Samples to the left of the vertical dashed line are `Wasted24`."}
ggplot(tmp, aes(x = WHZ24, y = ..count.., fill = treatment)) + 
  geom_density(alpha = 1 / 2) +
  scale_fill_manual(values = treatment_colours) +
  theme_minimal_hgrid() +
  ylab("scaled density") +
  guides(fill = guide_legend(override.aes = list(linetype = 0))) +
  theme(axis.line.x = element_blank()) +
  geom_vline(xintercept = -2, lty = 2)
```

Wasting at endline (`Wasted24`) is defined as `WHZ24` $< -2$.
The table below classifies subjects by `treatment` and `Wasted24`.

```{r}
knitr::kable(
  janitor::tabyl(tmp, Wasted24, treatment) %>%
    janitor::adorn_title("combined"))
```

#### `WHZ24`

We subset the data to just the endline `timepoint` and update the design matrix to include `WHZ24` in addition to `treatment`, `id` (i.e. subject ID), and `plate_number` (i.e. batch) information.

Contrasts for association of gene expression with `WHZ24`, accounting for `group`, `id`, and `plate_number`, are set up in `r BiocStyle::Biocpkg("limma")` using the `makeContrasts()` function.

```{r, results = "hide"}
x_ <- x[, !is.na(x$samples$WHZ24) & x$samples$timepoint == "endline"]

design_ <- model.matrix(~0 + treatment + WHZ24 + plate_number, x_$samples)
colnames(design_) <- sub("plate\\_number", "", colnames(design_))

contr.matrix_ <- makeContrasts(WHZ24 = WHZ24, levels = colnames(design_))

par(mfrow = c(1, 2))
v_ <- voom(x_, design_, plot = TRUE)
vfit_ <- lmFit(v_, design_)
vfit_ <- contrasts.fit(vfit_, contrasts = contr.matrix_)
efit_ <- eBayes(vfit_)
plotSA(efit_, main = "Final model: Mean-variance trend")
```

For a quick look at differential expression levels, the number of significantly up- and down-regulated genes can be summarised in a table.
Significance is defined using an adjusted p-value cutoff that is set at 5% by default.

```{r}
dt <- decideTests(efit_)
knitr::kable(
  t(summary(dt)),
  caption = "Number of DEGs per comparison. Each row corresponds to a comparison and each column corresponds to the number of downregulated genes (`Down`), the number of non-differentially expressed genes (`NotSig`), the number of upregulated genes (`Up`)")
```

**TODO** Describe other outputs

```{r, message = FALSE}
# Create outputs
deg_dir <- here("output", "DEGs")
dir.create(deg_dir)
createDEGOutputs(
  outdir = deg_dir,
  efit = efit_,
  v = v_,
  prefix = "WHZ24",
  o = order(
    v_$targets$plate_number,
    v_$targets$treatment,
    v_$targets$timepoint,
    v_$targets$WHZ24,
    v_$targets$Wasted24,
    v_$targets$id),
  groups = paste0(
    sub("LC47", "", v_$targets$plate_number),
    ".",
    v_$targets$treatment,
    ".",
    ifelse(v_$targets$Wasted24, "T", "F")), 
  sample.cols = v_$targets$Wasted24)
```

#### `Wasted24`

We subset the data to just the endline `timepoint` and update the design matrix to include `Wasted24` in addition to `treatment`, `id` (i.e. subject ID), and `plate_number` (i.e. batch) information.

Contrasts for association of gene expression with `Wasted24`, accounting for `group`, `id`, and `plate_number`, are set up in `r BiocStyle::Biocpkg("limma")` using the `makeContrasts()` function.

```{r, results = "hide"}
x_ <- x[, !is.na(x$samples$Wasted24) & x$samples$timepoint == "endline"]

design_ <- model.matrix(~0 + treatment + Wasted24 + plate_number, x_$samples)
colnames(design_) <- sub("plate\\_number", "", colnames(design_))

contr.matrix_ <- makeContrasts(
  Wasted24 = Wasted24TRUE,
  levels = colnames(design_))

par(mfrow = c(1, 2))
v_ <- voom(x_, design_, plot = TRUE)
vfit_ <- lmFit(v_, design_)
vfit_ <- contrasts.fit(vfit_, contrasts = contr.matrix_)
efit_ <- eBayes(vfit_)
plotSA(efit_, main = "Final model: Mean-variance trend")
```

For a quick look at differential expression levels, the number of significantly up- and down-regulated genes can be summarised in a table.
Significance is defined using an adjusted p-value cutoff that is set at 5% by default.

```{r}
dt <- decideTests(efit_)
knitr::kable(
  t(summary(dt)),
  caption = "Number of DEGs per comparison. Each row corresponds to a comparison and each column corresponds to the number of downregulated genes (`Down`), the number of non-differentially expressed genes (`NotSig`), the number of upregulated genes (`Up`)")
```

**TODO** Describe other outputs

```{r, message = FALSE}
# Create outputs
deg_dir <- here("output", "DEGs")
dir.create(deg_dir)
createDEGOutputs(
  outdir = deg_dir,
  efit = efit_,
  v = v_,
  prefix = "Wasted24",
  o = order(
    v_$targets$plate_number,
    v_$targets$treatment,
    v_$targets$timepoint,
    v_$targets$WHZ24,
    v_$targets$Wasted24,
    v_$targets$id),
  groups = paste0(
    sub("LC47", "", v_$targets$plate_number),
    ".",
    v_$targets$treatment,
    ".",
    ifelse(v_$targets$Wasted24, "T", "F")),
  sample.cols = v_$targets$Wasted24)
```

### Gene expression by Weight for Age at endline

```{r}
tmp <- unique(x$samples[, c("id", "treatment", "WAZ24")])
# Underweight is defined as WAZ < -2
# Overweight is defined as WAZ > 2
tmp$weightstatus <- dplyr::case_when(
  is.na(tmp$WAZ24) ~ NA_character_,
  tmp$WAZ24 < -2 ~ "underweight",
  tmp$WAZ24 > -2 & tmp$WAZ24 < 2 ~ "normalrange",
  tmp$WAZ24 > 2 ~ "overweight")
tmp$Underweight24 <- dplyr::case_when(
  is.na(tmp$WAZ24) ~ NA,
  tmp$WAZ24 < -2 ~ TRUE,
  tmp$WAZ24 > -2 & tmp$WAZ24 < 2 ~ FALSE,
  tmp$WAZ24 > 2 ~ FALSE)
```

Figure \@ref(fig:WAZ24-density) shows the distribution of weight for age at endline (`WAZ24`) by treatment arm.
There is `r sum(is.na(tmp$WAZ24))` samples without a `WAZ24` measurement (id: `r glue::glue_collapse(tmp$id[is.na(tmp$WAZ24)], sep = ", ", last = ", and ")`).

```{r WAZ24-density, fig.cap = "Density estimates of `WAZ24` in each treatment arm. The density curves were scaled such that the area under each curve corresponds to the total number of subjects in each treatment arm. Samples to the left of the vertical dashed line are `Underweight24`."}
ggplot(tmp, aes(x = WAZ24, y = ..count.., fill = treatment)) + 
  geom_density(alpha = 1 / 2) +
  scale_fill_manual(values = treatment_colours) +
  theme_minimal_hgrid() +
  ylab("scaled density") +
  guides(fill = guide_legend(override.aes = list(linetype = 0))) +
  theme(axis.line.x = element_blank()) +
  geom_vline(xintercept = -2, lty = 2)
```

Underweight at endline (`Underweight24`) is defined as `WAZ24` $< -2$.
The table below classifies subjects by `treatment` and `Underweight24`.

```{r}
knitr::kable(
  janitor::tabyl(tmp, Underweight24, treatment) %>%
    janitor::adorn_title("combined"))
```

#### `WAZ24`

We subset the data to just the endline `timepoint` and update the design matrix to include `WAZ24` in addition to `treatment`, `id` (i.e. subject ID), and `plate_number` (i.e. batch) information.

Contrasts for association of gene expression with `WAZ24`, accounting for `group`, `id`, and `plate_number`, are set up in `r BiocStyle::Biocpkg("limma")` using the `makeContrasts()` function.

```{r, results = "hide"}
x_ <- x[, !is.na(x$samples$WAZ24) & x$samples$timepoint == "endline"]

design_ <- model.matrix(~0 + treatment + WAZ24 + plate_number, x_$samples)
colnames(design_) <- sub("plate\\_number", "", colnames(design_))

contr.matrix_ <- makeContrasts(WAZ24 = WAZ24, levels = colnames(design_))

par(mfrow = c(1, 2))
v_ <- voom(x_, design_, plot = TRUE)
vfit_ <- lmFit(v_, design_)
vfit_ <- contrasts.fit(vfit_, contrasts = contr.matrix_)
efit_ <- eBayes(vfit_)
plotSA(efit_, main = "Final model: Mean-variance trend")
```

For a quick look at differential expression levels, the number of significantly up- and down-regulated genes can be summarised in a table.
Significance is defined using an adjusted p-value cutoff that is set at 5% by default.

```{r}
dt <- decideTests(efit_)
knitr::kable(
  t(summary(dt)),
  caption = "Number of DEGs per comparison. Each row corresponds to a comparison and each column corresponds to the number of downregulated genes (`Down`), the number of non-differentially expressed genes (`NotSig`), the number of upregulated genes (`Up`)")
```

**TODO** Describe other outputs

```{r, message = FALSE}
# Create outputs
deg_dir <- here("output", "DEGs")
dir.create(deg_dir)
createDEGOutputs(
  outdir = deg_dir,
  efit = efit_,
  v = v_,
  prefix = "WAZ24",
  o = order(
    v_$targets$plate_number,
    v_$targets$treatment,
    v_$targets$timepoint,
    v_$targets$WAZ24,
    v_$targets$Underweight24,
    v_$targets$id),
  groups = paste0(
    sub("LC47", "", v_$targets$plate_number),
    ".",
    v_$targets$treatment,
    ".",
    ifelse(v_$targets$Underweight24, "T", "F")),
  sample.cols = v_$targets$Underweight24)
```

#### `Underweight24`

We subset the data to just the endline `timepoint` and update the design matrix to include `Underweight24` in addition to `treatment`, `id` (i.e. subject ID), and `plate_number` (i.e. batch) information.

Contrasts for association of gene expression with `Underweight24`, accounting for `group`, `id`, and `plate_number`, are set up in `r BiocStyle::Biocpkg("limma")` using the `makeContrasts()` function.

```{r, results = "hide"}
x_ <- x[, !is.na(x$samples$Underweight24) & x$samples$timepoint == "endline"]

design_ <- model.matrix(
  ~0 + treatment + Underweight24 + plate_number,
  x_$samples)
colnames(design_) <- sub("plate\\_number", "", colnames(design_))

contr.matrix_ <- makeContrasts(
  Underweight24 = Underweight24TRUE,
  levels = colnames(design_))

par(mfrow = c(1, 2))
v_ <- voom(x_, design_, plot = TRUE)
vfit_ <- lmFit(v_, design_)
vfit_ <- contrasts.fit(vfit_, contrasts = contr.matrix_)
efit_ <- eBayes(vfit_)
plotSA(efit_, main = "Final model: Mean-variance trend")
```

For a quick look at differential expression levels, the number of significantly up- and down-regulated genes can be summarised in a table.
Significance is defined using an adjusted p-value cutoff that is set at 5% by default.

```{r}
dt <- decideTests(efit_)
knitr::kable(
  t(summary(dt)),
  caption = "Number of DEGs per comparison. Each row corresponds to a comparison and each column corresponds to the number of downregulated genes (`Down`), the number of non-differentially expressed genes (`NotSig`), the number of upregulated genes (`Up`)")
```

**TODO** Describe other outputs

```{r, message = FALSE}
# Create outputs
deg_dir <- here("output", "DEGs")
dir.create(deg_dir)
createDEGOutputs(
  outdir = deg_dir,
  efit = efit_,
  v = v_,
  prefix = "Underweight24",
  o = order(
    v_$targets$plate_number,
    v_$targets$treatment,
    v_$targets$timepoint,
    v_$targets$WAZ24,
    v_$targets$Underweight24,
    v_$targets$id),
  groups = paste0(
    sub("LC47", "", v_$targets$plate_number),
    ".",
    v_$targets$treatment,
    ".",
    ifelse(v_$targets$Underweight24, "T", "F")),
  sample.cols = v_$targets$Underweight24)
```

### Gene expression by age

**TODO**: It's not clear what comparison is being requested from Tom's email.

**TODO**: It's not clear which age column to use (`r glue::glue_collapse(grep("age", colnames(colData(sce)), value = TRUE, ignore.case = TRUE), ", ", last = ", or ")`).

```{r}
tmp <- unique(
  x$samples[, c(
    "id", 
    "treatment",
    grep("age", colnames(colData(sce)), value = TRUE, ignore.case = TRUE))])

# Someone with missing age variables
tmp$id[is.na(tmp$age0)]
tmp$id[is.na(tmp$age12)]
tmp$id[is.na(tmp$age24)]
tmp$id[is.na(tmp$EnrAgeMo)]

# EnrAgeMo and age0 are the same variable
stopifnot(all.equal(sce$EnrAgeMo, sce$age0))

# age0 density
p1 <- ggplot(tmp, aes(x = age0, y = ..count.., fill = treatment)) + 
  geom_density(alpha = 1 / 2) +
  scale_fill_manual(values = treatment_colours) +
  theme_minimal_hgrid() +
  ylab("scaled density") +
  guides(fill = guide_legend(override.aes = list(linetype = 0))) +
  theme(axis.line.x = element_blank())

# age12 density
p2 <- ggplot(tmp, aes(x = age12, y = ..count.., fill = treatment)) + 
  geom_density(alpha = 1 / 2) +
  scale_fill_manual(values = treatment_colours) +
  theme_minimal_hgrid() +
  ylab("scaled density") +
  guides(fill = guide_legend(override.aes = list(linetype = 0))) +
  theme(axis.line.x = element_blank())

# age24 density
p3 <- ggplot(tmp, aes(x = age24, y = ..count.., fill = treatment)) + 
  geom_density(alpha = 1 / 2) +
  scale_fill_manual(values = treatment_colours) +
  theme_minimal_hgrid() +
  ylab("scaled density") +
  guides(fill = guide_legend(override.aes = list(linetype = 0))) +
  theme(axis.line.x = element_blank())

# EnrAgeMo density
p4 <- ggplot(tmp, aes(x = EnrAgeMo, y = ..count.., fill = treatment)) + 
  geom_density(alpha = 1 / 2) +
  scale_fill_manual(values = treatment_colours) +
  theme_minimal_hgrid() +
  ylab("scaled density") +
  guides(fill = guide_legend(override.aes = list(linetype = 0))) +
  theme(axis.line.x = element_blank())

((p4| p1) / (p2 | p3)) + plot_layout(guides = "collect")
```

### Gene expression by sex

**TODO**: It's not clear what comparison is being requested from Tom's email.

## Figure 4

### Gene expression by Haemoglobin at endline

```{r}
tmp <- unique(x$samples[, c("id", "treatment", "hb24")])
# anaemic is defined as hb < 11
tmp$Anaemic24 <- dplyr::case_when(
  is.na(tmp$hb24) ~ NA,
  tmp$hb24 < 11 ~ TRUE,
  tmp$hb24 >= 11 ~ FALSE)
```

Figure \@ref(fig:hb24-density) shows the distribution of weight for age at endline (`hb24`) by treatment arm.
There is `r sum(is.na(tmp$hb24))` samples without a `hb24` measurement (id: `r glue::glue_collapse(tmp$id[is.na(tmp$hb24)], sep = ", ", last = " and ")`).

```{r hb24-density, fig.cap = "Density estimates of `hb24` in each treatment arm. The density curves were scaled such that the area under each curve corresponds to the total number of subjects in each treatment arm. Samples to the left of the vertical dashed line are `Anaemic24`."}
ggplot(tmp, aes(x = hb24, y = ..count.., fill = treatment)) + 
  geom_density(alpha = 1 / 2) +
  scale_fill_manual(values = treatment_colours) +
  theme_minimal_hgrid() +
  ylab("scaled density") +
  guides(fill = guide_legend(override.aes = list(linetype = 0))) +
  theme(axis.line.x = element_blank()) + 
  geom_vline(xintercept = 11, lty = 2)
```

Anaemic at endline (`Anaemic24`) is defined as `hb24` $< -2$.
The table below classifies subjects by `treatment` and `Anaemic24`.

```{r}
knitr::kable(
  janitor::tabyl(tmp, Anaemic24, treatment) %>%
    janitor::adorn_title("combined"))
```

#### `hb24`

We subset the data to just the endline `timepoint` and update the design matrix to include `hb24` in addition to `treatment`, `id` (i.e. subject ID), and `plate_number` (i.e. batch) information.

Contrasts for association of gene expression with `hb24`, accounting for `group`, `id`, and `plate_number`, are set up in `r BiocStyle::Biocpkg("limma")` using the `makeContrasts()` function.

```{r, results = "hide"}
x_ <- x[, !is.na(x$samples$hb24) & x$samples$timepoint == "endline"]

design_ <- model.matrix(~0 + treatment + hb24 + plate_number, x_$samples)
colnames(design_) <- sub("plate\\_number", "", colnames(design_))

contr.matrix_ <- makeContrasts(hb24 = hb24, levels = colnames(design_))

par(mfrow = c(1, 2))
v_ <- voom(x_, design_, plot = TRUE)
vfit_ <- lmFit(v_, design_)
vfit_ <- contrasts.fit(vfit_, contrasts = contr.matrix_)
efit_ <- eBayes(vfit_)
plotSA(efit_, main = "Final model: Mean-variance trend")
```

For a quick look at differential expression levels, the number of significantly up- and down-regulated genes can be summarised in a table.
Significance is defined using an adjusted p-value cutoff that is set at 5% by default.

```{r}
dt <- decideTests(efit_)
knitr::kable(
  t(summary(dt)),
  caption = "Number of DEGs per comparison. Each row corresponds to a comparison and each column corresponds to the number of downregulated genes (`Down`), the number of non-differentially expressed genes (`NotSig`), the number of upregulated genes (`Up`)")
```

**TODO** Describe other outputs

```{r, message = FALSE}
# Create outputs
deg_dir <- here("output", "DEGs")
dir.create(deg_dir)
createDEGOutputs(
  outdir = deg_dir,
  efit = efit_,
  v = v_,
  prefix = "hb24",
  o = order(
    v_$targets$plate_number,
    v_$targets$treatment,
    v_$targets$timepoint,
    v_$targets$hb24,
    v_$targets$Anaemic24,
    v_$targets$id),
  groups = paste0(
    sub("LC47", "", v_$targets$plate_number),
    ".",
    v_$targets$treatment,
    ".",
    ifelse(v_$targets$Anaemic24, "T", "F")),
  sample.cols = v_$targets$Anaemic24)
```

#### `Anaemic24`

We subset the data to just the endline `timepoint` and update the design matrix to include `Anaemic24` in addition to `treatment`, `id` (i.e. subject ID), and `plate_number` (i.e. batch) information.

Contrasts for association of gene expression with `Anaemic24`, accounting for `group`, `id`, and `plate_number`, are set up in `r BiocStyle::Biocpkg("limma")` using the `makeContrasts()` function.

```{r, results = "hide"}
x_ <- x[, !is.na(x$samples$Anaemic24) & x$samples$timepoint == "endline"]

design_ <- model.matrix(~0 + treatment + Anaemic24 + plate_number, x_$samples)
colnames(design_) <- sub("plate\\_number", "", colnames(design_))

contr.matrix_ <- makeContrasts(
  Anaemic24 = Anaemic24TRUE,
  levels = colnames(design_))

par(mfrow = c(1, 2))
v_ <- voom(x_, design_, plot = TRUE)
vfit_ <- lmFit(v_, design_)
vfit_ <- contrasts.fit(vfit_, contrasts = contr.matrix_)
efit_ <- eBayes(vfit_)
plotSA(efit_, main = "Final model: Mean-variance trend")
```

For a quick look at differential expression levels, the number of significantly up- and down-regulated genes can be summarised in a table.
Significance is defined using an adjusted p-value cutoff that is set at 5% by default.

```{r}
dt <- decideTests(efit_)
knitr::kable(
  t(summary(dt)),
  caption = "Number of DEGs per comparison. Each row corresponds to a comparison and each column corresponds to the number of downregulated genes (`Down`), the number of non-differentially expressed genes (`NotSig`), the number of upregulated genes (`Up`)")
```

**TODO** Describe other outputs

```{r, message = FALSE}
# Create outputs
deg_dir <- here("output", "DEGs")
dir.create(deg_dir)
createDEGOutputs(
  outdir = deg_dir,
  efit = efit_,
  v = v_,
  prefix = "Anaemic24",
  o = order(
    v_$targets$plate_number,
    v_$targets$treatment,
    v_$targets$timepoint,
    v_$targets$hb24,
    v_$targets$Anaemic24,
    v_$targets$id),
  groups = paste0(
    sub("LC47", "", v_$targets$plate_number),
    ".",
    v_$targets$treatment,
    ".",
    ifelse(v_$targets$Anaemic24, "T", "F")),
  sample.cols = v_$targets$Anaemic24)
```

### Correlation between recovering from anaemia and being in a treatment group

**TODO**: It's not clear what comparison is being requested from Tom's email.

```{r}
tmp <- unique(
  x$samples[, c("id", "treatment", "hb0", "hb24", "Anaemic0", "Anaemic24")])

library(gghighlight)
p <- ggplot(
  tidyr::pivot_longer(tmp, c(hb0, hb24), names_to = "timepoint"),
  aes(x = timepoint, y = value, colour = treatment)) +
  geom_point() +
  geom_line(aes(group = id), alpha = 1 / 3, size = 1) + 
  scale_colour_manual(values = treatment_colours) +
  theme_minimal_hgrid() +
  facet_grid(~treatment) + 
  guides(colour = FALSE)+
  ggtitle("Change in haemoglobin from baseline to endline")
p
```

```{r}
p1 <- p +
  geom_line(aes(group = id), alpha = 1, size = 2) + 
  gghighlight(
    !Anaemic0 & !Anaemic24,
    unhighlighted_params = list(size = 1 / 2, alpha = 1 / 3),
    use_direct_label = FALSE) +
  ggtitle("Children without anaemia (no change)")

p2 <- p +
  geom_line(aes(group = id), alpha = 1, size = 2) + 
  gghighlight(
    Anaemic0 & !Anaemic24,
    unhighlighted_params = list(size = 1 / 2, alpha = 1 / 3),
    use_direct_label = FALSE) +
    ggtitle("Children that recovered from anaemia")

p3 <- p +
  geom_line(aes(group = id), alpha = 1, size = 2) + 
  gghighlight(
    !Anaemic0 & Anaemic24,
    unhighlighted_params = list(size = 1 / 2, alpha = 1 / 3),
    use_direct_label = FALSE) +
    ggtitle("Children that developed anaemia")

p4 <- p +
  geom_line(aes(group = id), alpha = 1, size = 2) + 
  gghighlight(
    Anaemic0 & Anaemic24,
    unhighlighted_params = list(size = 1 / 2, alpha = 1 / 3),
    use_direct_label = FALSE) +
    ggtitle("Children with anaemia (no change)")
    
(p1 / p2) | (p3 / p4)
```

```{r}
tmp2 <- tmp[!is.na(tmp$Anaemic0) & !is.na(tmp$Anaemic24), ]
```

Anaemia rates between treatments at baseline.

```{r}
table(tmp2$treatment, tmp2$Anaemic0)
chisq.test(table(tmp2$treatment, tmp2$Anaemic0))
```

Anaemic rates between treatments at endline

```{r}
table(tmp2$treatment, tmp2$Anaemic24)
chisq.test(table(tmp2$treatment, tmp2$Anaemic24))
```

Anaemic rates between treatments at baseline and endline

```{r}
dplyr::count(tmp2, treatment, Anaemic0, Anaemic24)
z <- table(tmp2$Anaemic0, tmp2$Anaemic24, tmp2$treatment)
```

Treatment A

```{r}
z[, , "A"]
fisher.test(z[, , "A"])
```

Treatment B

```{r}
z[, , "B"]
fisher.test(z[, , "B"])
```

Treatment C

```{r}
z[, , "C"]
fisher.test(z[, , "C"])
```

Transitions between treatments

```{r}
zz <- table(tmp2$treatment, paste0(tmp2$Anaemic0, "->", tmp2$Anaemic24))
zz
chisq.test(zz)
```

Children that recovered from anaemia

```{r}
chisq.test(zz[, c("TRUE->FALSE", "TRUE->TRUE")])
```

Children that developed anaemia

```{r}
chisq.test(zz[, c("FALSE->FALSE", "FALSE->TRUE")])
```

Classifying as 'recovered' or 'not recovered' (including those that never go anaemia).

```{r}
zzz <- table(
  tmp2$treatment,
  dplyr::case_when(
    tmp2$Anaemic0 & !tmp2$Anaemic24 ~ "recovered",
    TRUE ~ "not recovered"))
zzz
chisq.test(zzz)
```

# Additional information {.appendix}

The following are available on request:

- Full CSV tables of any data presented.
- PDF/PNG files of any static plots.

# Session info {.appendix}

<summary>The analysis and this document were prepared using the following software (click triangle to expand)</summary>
<details>

```{r}
sessioninfo::session_info()
```

</details>

```{r}
knitr::knit_exit()
```

# OLD CODE

## Example results

**TODO** Combine the next subsubsections into 1 subsection (or just remove entirely and move into the results?)

### Examining the number of DE genes

For a quick look at differential expression levels, the number of significantly up- and down-regulated genes can be summarised in a table.
Significance is defined using an adjusted p-value cutoff that is set at 5% by default.
There are a number of DEGs between timepoints within each treatment but fewer DEGs between treatments at endline.

```{r}
dt <- decideTests(efit)
t(summary(dt))
```

```{r venn, fig.cap = "Venn diagrams showing the number of genes DE in the pairwise comparisons. The number of genes that are not DE in any comparison are marked in the bottom-right.", fig.asp = 1 / 2}
par(mfrow = c(1, 2))
vennDiagram(dt[, 1:3], main = "DE between timepoints within treatment", cex = 0.7)
vennDiagram(dt[, 4:6], main = "DE between treatments at endline", cex = 0.7)
```

```{r}
# TODO: Create my own output function.
deg_dir <- here("output", "DEGs")
dir.create(deg_dir)
write.fit(efit, dt, file = file.path(deg_dir, "results.txt"))
```

### Examining individual DE genes from top to bottom

```{r}
topTable(efit, coef = "A.endline_vs_A.baseline")
topTable(efit, coef = "B.endline_vs_B.baseline")
topTable(efit, coef = "C.endline_vs_C.baseline")

topTable(efit, coef = "A.endline_vs_B.endline")
topTable(efit, coef = "A.endline_vs_C.endline")
topTable(efit, coef = "B.endline_vs_C.endline")
```

### Useful graphical representations of differential expression results

To summarise results for all genes visually, mean-difference plots, which display log-FCs from the linear model fit against the average log-CPM values can be generated using the `plotMD()` function, with the differentially expressed genes highlighted.

```{r, fig.asp = 2 / 3}
par(mfrow = c(2, 3))
plotMD(
  efit,
  coef = "A.endline_vs_A.baseline",
  status = dt[, "A.endline_vs_A.baseline"],
  main = "A.endline_vs_A.baseline")

plotMD(
  efit,
  coef = "B.endline_vs_B.baseline",
  status = dt[, "B.endline_vs_B.baseline"],
  main = "B.endline_vs_B.baseline")

plotMD(
  efit,
  coef = "C.endline_vs_C.baseline",
  status = dt[, "C.endline_vs_C.baseline"],
  main = "C.endline_vs_C.baseline")

plotMD(
  efit,
  coef = "A.endline_vs_B.endline",
  status = dt[, "A.endline_vs_B.endline"],
  main = "A.endline_vs_B.endline")

plotMD(
  efit,
  coef = "A.endline_vs_C.endline",
  status = dt[, "A.endline_vs_C.endline"],
  main = "A.endline_vs_C.endline")

plotMD(
  efit,
  coef = "B.endline_vs_C.endline",
  status = dt[, "B.endline_vs_C.endline"],
  main = "B.endline_vs_C.endline")
```

`r BiocStyle::Biocpkg("Glimma")` extends this functionality by providing an interactive mean-difference plot.
The output of this function is an html page, with summarised results in the left panel (similar to what is output by `plotMD()`), and the log-CPM values from individual samples for a selected gene in the right panel, with a table of results below the plots.
This interactive display allows the user to search for particular genes based on the annotation provided (e.g. Gene symbol identifier), which is not possible in a static R plot.

# Results

## Figure 1

### Endline pairwise analyses between treatments

```{r}
design <- model.matrix(~0 + group + id + plate_number, x$samples)
colnames(design) <- sub("plate\\_number|group", "", colnames(design))

contr.matrix <- makeContrasts(
  A.endline_vs_B.endline = A.endline - B.endline,
  A.endline_vs_C.endline = A.endline - C.endline,
  B.endline_vs_C.endline = B.endline - C.endline,
  levels = colnames(design))

par(mfrow = c(1, 2))
v <- voom(x, design, plot = TRUE)
vfit <- lmFit(v, design)
vfit <- contrasts.fit(vfit, contrasts = contr.matrix)
efit <- eBayes(vfit)
plotSA(efit, main = "Final model: Mean-variance trend")

dt <- decideTests(efit)
summary(dt)
```

#### Adjusted by sex

**TODO**: Decide on strategy of fixed/random effects to achieve this aim.

```{r}

```

## Figure 2

### Endline vs. baseline

```{r}
design <- model.matrix(~0 + group + id + plate_number, x$samples)
colnames(design) <- sub("plate\\_number|group", "", colnames(design))

contr.matrix <- makeContrasts(
  A.endline_vs_A.baseline = A.endline - A.baseline,
  B.endline_vs_B.baseline = B.endline - B.baseline,
  C.endline_vs_C.baseline = C.endline - C.baseline,
  levels = colnames(design))

par(mfrow = c(1, 2))
v <- voom(x, design, plot = TRUE)
vfit <- lmFit(v, design)
vfit <- contrasts.fit(vfit, contrasts = contr.matrix)
efit <- eBayes(vfit)
plotSA(efit, main = "Final model: Mean-variance trend")

dt <- decideTests(efit)
summary(dt)
```

#### Check for differences in age at enrolment

```{r, results = "show"}
tmp <- unique(x$samples[, c("id", "treatment", "age0")])

boxplot(tmp$age0 ~ tmp$treatment)
summary(aov(tmp$age0 ~ tmp$treatment))
anova(lm(tmp$age0 ~ tmp$treatment))

plot(density(tmp$age0))
tmp$age0_cut <- cut(tmp$age0, breaks = c(8.5, 10, 11.5))
boxplot(tmp$age0 ~ tmp$treatment + tmp$age0_cut)
chisq.test(tmp$treatment, tmp$age0_cut)
```

## Figure 3

### Gene expression by Height for Age at endline

```{r}
tmp <- unique(x$samples[, c("id", "treatment", "HAZ24", "Stunted24")])

# Someone with missing HAZ24
tmp$id[is.na(tmp$HAZ24)]

# Stunting is defined as HAZ < -2
table(tmp$HAZ24 < -2, tmp$Stunted24)
tapply(tmp$HAZ24, tmp$Stunted24, summary)

ggplot(tmp, aes(x = HAZ24, y = ..count.., fill = treatment)) + 
  geom_density(alpha = 1 / 2) +
  scale_fill_manual(values = treatment_colours) +
  theme_minimal_hgrid() +
  ylab("scaled density") +
  guides(fill = guide_legend(override.aes = list(linetype = 0))) +
  theme(axis.line.x = element_blank())
```

#### HAZ24

```{r}
x_ <- x[, !is.na(x$samples$HAZ24) & x$samples$timepoint == "endline"]

design <- model.matrix(~0 + treatment + HAZ24 + plate_number, x_$samples)
colnames(design) <- sub("plate\\_number", "", colnames(design))

contr.matrix <- makeContrasts(HAZ24 = HAZ24, levels = colnames(design))

par(mfrow = c(1, 2))
v <- voom(x_, design, plot = TRUE)
vfit <- lmFit(v, design)
vfit <- contrasts.fit(vfit, contrasts = contr.matrix)
efit <- eBayes(vfit)
plotSA(efit, main = "Final model: Mean-variance trend")

dt <- decideTests(efit)
summary(dt)
```

#### Stunted24

```{r}
x_ <- x[, !is.na(x$samples$Stunted24) & x$samples$timepoint == "endline"]

design <- model.matrix(~0 + treatment + Stunted24 + plate_number, x_$samples)
colnames(design) <- sub("plate\\_number|treatment", "", colnames(design))

contr.matrix <- makeContrasts(
  Stunted24 = Stunted24TRUE,
  levels = colnames(design))

par(mfrow = c(1, 2))
v <- voom(x_, design, plot = TRUE)
vfit <- lmFit(v, design)
vfit <- contrasts.fit(vfit, contrasts = contr.matrix)
efit <- eBayes(vfit)
plotSA(efit, main = "Final model: Mean-variance trend")

dt <- decideTests(efit)
summary(dt)
```

### Gene expression by Weight for Height at endline

```{r}
tmp <- unique(x$samples[, c("id", "treatment", "WHZ24", "Wasted24")])

# Someone with missing WHZ24
tmp$id[is.na(tmp$WHZ24)]

# Wasting is defined as WHZ < -2
table(tmp$WHZ24 < -2, tmp$Wasted24)
tapply(tmp$WHZ24, tmp$Wasted24, summary)

# WHZ24 density
ggplot(tmp, aes(x = WHZ24, y = ..count.., fill = treatment)) + 
  geom_density(alpha = 1 / 2) +
  scale_fill_manual(values = treatment_colours) +
  theme_minimal_hgrid() +
  ylab("scaled density") +
  guides(fill = guide_legend(override.aes = list(linetype = 0))) +
  theme(axis.line.x = element_blank())
```

#### WHZ24

```{r}
x_ <- x[, !is.na(x$samples$WHZ24) & x$samples$timepoint == "endline"]

design <- model.matrix(~0 + treatment + WHZ24 + plate_number, x_$samples)
colnames(design) <- sub("plate\\_number|treatment", "", colnames(design))

contr.matrix <- makeContrasts(WHZ24 = WHZ24, levels = colnames(design))

par(mfrow = c(1, 2))
v <- voom(x_, design, plot = TRUE)
vfit <- lmFit(v, design)
vfit <- contrasts.fit(vfit, contrasts = contr.matrix)
efit <- eBayes(vfit)
plotSA(efit, main = "Final model: Mean-variance trend")

dt <- decideTests(efit)
summary(dt)
```

#### Wasted24

```{r}
x_ <- x[, !is.na(x$samples$Wasted24) & x$samples$timepoint == "endline"]

design <- model.matrix(~0 + treatment + Wasted24 + plate_number, x_$samples)
colnames(design) <- sub("plate\\_number|treatment", "", colnames(design))

contr.matrix <- makeContrasts(
  Wasted24 = Wasted24TRUE,
  levels = colnames(design))

par(mfrow = c(1, 2))
v <- voom(x_, design, plot = TRUE)
vfit <- lmFit(v, design)
vfit <- contrasts.fit(vfit, contrasts = contr.matrix)
efit <- eBayes(vfit)
plotSA(efit, main = "Final model: Mean-variance trend")

dt <- decideTests(efit)
summary(dt)
```

### Gene expression by Weight for Age at endline

```{r}
tmp <- unique(x$samples[, c("id", "treatment", "WAZ24")])

# Someone with missing WAZ24
tmp$id[is.na(tmp$WAZ24)]

# Underweight is defined as WAZ < -2
# Overweight is defined as WAZ > 2
tmp$weightstatus <- dplyr::case_when(
  is.na(tmp$WAZ24) ~ NA_character_,
  tmp$WAZ24 < -2 ~ "underweight",
  tmp$WAZ24 > -2 & tmp$WAZ24 < 2 ~ "normalrange",
  tmp$WAZ24 > 2 ~ "overweight")
table(tmp$weightstatus)

tapply(tmp$WAZ24, tmp$weightstatus, summary)

# WAZ24 density
ggplot(tmp, aes(x = WAZ24, y = ..count.., fill = treatment)) + 
  geom_density(alpha = 1 / 2) +
  scale_fill_manual(values = treatment_colours) +
  theme_minimal_hgrid() +
  ylab("scaled density") +
  guides(fill = guide_legend(override.aes = list(linetype = 0))) +
  theme(axis.line.x = element_blank())
```

#### WAZ24

```{r}
x_ <- x[, !is.na(x$samples$WAZ24) & x$samples$timepoint == "endline"]

design <- model.matrix(~0 + treatment + WAZ24 + plate_number, x_$samples)
colnames(design) <- sub("plate\\_number|treatment", "", colnames(design))

contr.matrix <- makeContrasts(WAZ24 = WAZ24, levels = colnames(design))

par(mfrow = c(1, 2))
v <- voom(x_, design, plot = TRUE)
vfit <- lmFit(v, design)
vfit <- contrasts.fit(vfit, contrasts = contr.matrix)
efit <- eBayes(vfit)
plotSA(efit, main = "Final model: Mean-variance trend")

dt <- decideTests(efit)
summary(dt)
```

#### Underweight24

```{r}
x_ <- x[, x$samples$timepoint == "endline"]
# NOTE: There are no overweight subjects, so can ignore that category
x_$samples$Underweight24 <- dplyr::case_when(
  is.na(x_$samples$WAZ24) ~ NA,
  x_$samples$WAZ24 < -2 ~ TRUE,
  x_$samples$WAZ24 > -2 & x_$samples$WAZ24 < 2 ~ FALSE,
  x_$samples$WAZ24 > 2 ~ FALSE)
x_ <- x_[, !is.na(x_$samples$Underweight24)]

design <- model.matrix(~0 + treatment + Underweight24 + plate_number, x_$samples)
colnames(design) <- sub("plate\\_number|treatment", "", colnames(design))

contr.matrix <- makeContrasts(
  Underweight24 = Underweight24TRUE,
  levels = colnames(design))

par(mfrow = c(1, 2))
v <- voom(x_, design, plot = TRUE)
vfit <- lmFit(v, design)
vfit <- contrasts.fit(vfit, contrasts = contr.matrix)
efit <- eBayes(vfit)
plotSA(efit, main = "Final model: Mean-variance trend")

dt <- decideTests(efit)
summary(dt)
```

### Gene expression by age

**TODO**: It's not clear what comparison is being requested from Tom's email.

**TODO**: It's not clear which age column to use (`r glue::glue_collapse(grep("age", colnames(colData(sce)), value = TRUE, ignore.case = TRUE), ", ", last = ", or ")`).

```{r}
tmp <- unique(
  x$samples[, c(
    "id", 
    "treatment",
    grep("age", colnames(colData(sce)), value = TRUE, ignore.case = TRUE))])

# Someone with missing age variables
tmp$id[is.na(tmp$age0)]
tmp$id[is.na(tmp$age12)]
tmp$id[is.na(tmp$age24)]
tmp$id[is.na(tmp$EnrAgeMo)]

# EnrAgeMo and age0 are the same variable
stopifnot(all.equal(sce$EnrAgeMo, sce$age0))

# age0 density
p1 <- ggplot(tmp, aes(x = age0, y = ..count.., fill = treatment)) + 
  geom_density(alpha = 1 / 2) +
  scale_fill_manual(values = treatment_colours) +
  theme_minimal_hgrid() +
  ylab("scaled density") +
  guides(fill = guide_legend(override.aes = list(linetype = 0))) +
  theme(axis.line.x = element_blank())

# age12 density
p2 <- ggplot(tmp, aes(x = age12, y = ..count.., fill = treatment)) + 
  geom_density(alpha = 1 / 2) +
  scale_fill_manual(values = treatment_colours) +
  theme_minimal_hgrid() +
  ylab("scaled density") +
  guides(fill = guide_legend(override.aes = list(linetype = 0))) +
  theme(axis.line.x = element_blank())

# age24 density
p3 <- ggplot(tmp, aes(x = age24, y = ..count.., fill = treatment)) + 
  geom_density(alpha = 1 / 2) +
  scale_fill_manual(values = treatment_colours) +
  theme_minimal_hgrid() +
  ylab("scaled density") +
  guides(fill = guide_legend(override.aes = list(linetype = 0))) +
  theme(axis.line.x = element_blank())

# EnrAgeMo density
p4 <- ggplot(tmp, aes(x = EnrAgeMo, y = ..count.., fill = treatment)) + 
  geom_density(alpha = 1 / 2) +
  scale_fill_manual(values = treatment_colours) +
  theme_minimal_hgrid() +
  ylab("scaled density") +
  guides(fill = guide_legend(override.aes = list(linetype = 0))) +
  theme(axis.line.x = element_blank())

((p4| p1) / (p2 | p3)) + plot_layout(guides = "collect")
```

### Gene expression by sex

**TODO**: It's not clear what comparison is being requested from Tom's email.

```{r}
tmp <- unique(x$samples[, c("id", "treatment", "timepoint", "sex")])
head(tmp)
table(tmp$sex, tmp$treatment)
chisq.test(table(tmp$sex, tmp$treatment))
```

The following analysis identifies `sex`-associated genes by treating `treatment`, `timepoint`, `plate_number`, and `sex` as fixed effects and `id` as a random effect.

```{r, results = "hide"}
block <- x$samples$id

design <- model.matrix(
  ~0 + treatment + timepoint + plate_number + sex,
  x$samples)
colnames(design) <- sub(
  "plate\\_number|treatment|timepoint|sex",
  "",
  colnames(design))
contr.matrix <- makeContrasts(
  M_vs_F = M,
  levels = colnames(design))

par(mfrow = c(2, 2))
v <- voom(x, design, plot = TRUE)
cor <- duplicateCorrelation(v, design, block = block)
cor$cor
v <- voom(v, design, plot = TRUE, block = block, correlation = cor$cor)
cor <- duplicateCorrelation(v, design, block = block)
cor$cor
vfit <- lmFit(v, design, block = block, correlation = cor$cor)
vfit <- contrasts.fit(vfit, contrasts = contr.matrix)
efit <- eBayes(vfit)
plotSA(efit, main = "Final model: Mean-variance trend")

dt <- decideTests(efit)
summary(dt)

topTable(efit)
```

## Figure 4

### Gene expression by Haemoglobin at endline

```{r}
tmp <- unique(x$samples[, c("id", "treatment", "hb24")])

# Someone with missing hb24
tmp$id[is.na(tmp$hb24)]

# anaemic is defined as hb < 11
tmp$anaemic <- dplyr::case_when(
  is.na(tmp$hb24) ~ NA,
  tmp$hb24 < 11 ~ TRUE,
  tmp$hb24 >= 11 ~ FALSE)
table(tmp$anaemic)

tapply(tmp$hb24, tmp$anaemic, summary)

# hb24 density
ggplot(tmp, aes(x = hb24, y = ..count.., fill = treatment)) + 
  geom_density(alpha = 1 / 2) +
  scale_fill_manual(values = treatment_colours) +
  theme_minimal_hgrid() +
  ylab("scaled density") +
  guides(fill = guide_legend(override.aes = list(linetype = 0))) +
  theme(axis.line.x = element_blank())
```

#### hb24

```{r}
x_ <- x[, !is.na(x$samples$hb24) & x$samples$timepoint == "endline"]

design <- model.matrix(~0 + treatment + hb24 + plate_number, x_$samples)
colnames(design) <- sub("plate\\_number|treatment", "", colnames(design))

contr.matrix <- makeContrasts(hb24 = hb24, levels = colnames(design))

par(mfrow = c(1, 2))
v <- voom(x_, design, plot = TRUE)
vfit <- lmFit(v, design)
vfit <- contrasts.fit(vfit, contrasts = contr.matrix)
efit <- eBayes(vfit)
plotSA(efit, main = "Final model: Mean-variance trend")

dt <- decideTests(efit)
summary(dt)
```

#### Anaemic24

```{r}
x_ <- x[, x$samples$timepoint == "endline"]
x_$samples$Anaemic24 <- dplyr::case_when(
  is.na(x_$samples$hb24) ~ NA,
  x_$samples$hb24 < 11 ~ TRUE,
  x_$samples$hb24 >= 11 ~ FALSE)
x_ <- x_[, !is.na(x_$samples$Anaemic24)]

design <- model.matrix(~0 + treatment + Anaemic24 + plate_number, x_$samples)
colnames(design) <- sub("plate\\_number|treatment", "", colnames(design))

contr.matrix <- makeContrasts(
  Anaemic24 = Anaemic24TRUE,
  levels = colnames(design))

par(mfrow = c(1, 2))
v <- voom(x_, design, plot = TRUE)
vfit <- lmFit(v, design)
vfit <- contrasts.fit(vfit, contrasts = contr.matrix)
efit <- eBayes(vfit)
plotSA(efit, main = "Final model: Mean-variance trend")

dt <- decideTests(efit)
summary(dt)
```

### Correlation between Recovering from anaemia (Hb < 11 -> Hb > 11) and being in a treatment group.

```{r}
tmp <- unique(x$samples[, c("id", "treatment", "hb0", "hb24")])

# anaemic is defined as hb < 11
tmp$Anaemic0 <- dplyr::case_when(
  is.na(tmp$hb0) ~ NA,
  tmp$hb0 < 11 ~ TRUE,
  tmp$hb0 >= 11 ~ FALSE)
tmp$Anaemic24 <- dplyr::case_when(
  is.na(tmp$hb24) ~ NA,
  tmp$hb24 < 11 ~ TRUE,
  tmp$hb24 >= 11 ~ FALSE)

library(gghighlight)
p <- ggplot(
  tidyr::pivot_longer(tmp, c(hb0, hb24), names_to = "timepoint"),
  aes(x = timepoint, y = value, colour = treatment)) +
  geom_point() +
  geom_line(aes(group = id), alpha = 1 / 3, size = 1) + 
  scale_colour_manual(values = treatment_colours) +
  theme_minimal_hgrid() +
  facet_grid(~treatment) + 
  guides(colour = FALSE)+
  ggtitle("Change in haemoglobin from baseline to endline")
p 

p1 <- p +
  geom_line(aes(group = id), alpha = 1, size = 2) + 
  gghighlight(
    !Anaemic0 & !Anaemic24,
    unhighlighted_params = list(size = 1 / 2, alpha = 1 / 3),
    use_direct_label = FALSE) +
  ggtitle("Children without anaemia (no change)")

p2 <- p +
  geom_line(aes(group = id), alpha = 1, size = 2) + 
  gghighlight(
    Anaemic0 & !Anaemic24,
    unhighlighted_params = list(size = 1 / 2, alpha = 1 / 3),
    use_direct_label = FALSE) +
    ggtitle("Children that recovered from anaemia")

p3 <- p +
  geom_line(aes(group = id), alpha = 1, size = 2) + 
  gghighlight(
    !Anaemic0 & Anaemic24,
    unhighlighted_params = list(size = 1 / 2, alpha = 1 / 3),
    use_direct_label = FALSE) +
    ggtitle("Children that developed anaemia")

p4 <- p +
  geom_line(aes(group = id), alpha = 1, size = 2) + 
  gghighlight(
    Anaemic0 & Anaemic24,
    unhighlighted_params = list(size = 1 / 2, alpha = 1 / 3),
    use_direct_label = FALSE) +
    ggtitle("Children with anaemia (no change)")
    
(p1 / p2) | (p3 / p4)

tmp2 <- tmp[!is.na(tmp$Anaemic0) & !is.na(tmp$Anaemic24), ]

# Anaemia rates between treatments at baseline
table(tmp2$treatment, tmp2$Anaemic0)
chisq.test(table(tmp2$treatment, tmp2$Anaemic0))

# Anaemic rates between treatments at endline
table(tmp2$treatment, tmp2$Anaemic24)
chisq.test(table(tmp2$treatment, tmp2$Anaemic24))

dplyr::count(tmp2, treatment, Anaemic0, Anaemic24)

z <- table(tmp2$Anaemic0, tmp2$Anaemic24, tmp2$treatment)

z[, , "A"]
fisher.test(z[, , "A"])

z[, , "B"]
fisher.test(z[, , "B"])

z[, , "C"]
fisher.test(z[, , "C"])

zz <- table(tmp2$treatment, paste0(tmp2$Anaemic0, "->", tmp2$Anaemic24))
zz
chisq.test(zz)

# Children that recovered from anaemia
chisq.test(zz[, c("TRUE->FALSE", "TRUE->TRUE")])
# Children that developed anaemia
chisq.test(zz[, c("FALSE->FALSE", "FALSE->TRUE")])

zzz <- table(
  tmp2$treatment,
  dplyr::case_when(
    tmp2$Anaemic0 & !tmp2$Anaemic24 ~ "recovered",
    TRUE ~ "not recovered"))
zzz
chisq.test(zzz)
```
